{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "65376208",
   "metadata": {},
   "source": [
    "# Data PreProcessing & Preparation for ML--NO1\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a1d1c93",
   "metadata": {},
   "source": [
    "> ## Import libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d12ca1c2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python: 3.11.0 | packaged by conda-forge | (main, Oct 25 2022, 06:18:27) [GCC 10.4.0]\n",
      "Pandas: 1.5.1\n",
      "Numpy: 1.23.4\n"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "import pandas as pd\n",
    "import os\n",
    "import numpy as np\n",
    "from sklearn.preprocessing import OneHotEncoder, OrdinalEncoder\n",
    "from sklearn.preprocessing import LabelEncoder\n",
    "from tqdm.notebook import tqdm\n",
    "\n",
    "\n",
    "print('python: {}'.format(sys.version))\n",
    "print('Pandas: {}'.format(pd.__version__))\n",
    "print('Numpy: {}'.format(np.__version__))\n",
    "#print('OS: {}'.format(os.__version__))\n",
    "#print('Sklearn: {}'.format(sklearn.__version__))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1168158e",
   "metadata": {},
   "source": [
    "# Read SNPs files pathes for all 3 groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4166e289",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gp1_Ready_SNPs_VCF = '/home/phoebe/phd/ML_csv/GP_1/1_GP1_Ready_SNPs_VCF/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "caf6ccd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gp2_Ready_SNPs_VCF = '/home/phoebe/phd/ML_csv/GP_2/Gp2_SNP_Variant_files/1_Gp2_Ready_SNPs_F1_VCF/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "96a45031",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gp3_Ready_SNPs_VCF = '/home/phoebe/phd/ML_csv/GP_3/Gp3_SNP_Variant_files/1_Gp3_Ready_SNPs_F2_VCF/'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a0c57de",
   "metadata": {},
   "source": [
    "# Read INDELs files pathes for all 3 groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "800e0c7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gp1_Ready_INDELs_VCF = '/home/phoebe/phd/ML_csv/GP_1/1_GP1_Ready_INDELs_VCF/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7be91a0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gp2_Ready_INDELs_VCF = '/home/phoebe/phd/ML_csv/GP_2/Gp2_INDEL_Variant_files/1_Gp2_Ready_INDELs_F1_VCF/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "48ed58f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gp3_Ready_INDELs_VCF = '/home/phoebe/phd/ML_csv/GP_3/Gp3_INDEL_Variant_files/1_Gp3_Ready_INDELs_F2_VCF/'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6c06b33c",
   "metadata": {},
   "source": [
    "# Create directories for merged SNPs + INDELs files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "63d1cdce",
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists ('/home/phoebe/phd/ML_csv/GP_1/Gp1_merged/') == False:\n",
    "    os.makedirs('/home/phoebe/phd/ML_csv/GP_1/Gp1_merged/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e21d0355",
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists ('/home/phoebe/phd/ML_csv/GP_2/Gp2_merged/') == False:\n",
    "    os.makedirs('/home/phoebe/phd/ML_csv/GP_2/Gp2_merged/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "688535ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists ('/home/phoebe/phd/ML_csv/GP_3/Gp3_merged/') == False:\n",
    "    os.makedirs('/home/phoebe/phd/ML_csv/GP_3/Gp3_merged/')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d0b11d17",
   "metadata": {},
   "source": [
    "# Read MERGED (SNPs + INDELs) files pathes for all 3 groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "18214904",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gp1_merged = '/home/phoebe/phd/ML_csv/GP_1/Gp1_merged/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5779640d",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gp2_merged = '/home/phoebe/phd/ML_csv/GP_2/Gp2_merged/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "64ee311e",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gp3_merged = '/home/phoebe/phd/ML_csv/GP_3/Gp3_merged/'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f48f1c78",
   "metadata": {},
   "source": [
    "# First we need to combine SNPs and INDELs of every sample for all groups"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bfe58bcb",
   "metadata": {},
   "source": [
    "# Group_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "63948152",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G1_Sample_014_merged.csv (70, 23)\n",
      "G1_Sample_015_merged.csv (88, 23)\n",
      "G1_Sample_006_merged.csv (72, 23)\n",
      "G1_Sample_024_merged.csv (65, 23)\n",
      "G1_Sample_021_merged.csv (73, 23)\n",
      "G1_Sample_007_merged.csv (67, 23)\n",
      "G1_Sample_013_merged.csv (77, 23)\n",
      "G1_Sample_022_merged.csv (78, 23)\n",
      "G1_Sample_017_merged.csv (76, 23)\n"
     ]
    }
   ],
   "source": [
    "for file in os.listdir(Gp1_Ready_SNPs_VCF):\n",
    "    SNP_file_path = os.path.join(Gp1_Ready_SNPs_VCF,file)\n",
    "    SNP_file_basename = '_'.join(os.path.basename(SNP_file_path).split('_')[0:3])\n",
    "    #print('file_basename: ',SNP_file_basename)\n",
    "    SNP_df = pd.read_csv(SNP_file_path)  \n",
    "    #print(SNP_df)\n",
    "    for f in os.listdir(Gp1_Ready_INDELs_VCF):\n",
    "            INDEL_file_path = os.path.join(Gp1_Ready_INDELs_VCF,f)\n",
    "            INDEL_file_basename = '_'.join(os.path.basename(INDEL_file_path).split('_')[0:3])\n",
    "            if SNP_file_basename == INDEL_file_basename:\n",
    "                INDEL_df = pd.read_csv(INDEL_file_path) \n",
    "                #rint(SNP_file_basename, SNP_df.shape)\n",
    "                #rint (INDEL_file_basename, INDEL_df.shape)\n",
    "                #print(INDEL_df)\n",
    "\n",
    "                final_df = pd.concat([SNP_df,INDEL_df],ignore_index=True, axis=0)\n",
    "                #print (final_df)\n",
    "                final_Out_name = SNP_file_basename + '_merged.csv'\n",
    "                print (final_Out_name, final_df.shape)\n",
    "                final_Out_path = os.path.join(Gp1_merged, final_Out_name)\n",
    "                final_df.to_csv(final_Out_path, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d002207",
   "metadata": {},
   "source": [
    "# Group_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "52956836",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G2_Sample_38 (131308, 24)\n",
      "G2_Sample_38 (17261, 24)\n",
      "G2_Sample_38_merged.csv (148569, 24)\n",
      "G2_Sample_41 (134660, 24)\n",
      "G2_Sample_41 (17516, 24)\n",
      "G2_Sample_41_merged.csv (152176, 24)\n",
      "G2_Sample_35 (133145, 24)\n",
      "G2_Sample_35 (17690, 24)\n",
      "G2_Sample_35_merged.csv (150835, 24)\n",
      "G2_Sample_52 (127372, 24)\n",
      "G2_Sample_52 (16922, 24)\n",
      "G2_Sample_52_merged.csv (144294, 24)\n",
      "G2_Sample_36 (133286, 24)\n",
      "G2_Sample_36 (17974, 24)\n",
      "G2_Sample_36_merged.csv (151260, 24)\n",
      "G2_Sample_11 (120250, 24)\n",
      "G2_Sample_11 (14378, 24)\n",
      "G2_Sample_11_merged.csv (134628, 24)\n",
      "G2_Sample_49 (135108, 24)\n",
      "G2_Sample_49 (19119, 24)\n",
      "G2_Sample_49_merged.csv (154227, 24)\n",
      "G2_Sample_33 (109955, 24)\n",
      "G2_Sample_33 (13287, 24)\n",
      "G2_Sample_33_merged.csv (123242, 24)\n",
      "G2_Sample_12 (116748, 24)\n",
      "G2_Sample_12 (13632, 24)\n",
      "G2_Sample_12_merged.csv (130380, 24)\n",
      "G2_Sample_53 (116229, 24)\n",
      "G2_Sample_53 (14709, 24)\n",
      "G2_Sample_53_merged.csv (130938, 24)\n",
      "G2_Sample_50 (125829, 24)\n",
      "G2_Sample_50 (17570, 24)\n",
      "G2_Sample_50_merged.csv (143399, 24)\n",
      "G2_Sample_47 (127933, 24)\n",
      "G2_Sample_47 (17233, 24)\n",
      "G2_Sample_47_merged.csv (145166, 24)\n",
      "G2_Sample_51 (126171, 24)\n",
      "G2_Sample_51 (16751, 24)\n",
      "G2_Sample_51_merged.csv (142922, 24)\n",
      "G2_Sample_43 (133417, 24)\n",
      "G2_Sample_43 (18300, 24)\n",
      "G2_Sample_43_merged.csv (151717, 24)\n",
      "G2_Sample_42 (130864, 24)\n",
      "G2_Sample_42 (17555, 24)\n",
      "G2_Sample_42_merged.csv (148419, 24)\n",
      "G2_Sample_5 (132989, 24)\n",
      "G2_Sample_5 (17216, 24)\n",
      "G2_Sample_5_merged.csv (150205, 24)\n",
      "G2_Sample_46 (127347, 24)\n",
      "G2_Sample_46 (16355, 24)\n",
      "G2_Sample_46_merged.csv (143702, 24)\n",
      "G2_Sample_34 (114363, 24)\n",
      "G2_Sample_34 (14335, 24)\n",
      "G2_Sample_34_merged.csv (128698, 24)\n",
      "G2_Sample_44 (125095, 24)\n",
      "G2_Sample_44 (16060, 24)\n",
      "G2_Sample_44_merged.csv (141155, 24)\n",
      "G2_Sample_39 (138015, 24)\n",
      "G2_Sample_39 (18508, 24)\n",
      "G2_Sample_39_merged.csv (156523, 24)\n",
      "G2_Sample_48 (134264, 24)\n",
      "G2_Sample_48 (17897, 24)\n",
      "G2_Sample_48_merged.csv (152161, 24)\n",
      "G2_Sample_37 (145896, 24)\n",
      "G2_Sample_37 (20684, 24)\n",
      "G2_Sample_37_merged.csv (166580, 24)\n",
      "G2_Sample_40 (128634, 24)\n",
      "G2_Sample_40 (16860, 24)\n",
      "G2_Sample_40_merged.csv (145494, 24)\n"
     ]
    }
   ],
   "source": [
    "for file in tqdm(os.listdir(Gp2_Ready_SNPs_VCF)):\n",
    "    SNP_file_path = os.path.join(Gp2_Ready_SNPs_VCF,file)\n",
    "    SNP_file_basename = 'G2_' + '_'.join(os.path.basename(SNP_file_path).split('_')[0:2])\n",
    "    #print('file_basename: ',SNP_file_basename)\n",
    "    SNP_df = pd.read_csv(SNP_file_path)  \n",
    "    #print(SNP_df)\n",
    "    for f in os.listdir(Gp2_Ready_INDELs_VCF):\n",
    "            INDEL_file_path = os.path.join(Gp2_Ready_INDELs_VCF,f)\n",
    "            INDEL_file_basename = 'G2_' + '_'.join(os.path.basename(INDEL_file_path).split('_')[0:2])\n",
    "            if SNP_file_basename == INDEL_file_basename:\n",
    "                INDEL_df = pd.read_csv(INDEL_file_path) \n",
    "                #print(SNP_file_basename, SNP_df.shape)\n",
    "                #print (INDEL_file_basename, INDEL_df.shape)\n",
    "                #print(INDEL_df)\n",
    "\n",
    "                final_df = pd.concat([SNP_df,INDEL_df],ignore_index=True, axis=0)\n",
    "                #print (final_df)\n",
    "                final_Out_name = SNP_file_basename + '_merged.csv'\n",
    "                print (final_Out_name, final_df.shape)\n",
    "                final_Out_path = os.path.join(Gp2_merged, final_Out_name)\n",
    "                final_df.to_csv(final_Out_path, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f509c15",
   "metadata": {},
   "source": [
    "# Group_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "78745bde",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "7c24d2a8dc9144469cc40b578c7bc1ff",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/9 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G3_Sample_3 (264814, 40)\n",
      "G3_Sample_3 (52987, 40)\n",
      "G3_Sample_3_merged.csv (317801, 40)\n",
      "G3_Sample_2 (262398, 40)\n",
      "G3_Sample_2 (50308, 40)\n",
      "G3_Sample_2_merged.csv (312706, 40)\n",
      "G3_Sample_5 (375401, 40)\n",
      "G3_Sample_5 (68724, 40)\n",
      "G3_Sample_5_merged.csv (444125, 40)\n",
      "G3_Sample_6 (240513, 40)\n",
      "G3_Sample_6 (42822, 40)\n",
      "G3_Sample_6_merged.csv (283335, 40)\n",
      "G3_Sample_4 (307793, 40)\n",
      "G3_Sample_4 (60653, 40)\n",
      "G3_Sample_4_merged.csv (368446, 40)\n",
      "G3_Sample_8 (327686, 40)\n",
      "G3_Sample_8 (61429, 40)\n",
      "G3_Sample_8_merged.csv (389115, 40)\n",
      "G3_Sample_7 (317504, 40)\n",
      "G3_Sample_7 (57846, 40)\n",
      "G3_Sample_7_merged.csv (375350, 40)\n",
      "G3_Sample_9 (250867, 40)\n",
      "G3_Sample_9 (48741, 40)\n",
      "G3_Sample_9_merged.csv (299608, 40)\n",
      "G3_Sample_1 (268943, 40)\n",
      "G3_Sample_1 (52943, 40)\n",
      "G3_Sample_1_merged.csv (321886, 40)\n"
     ]
    }
   ],
   "source": [
    "for file in tqdmdm(os.listdir(Gp3_Ready_SNPs_VCF)):\n",
    "    SNP_file_path = os.path.join(Gp3_Ready_SNPs_VCF,file)\n",
    "    SNP_file_basename = 'G3_' + '_'.join(os.path.basename(SNP_file_path).split('_')[0:2])\n",
    "    SNP_file_basename = SNP_file_basename.replace('s','S')\n",
    "    #print('file_basename: ',SNP_file_basename)\n",
    "    SNP_df = pd.read_csv(SNP_file_path)  \n",
    "    #print(SNP_df)\n",
    "    for f in os.listdir(Gp3_Ready_INDELs_VCF):\n",
    "            INDEL_file_path = os.path.join(Gp3_Ready_INDELs_VCF,f)\n",
    "            INDEL_file_basename = 'G3_' + '_'.join(os.path.basename(INDEL_file_path).split('_')[0:2])\n",
    "            INDEL_file_basename = INDEL_file_basename.replace('s','S')\n",
    "            if SNP_file_basename == INDEL_file_basename:\n",
    "                INDEL_df = pd.read_csv(INDEL_file_path) \n",
    "                #print(SNP_file_basename, SNP_df.shape)\n",
    "                #print (INDEL_file_basename, INDEL_df.shape)\n",
    "                #print(INDEL_df)\n",
    "\n",
    "                final_df = pd.concat([SNP_df,INDEL_df],ignore_index=True, axis=0)\n",
    "                #print (final_df)\n",
    "                final_Out_name = SNP_file_basename + '_merged.csv'\n",
    "                print (final_Out_name, final_df.shape)\n",
    "                final_Out_path = os.path.join(Gp3_merged, final_Out_name)\n",
    "                final_df.to_csv(final_Out_path, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f1aafbe",
   "metadata": {},
   "source": [
    "> # Genotype Dataframe Without Encoding"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "efd2f2e3",
   "metadata": {},
   "source": [
    "# Group_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "80ff043d",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "5c2291f588f7451b86c6d8604d72072d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/9 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G1_Sample_006_merged (72, 23)\n",
      "GT: \n",
      "0/1    50\n",
      "1/1    22\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    45\n",
      "1/1    22\n",
      "Name: GT, dtype: int64\n",
      "G1_Sample_006_merged (67, 24)\n",
      "G1_Sample_006_merged (1, 67)\n",
      "*********************************************************\n",
      "G1_Sample_007_merged (67, 23)\n",
      "GT: \n",
      "0/1    43\n",
      "1/1    23\n",
      "./.     1\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    40\n",
      "1/1    23\n",
      "Name: GT, dtype: int64\n",
      "G1_Sample_007_merged (63, 24)\n",
      "G1_Sample_007_merged (1, 63)\n",
      "*********************************************************\n",
      "G1_Sample_017_merged (76, 23)\n",
      "GT: \n",
      "0/1    48\n",
      "1/1    27\n",
      "./.     1\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    45\n",
      "1/1    27\n",
      "Name: GT, dtype: int64\n",
      "G1_Sample_017_merged (72, 24)\n",
      "G1_Sample_017_merged (1, 72)\n",
      "*********************************************************\n",
      "G1_Sample_013_merged (77, 23)\n",
      "GT: \n",
      "0/1    44\n",
      "1/1    31\n",
      "1/.     1\n",
      "./.     1\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    40\n",
      "1/1    31\n",
      "Name: GT, dtype: int64\n",
      "G1_Sample_013_merged (71, 24)\n",
      "G1_Sample_013_merged (1, 71)\n",
      "*********************************************************\n",
      "G1_Sample_022_merged (78, 23)\n",
      "GT: \n",
      "0/1    47\n",
      "1/1    29\n",
      "1/2     1\n",
      "./.     1\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    44\n",
      "1/1    30\n",
      "Name: GT, dtype: int64\n",
      "G1_Sample_022_merged (74, 24)\n",
      "G1_Sample_022_merged (1, 74)\n",
      "*********************************************************\n",
      "G1_Sample_024_merged (65, 23)\n",
      "GT: \n",
      "0/1    33\n",
      "1/1    31\n",
      "./.     1\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "1/1    31\n",
      "0/1    30\n",
      "Name: GT, dtype: int64\n",
      "G1_Sample_024_merged (61, 24)\n",
      "G1_Sample_024_merged (1, 61)\n",
      "*********************************************************\n",
      "G1_Sample_021_merged (73, 23)\n",
      "GT: \n",
      "0/1    46\n",
      "1/1    27\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    43\n",
      "1/1    27\n",
      "Name: GT, dtype: int64\n",
      "G1_Sample_021_merged (70, 24)\n",
      "G1_Sample_021_merged (1, 70)\n",
      "*********************************************************\n",
      "G1_Sample_015_merged (88, 23)\n",
      "GT: \n",
      "0/1    57\n",
      "1/1    29\n",
      "1/2     1\n",
      "./.     1\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    53\n",
      "1/1    30\n",
      "Name: GT, dtype: int64\n",
      "G1_Sample_015_merged (83, 24)\n",
      "G1_Sample_015_merged (1, 83)\n",
      "*********************************************************\n",
      "G1_Sample_014_merged (70, 23)\n",
      "GT: \n",
      "0/1    39\n",
      "1/1    29\n",
      "./.     2\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    38\n",
      "1/1    29\n",
      "Name: GT, dtype: int64\n",
      "G1_Sample_014_merged (67, 24)\n",
      "G1_Sample_014_merged (1, 67)\n",
      "*********************************************************\n"
     ]
    }
   ],
   "source": [
    "### Extract Genotype Information for each and every single SNP in every single patient\n",
    "### and store these Informations from each patient in a list to then concatenate them in a single Dataframe \n",
    "\n",
    "G1_list= []\n",
    "for file in tqdm(os.listdir(Gp1_merged)):\n",
    "    file_path = os.path.join(Gp1_merged,file)\n",
    "    Sample_ID = os.path.basename(file).split('.')[0]\n",
    "\n",
    "## Read every single file in a separate df\n",
    "## print df.shape to get idea about ur df dimensions.\n",
    "    df =pd.read_csv(file_path) \n",
    "    print(Sample_ID, df.shape)                         \n",
    "    \n",
    "    ## Before apply ordinal encoding which is additive encoding counting number of minor alleles i.e. \n",
    "    ## genotype \"0/0\" will be coded as '0' caz it contains zero minor alelles\n",
    "    ## genotype \"0/1\" will be coded as '1' caz it contains 1 minor alelles\n",
    "    ## genotype \"1/1\" will be coded as '2' caz it contains 2 minor alelles\n",
    "    ## first replace different genotypes as following:\n",
    "    print ('GT: ')\n",
    "    print(df['GT'].value_counts())\n",
    "    df['GT'] = df['GT'].str.replace('0/2','0/1')\n",
    "    df['GT'] = df['GT'].str.replace('1/2','1/1') \n",
    "    df['GT'] = df['GT'].str.replace('0|1','0/1', regex=False) \n",
    "    df['GT'] = df['GT'].str.replace('1|1','1/1', regex=False)\n",
    "    df['GT'] = df['GT'].str.replace('1|2','1/1', regex=False) \n",
    "    \n",
    "## combine CHROM + POS in one variable\n",
    "## Create new column called SNP_ID and insert it into df\n",
    "    \n",
    "    Head = df[\"CHROM\"] + '-' + df[\"POS\"].astype(str)   \n",
    "    df.insert(loc=1, column= 'SNP_ID', value= Head) \n",
    "    \n",
    "## Now use multiple filters on ur df to include only SNPs that has     \n",
    "## 'FILTER' == PASS and that has \n",
    "## minor allele frequency 'AF' >= 0.01  .....???\n",
    "## 'QUAL' >= 30.0\n",
    "## Also exclude SNPs with missing genotypes like '1/.' or '0/.' 0r './.' \n",
    "## i.e include onle those SNPs with genotype '0/1' or '1/1'\n",
    "## include SNPs related to specific chromosomes that is known to be associated with our disease of interest \n",
    "##'CHROM' == ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "\n",
    "    df = df[(df['FILTER']=='PASS')&(df['QUAL']>=30)& ((df['GT'] == '0/1') | (df['GT'] == '1/1'))] \n",
    "   \n",
    "    chromosome = ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "    df = df[df[\"CHROM\"].isin(chromosome)]\n",
    "    \n",
    "    print ('----------------')\n",
    "    print ('GT: ')\n",
    "    print(df['GT'].value_counts())\n",
    "    \n",
    "    print(Sample_ID, df.shape)          ## Note that the df.shape now has fewer rows for SNPs that was filtered out\n",
    "    df = df[['SNP_ID','GT']]\n",
    "    df = df.rename(columns = {'GT': Sample_ID})\n",
    "    df = df.T  \n",
    "    df.columns= df.iloc[0]\n",
    "    df = df.drop(['SNP_ID'])\n",
    "    print(Sample_ID, df.shape)\n",
    "    print('*********************************************************')\n",
    "    G1_list.append(df)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "622fb98f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 24.2 ms, sys: 3.89 ms, total: 28.1 ms\n",
      "Wall time: 27.2 ms\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>SNP_ID</th>\n",
       "      <th>chr1-156785617</th>\n",
       "      <th>chr1-156846233</th>\n",
       "      <th>chr1-156848995</th>\n",
       "      <th>chr1-201285759</th>\n",
       "      <th>chr1-201285909</th>\n",
       "      <th>chr1-201286678</th>\n",
       "      <th>chr1-201286771</th>\n",
       "      <th>chr1-201288007</th>\n",
       "      <th>chr1-201291066</th>\n",
       "      <th>chr1-201293617</th>\n",
       "      <th>...</th>\n",
       "      <th>chr1-236572539</th>\n",
       "      <th>chr2-109526837</th>\n",
       "      <th>chr2-109527172</th>\n",
       "      <th>chrX-69247672</th>\n",
       "      <th>chr2-167142786</th>\n",
       "      <th>chr2-167142979</th>\n",
       "      <th>chr2-167145106</th>\n",
       "      <th>chr9-133542208</th>\n",
       "      <th>chrX-69253319</th>\n",
       "      <th>chr2-167133479</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G1_Sample_006_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_007_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_017_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_013_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_022_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_024_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_021_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_015_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_014_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9 rows × 191 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "SNP_ID               chr1-156785617 chr1-156846233 chr1-156848995  \\\n",
       "G1_Sample_006_merged            0/1            0/1            0/1   \n",
       "G1_Sample_007_merged            1/1            NaN            0/1   \n",
       "G1_Sample_017_merged            1/1            0/1            NaN   \n",
       "G1_Sample_013_merged            1/1            NaN            0/1   \n",
       "G1_Sample_022_merged            1/1            NaN            1/1   \n",
       "G1_Sample_024_merged            0/1            NaN            1/1   \n",
       "G1_Sample_021_merged            1/1            0/1            0/1   \n",
       "G1_Sample_015_merged            1/1            NaN            1/1   \n",
       "G1_Sample_014_merged            0/1            NaN            0/1   \n",
       "\n",
       "SNP_ID               chr1-201285759 chr1-201285909 chr1-201286678  \\\n",
       "G1_Sample_006_merged            1/1            1/1            0/1   \n",
       "G1_Sample_007_merged            0/1            0/1            NaN   \n",
       "G1_Sample_017_merged            1/1            1/1            NaN   \n",
       "G1_Sample_013_merged            1/1            1/1            NaN   \n",
       "G1_Sample_022_merged            0/1            0/1            NaN   \n",
       "G1_Sample_024_merged            1/1            1/1            NaN   \n",
       "G1_Sample_021_merged            1/1            0/1            0/1   \n",
       "G1_Sample_015_merged            1/1            0/1            0/1   \n",
       "G1_Sample_014_merged            1/1            1/1            NaN   \n",
       "\n",
       "SNP_ID               chr1-201286771 chr1-201288007 chr1-201291066  \\\n",
       "G1_Sample_006_merged            0/1            1/1            0/1   \n",
       "G1_Sample_007_merged            NaN            0/1            0/1   \n",
       "G1_Sample_017_merged            NaN            1/1            1/1   \n",
       "G1_Sample_013_merged            NaN            0/1            0/1   \n",
       "G1_Sample_022_merged            NaN            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN            1/1            1/1   \n",
       "G1_Sample_021_merged            0/1            1/1            0/1   \n",
       "G1_Sample_015_merged            0/1            0/1            NaN   \n",
       "G1_Sample_014_merged            NaN            0/1            0/1   \n",
       "\n",
       "SNP_ID               chr1-201293617  ... chr1-236572539 chr2-109526837  \\\n",
       "G1_Sample_006_merged            0/1  ...            NaN            NaN   \n",
       "G1_Sample_007_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_017_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_013_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_022_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_021_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_015_merged            NaN  ...            1/1            0/1   \n",
       "G1_Sample_014_merged            NaN  ...            NaN            NaN   \n",
       "\n",
       "SNP_ID               chr2-109527172 chrX-69247672 chr2-167142786  \\\n",
       "G1_Sample_006_merged            NaN           NaN            NaN   \n",
       "G1_Sample_007_merged            NaN           NaN            NaN   \n",
       "G1_Sample_017_merged            NaN           NaN            NaN   \n",
       "G1_Sample_013_merged            NaN           NaN            NaN   \n",
       "G1_Sample_022_merged            NaN           NaN            NaN   \n",
       "G1_Sample_024_merged            NaN           NaN            NaN   \n",
       "G1_Sample_021_merged            NaN           NaN            NaN   \n",
       "G1_Sample_015_merged            0/1           0/1            NaN   \n",
       "G1_Sample_014_merged            NaN           NaN            0/1   \n",
       "\n",
       "SNP_ID               chr2-167142979 chr2-167145106 chr9-133542208  \\\n",
       "G1_Sample_006_merged            NaN            NaN            NaN   \n",
       "G1_Sample_007_merged            NaN            NaN            NaN   \n",
       "G1_Sample_017_merged            NaN            NaN            NaN   \n",
       "G1_Sample_013_merged            NaN            NaN            NaN   \n",
       "G1_Sample_022_merged            NaN            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN            NaN            NaN   \n",
       "G1_Sample_021_merged            NaN            NaN            NaN   \n",
       "G1_Sample_015_merged            NaN            NaN            NaN   \n",
       "G1_Sample_014_merged            0/1            0/1            0/1   \n",
       "\n",
       "SNP_ID               chrX-69253319 chr2-167133479  \n",
       "G1_Sample_006_merged           NaN            NaN  \n",
       "G1_Sample_007_merged           NaN            NaN  \n",
       "G1_Sample_017_merged           NaN            NaN  \n",
       "G1_Sample_013_merged           NaN            NaN  \n",
       "G1_Sample_022_merged           NaN            NaN  \n",
       "G1_Sample_024_merged           NaN            NaN  \n",
       "G1_Sample_021_merged           NaN            NaN  \n",
       "G1_Sample_015_merged           NaN            NaN  \n",
       "G1_Sample_014_merged           1/1            0/1  \n",
       "\n",
       "[9 rows x 191 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "## Dataframe containing all Genotypes of All SNPs in all samples\n",
    "\n",
    "G1_raw_genotype = pd.concat(G1_list,axis=0)\n",
    "G1_raw_genotype"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3219bd4f",
   "metadata": {},
   "source": [
    "# Group_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "02fa50ef",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "32599c07d48d433a9742d7e974a493c7",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/23 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G2_Sample_50_merged (143399, 24)\n",
      "GT: \n",
      "0/1    89726\n",
      "1/1    52282\n",
      "1/2     1346\n",
      "0/0       45\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    27011\n",
      "1/1    16844\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_50_merged (43855, 25)\n",
      "G2_Sample_50_merged (1, 43855)\n",
      "*********************************************************\n",
      "G2_Sample_43_merged (151717, 24)\n",
      "GT: \n",
      "0/1    95614\n",
      "1/1    54621\n",
      "1/2     1427\n",
      "0/0       51\n",
      "0/2        4\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    29310\n",
      "1/1    16346\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_43_merged (45656, 25)\n",
      "G2_Sample_43_merged (1, 45656)\n",
      "*********************************************************\n",
      "G2_Sample_48_merged (152161, 24)\n",
      "GT: \n",
      "0/1    97203\n",
      "1/1    53577\n",
      "1/2     1302\n",
      "0/0       74\n",
      "0/2        5\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    29264\n",
      "1/1    16885\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_48_merged (46149, 25)\n",
      "G2_Sample_48_merged (1, 46149)\n",
      "*********************************************************\n",
      "G2_Sample_33_merged (123242, 24)\n",
      "GT: \n",
      "0/1    78288\n",
      "1/1    43947\n",
      "1/2      964\n",
      "0/0       40\n",
      "0/2        3\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    22640\n",
      "1/1    14041\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_33_merged (36681, 25)\n",
      "G2_Sample_33_merged (1, 36681)\n",
      "*********************************************************\n",
      "G2_Sample_42_merged (148419, 24)\n",
      "GT: \n",
      "0/1    95043\n",
      "1/1    52021\n",
      "1/2     1295\n",
      "0/0       57\n",
      "0/2        3\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    27767\n",
      "1/1    16586\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_42_merged (44353, 25)\n",
      "G2_Sample_42_merged (1, 44353)\n",
      "*********************************************************\n",
      "G2_Sample_44_merged (141155, 24)\n",
      "GT: \n",
      "0/1    88255\n",
      "1/1    51676\n",
      "1/2     1149\n",
      "0/0       74\n",
      "0/2        1\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    27221\n",
      "1/1    16088\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_44_merged (43309, 25)\n",
      "G2_Sample_44_merged (1, 43309)\n",
      "*********************************************************\n",
      "G2_Sample_5_merged (150205, 24)\n",
      "GT: \n",
      "0/1    91076\n",
      "1/1    57807\n",
      "1/2     1250\n",
      "0/0       68\n",
      "0/2        4\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    29241\n",
      "1/1    17380\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_5_merged (46621, 25)\n",
      "G2_Sample_5_merged (1, 46621)\n",
      "*********************************************************\n",
      "G2_Sample_39_merged (156523, 24)\n",
      "GT: \n",
      "0/1    102359\n",
      "1/1     52708\n",
      "1/2      1399\n",
      "0/0        55\n",
      "0/2         2\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    29454\n",
      "1/1    16952\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_39_merged (46406, 25)\n",
      "G2_Sample_39_merged (1, 46406)\n",
      "*********************************************************\n",
      "G2_Sample_41_merged (152176, 24)\n",
      "GT: \n",
      "0/1    97441\n",
      "1/1    53329\n",
      "1/2     1328\n",
      "0/0       75\n",
      "0/2        3\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    30342\n",
      "1/1    16092\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_41_merged (46434, 25)\n",
      "G2_Sample_41_merged (1, 46434)\n",
      "*********************************************************\n",
      "G2_Sample_49_merged (154227, 24)\n",
      "GT: \n",
      "0/1    98448\n",
      "1/1    54196\n",
      "1/2     1506\n",
      "0/0       71\n",
      "0/2        6\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    30932\n",
      "1/1    16754\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_49_merged (47686, 25)\n",
      "G2_Sample_49_merged (1, 47686)\n",
      "*********************************************************\n",
      "G2_Sample_51_merged (142922, 24)\n",
      "GT: \n",
      "0/1    91253\n",
      "1/1    50318\n",
      "1/2     1303\n",
      "0/0       41\n",
      "0/2        7\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    27255\n",
      "1/1    15967\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_51_merged (43222, 25)\n",
      "G2_Sample_51_merged (1, 43222)\n",
      "*********************************************************\n",
      "G2_Sample_53_merged (130938, 24)\n",
      "GT: \n",
      "0/1    83854\n",
      "1/1    45922\n",
      "1/2     1128\n",
      "0/0       32\n",
      "0/2        2\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    25124\n",
      "1/1    14790\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_53_merged (39914, 25)\n",
      "G2_Sample_53_merged (1, 39914)\n",
      "*********************************************************\n",
      "G2_Sample_40_merged (145494, 24)\n",
      "GT: \n",
      "0/1    94832\n",
      "1/1    49278\n",
      "1/2     1293\n",
      "0/0       88\n",
      "0/2        3\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    27954\n",
      "1/1    15447\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_40_merged (43401, 25)\n",
      "G2_Sample_40_merged (1, 43401)\n",
      "*********************************************************\n",
      "G2_Sample_36_merged (151260, 24)\n",
      "GT: \n",
      "0/1    100186\n",
      "1/1     49646\n",
      "1/2      1372\n",
      "0/0        52\n",
      "0/2         4\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    30279\n",
      "1/1    15424\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_36_merged (45703, 25)\n",
      "G2_Sample_36_merged (1, 45703)\n",
      "*********************************************************\n",
      "G2_Sample_47_merged (145166, 24)\n",
      "GT: \n",
      "0/1    96296\n",
      "1/1    47518\n",
      "1/2     1290\n",
      "0/0       58\n",
      "0/2        4\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    28086\n",
      "1/1    15135\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_47_merged (43221, 25)\n",
      "G2_Sample_47_merged (1, 43221)\n",
      "*********************************************************\n",
      "G2_Sample_38_merged (148569, 24)\n",
      "GT: \n",
      "0/1    97131\n",
      "1/1    50067\n",
      "1/2     1301\n",
      "0/0       68\n",
      "0/2        2\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    28467\n",
      "1/1    15921\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_38_merged (44388, 25)\n",
      "G2_Sample_38_merged (1, 44388)\n",
      "*********************************************************\n",
      "G2_Sample_52_merged (144294, 24)\n",
      "GT: \n",
      "0/1    93465\n",
      "1/1    49504\n",
      "1/2     1268\n",
      "0/0       56\n",
      "0/2        1\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    28218\n",
      "1/1    15052\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_52_merged (43270, 25)\n",
      "G2_Sample_52_merged (1, 43270)\n",
      "*********************************************************\n",
      "G2_Sample_37_merged (166580, 24)\n",
      "GT: \n",
      "0/1    108165\n",
      "1/1     56635\n",
      "1/2      1710\n",
      "0/0        68\n",
      "0/2         2\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    30990\n",
      "1/1    17882\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_37_merged (48872, 25)\n",
      "G2_Sample_37_merged (1, 48872)\n",
      "*********************************************************\n",
      "G2_Sample_11_merged (134628, 24)\n",
      "GT: \n",
      "0/1    90028\n",
      "1/1    43519\n",
      "1/2     1022\n",
      "0/0       53\n",
      "0/2        6\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    27669\n",
      "1/1    13490\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_11_merged (41159, 25)\n",
      "G2_Sample_11_merged (1, 41159)\n",
      "*********************************************************\n",
      "G2_Sample_12_merged (130380, 24)\n",
      "GT: \n",
      "0/1    83481\n",
      "1/1    45904\n",
      "1/2      942\n",
      "0/0       51\n",
      "0/2        2\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    24825\n",
      "1/1    14346\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_12_merged (39171, 25)\n",
      "G2_Sample_12_merged (1, 39171)\n",
      "*********************************************************\n",
      "G2_Sample_35_merged (150835, 24)\n",
      "GT: \n",
      "0/1    93687\n",
      "1/1    55659\n",
      "1/2     1401\n",
      "0/0       82\n",
      "0/2        6\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    28235\n",
      "1/1    16788\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_35_merged (45023, 25)\n",
      "G2_Sample_35_merged (1, 45023)\n",
      "*********************************************************\n",
      "G2_Sample_34_merged (128698, 24)\n",
      "GT: \n",
      "0/1    78945\n",
      "1/1    48725\n",
      "1/2      973\n",
      "0/0       52\n",
      "0/2        3\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    22833\n",
      "1/1    15611\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_34_merged (38444, 25)\n",
      "G2_Sample_34_merged (1, 38444)\n",
      "*********************************************************\n",
      "G2_Sample_46_merged (143702, 24)\n",
      "GT: \n",
      "0/1    84601\n",
      "1/1    57820\n",
      "1/2     1223\n",
      "0/0       56\n",
      "0/2        2\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "0/1    27425\n",
      "1/1    17286\n",
      "Name: GT, dtype: int64\n",
      "G2_Sample_46_merged (44711, 25)\n",
      "G2_Sample_46_merged (1, 44711)\n",
      "*********************************************************\n"
     ]
    }
   ],
   "source": [
    "### Extract Genotype Information for each and every single SNP in every single patient\n",
    "### and store these Informations from each patient in a list to then concatenate them in a single Dataframe \n",
    "\n",
    "\n",
    "G2_list= []\n",
    "for file in tqdm(os.listdir(Gp2_merged)):\n",
    "    file_path = os.path.join(Gp2_merged,file)\n",
    "    Sample_ID = os.path.basename(file).split('.')[0]\n",
    "\n",
    "## Read every single file in a separate df\n",
    "## print df.shape to get idea about ur df dimensions.\n",
    "    df =pd.read_csv(file_path) \n",
    "    print(Sample_ID, df.shape)                         \n",
    "    \n",
    "    ## Before apply ordinal encoding which is additive encoding counting number of minor alleles i.e. \n",
    "    ## genotype \"0/0\" will be coded as '0' caz it contains zero minor alelles\n",
    "    ## genotype \"0/1\" will be coded as '1' caz it contains 1 minor alelles\n",
    "    ## genotype \"1/1\" will be coded as '2' caz it contains 2 minor alelles\n",
    "    ## first replace different genotypes as following:\n",
    "    print ('GT: ')\n",
    "    print(df['GT'].value_counts())\n",
    "    df['GT'] = df['GT'].str.replace('0/2','0/1')\n",
    "    df['GT'] = df['GT'].str.replace('1/2','1/1') \n",
    "    df['GT'] = df['GT'].str.replace('0|1','0/1', regex=False) \n",
    "    df['GT'] = df['GT'].str.replace('1|1','1/1', regex=False)\n",
    "    df['GT'] = df['GT'].str.replace('1|2','1/1', regex=False) \n",
    "    \n",
    "## combine CHROM + POS in one variable\n",
    "## Create new column called SNP_ID and insert it into df\n",
    "    \n",
    "    Head = df[\"CHROM\"] + '-' + df[\"POS\"].astype(str)   \n",
    "    df.insert(loc=1, column= 'SNP_ID', value= Head) \n",
    "    \n",
    "## Now use multiple filters on ur df to include only SNPs that has     \n",
    "## 'FILTER' == PASS and that has \n",
    "## minor allele frequency 'AF' >= 0.01  .....???\n",
    "## 'QUAL' >= 30.0\n",
    "## Also exclude SNPs with missing genotypes like '1/.' or '0/.' 0r './.' \n",
    "## i.e include onle those SNPs with genotype '0/1' or '1/1'\n",
    "## include SNPs related to specific chromosomes that is known to be associated with our disease of interest \n",
    "##'CHROM' == ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "\n",
    "    df = df[(df['FILTER']=='PASS')&(df['QUAL']>=30)& ((df['GT'] == '0/1') | (df['GT'] == '1/1'))] \n",
    "   \n",
    "    chromosome = ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "    df = df[df[\"CHROM\"].isin(chromosome)]\n",
    "    \n",
    "    print ('----------------')\n",
    "    print ('GT: ')\n",
    "    print(df['GT'].value_counts())\n",
    "    \n",
    "    print(Sample_ID, df.shape)          ## Note that the df.shape now has fewer rows for SNPs that was filtered out\n",
    "    df = df[['SNP_ID','GT']]\n",
    "    df = df.rename(columns = {'GT': Sample_ID})\n",
    "    df = df.T  \n",
    "    df.columns= df.iloc[0]\n",
    "    df = df.drop(['SNP_ID'])\n",
    "    print(Sample_ID, df.shape)\n",
    "    print('*********************************************************')\n",
    "    G2_list.append(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "fc119907",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 4s, sys: 918 ms, total: 1min 4s\n",
      "Wall time: 1min 4s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>SNP_ID</th>\n",
       "      <th>chr1-14653</th>\n",
       "      <th>chr1-14671</th>\n",
       "      <th>chr1-14907</th>\n",
       "      <th>chr1-14930</th>\n",
       "      <th>chr1-15211</th>\n",
       "      <th>chr1-16534</th>\n",
       "      <th>chr1-16571</th>\n",
       "      <th>chr1-16719</th>\n",
       "      <th>chr1-16841</th>\n",
       "      <th>chr1-17375</th>\n",
       "      <th>...</th>\n",
       "      <th>chrX-114658525</th>\n",
       "      <th>chrX-119419850</th>\n",
       "      <th>chrX-138698651</th>\n",
       "      <th>chrX-140802529</th>\n",
       "      <th>chrX-151290013</th>\n",
       "      <th>chrX-152928065</th>\n",
       "      <th>chrX-152939900</th>\n",
       "      <th>chrX-152980909</th>\n",
       "      <th>chrX-153551757</th>\n",
       "      <th>chrX-153664247</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G2_Sample_50_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_43_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_48_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_33_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_42_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_44_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_5_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_39_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_41_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_49_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_51_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_53_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_40_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_36_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_47_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_38_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_52_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_37_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_11_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_12_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_35_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_34_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_46_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23 rows × 169532 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "SNP_ID              chr1-14653 chr1-14671 chr1-14907 chr1-14930 chr1-15211  \\\n",
       "G2_Sample_50_merged        0/1        0/1        0/1        0/1        0/1   \n",
       "G2_Sample_43_merged        0/1        NaN        0/1        0/1        0/1   \n",
       "G2_Sample_48_merged        0/1        NaN        0/1        NaN        0/1   \n",
       "G2_Sample_33_merged        0/1        NaN        0/1        NaN        0/1   \n",
       "G2_Sample_42_merged        NaN        NaN        0/1        0/1        0/1   \n",
       "G2_Sample_44_merged        0/1        NaN        0/1        NaN        0/1   \n",
       "G2_Sample_5_merged         0/1        NaN        0/1        0/1        1/1   \n",
       "G2_Sample_39_merged        0/1        NaN        0/1        NaN        1/1   \n",
       "G2_Sample_41_merged        0/1        NaN        0/1        0/1        NaN   \n",
       "G2_Sample_49_merged        0/1        NaN        0/1        NaN        1/1   \n",
       "G2_Sample_51_merged        NaN        NaN        0/1        0/1        NaN   \n",
       "G2_Sample_53_merged        1/1        NaN        0/1        0/1        NaN   \n",
       "G2_Sample_40_merged        0/1        NaN        0/1        0/1        0/1   \n",
       "G2_Sample_36_merged        0/1        NaN        0/1        0/1        1/1   \n",
       "G2_Sample_47_merged        NaN        NaN        0/1        0/1        0/1   \n",
       "G2_Sample_38_merged        0/1        NaN        1/1        NaN        1/1   \n",
       "G2_Sample_52_merged        0/1        NaN        0/1        NaN        1/1   \n",
       "G2_Sample_37_merged        0/1        NaN        0/1        NaN        1/1   \n",
       "G2_Sample_11_merged        NaN        NaN        0/1        0/1        0/1   \n",
       "G2_Sample_12_merged        0/1        NaN        0/1        NaN        0/1   \n",
       "G2_Sample_35_merged        0/1        NaN        0/1        0/1        0/1   \n",
       "G2_Sample_34_merged        0/1        NaN        0/1        0/1        1/1   \n",
       "G2_Sample_46_merged        NaN        NaN        0/1        NaN        0/1   \n",
       "\n",
       "SNP_ID              chr1-16534 chr1-16571 chr1-16719 chr1-16841 chr1-17375  \\\n",
       "G2_Sample_50_merged        0/1        0/1        0/1        0/1        0/1   \n",
       "G2_Sample_43_merged        0/1        1/1        NaN        NaN        NaN   \n",
       "G2_Sample_48_merged        0/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_33_merged        0/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_42_merged        0/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_44_merged        0/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_5_merged         0/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_39_merged        0/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_41_merged        0/1        0/1        NaN        0/1        NaN   \n",
       "G2_Sample_49_merged        NaN        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_51_merged        NaN        NaN        NaN        NaN        NaN   \n",
       "G2_Sample_53_merged        0/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_40_merged        0/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_36_merged        0/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_47_merged        0/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_38_merged        0/1        1/1        NaN        NaN        NaN   \n",
       "G2_Sample_52_merged        1/1        1/1        NaN        NaN        NaN   \n",
       "G2_Sample_37_merged        1/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_11_merged        NaN        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_12_merged        NaN        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_35_merged        0/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_34_merged        0/1        0/1        NaN        NaN        NaN   \n",
       "G2_Sample_46_merged        0/1        0/1        NaN        NaN        NaN   \n",
       "\n",
       "SNP_ID               ... chrX-114658525 chrX-119419850 chrX-138698651  \\\n",
       "G2_Sample_50_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_43_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_48_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_33_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_42_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_44_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_5_merged   ...            NaN            NaN            NaN   \n",
       "G2_Sample_39_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_41_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_49_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_51_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_53_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_40_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_36_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_47_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_38_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_52_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_37_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_11_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_12_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_35_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_34_merged  ...            NaN            NaN            NaN   \n",
       "G2_Sample_46_merged  ...            1/1            1/1            1/1   \n",
       "\n",
       "SNP_ID              chrX-140802529 chrX-151290013 chrX-152928065  \\\n",
       "G2_Sample_50_merged            NaN            NaN            NaN   \n",
       "G2_Sample_43_merged            NaN            NaN            NaN   \n",
       "G2_Sample_48_merged            NaN            NaN            NaN   \n",
       "G2_Sample_33_merged            NaN            NaN            NaN   \n",
       "G2_Sample_42_merged            NaN            NaN            NaN   \n",
       "G2_Sample_44_merged            NaN            NaN            NaN   \n",
       "G2_Sample_5_merged             NaN            NaN            NaN   \n",
       "G2_Sample_39_merged            NaN            NaN            NaN   \n",
       "G2_Sample_41_merged            NaN            NaN            NaN   \n",
       "G2_Sample_49_merged            NaN            NaN            NaN   \n",
       "G2_Sample_51_merged            NaN            NaN            NaN   \n",
       "G2_Sample_53_merged            NaN            NaN            NaN   \n",
       "G2_Sample_40_merged            NaN            NaN            NaN   \n",
       "G2_Sample_36_merged            NaN            NaN            NaN   \n",
       "G2_Sample_47_merged            NaN            NaN            NaN   \n",
       "G2_Sample_38_merged            NaN            NaN            NaN   \n",
       "G2_Sample_52_merged            NaN            NaN            NaN   \n",
       "G2_Sample_37_merged            NaN            NaN            NaN   \n",
       "G2_Sample_11_merged            NaN            NaN            NaN   \n",
       "G2_Sample_12_merged            NaN            NaN            NaN   \n",
       "G2_Sample_35_merged            NaN            NaN            NaN   \n",
       "G2_Sample_34_merged            NaN            NaN            NaN   \n",
       "G2_Sample_46_merged            1/1            1/1            1/1   \n",
       "\n",
       "SNP_ID              chrX-152939900 chrX-152980909 chrX-153551757  \\\n",
       "G2_Sample_50_merged            NaN            NaN            NaN   \n",
       "G2_Sample_43_merged            NaN            NaN            NaN   \n",
       "G2_Sample_48_merged            NaN            NaN            NaN   \n",
       "G2_Sample_33_merged            NaN            NaN            NaN   \n",
       "G2_Sample_42_merged            NaN            NaN            NaN   \n",
       "G2_Sample_44_merged            NaN            NaN            NaN   \n",
       "G2_Sample_5_merged             NaN            NaN            NaN   \n",
       "G2_Sample_39_merged            NaN            NaN            NaN   \n",
       "G2_Sample_41_merged            NaN            NaN            NaN   \n",
       "G2_Sample_49_merged            NaN            NaN            NaN   \n",
       "G2_Sample_51_merged            NaN            NaN            NaN   \n",
       "G2_Sample_53_merged            NaN            NaN            NaN   \n",
       "G2_Sample_40_merged            NaN            NaN            NaN   \n",
       "G2_Sample_36_merged            NaN            NaN            NaN   \n",
       "G2_Sample_47_merged            NaN            NaN            NaN   \n",
       "G2_Sample_38_merged            NaN            NaN            NaN   \n",
       "G2_Sample_52_merged            NaN            NaN            NaN   \n",
       "G2_Sample_37_merged            NaN            NaN            NaN   \n",
       "G2_Sample_11_merged            NaN            NaN            NaN   \n",
       "G2_Sample_12_merged            NaN            NaN            NaN   \n",
       "G2_Sample_35_merged            NaN            NaN            NaN   \n",
       "G2_Sample_34_merged            NaN            NaN            NaN   \n",
       "G2_Sample_46_merged            1/1            1/1            1/1   \n",
       "\n",
       "SNP_ID              chrX-153664247  \n",
       "G2_Sample_50_merged            NaN  \n",
       "G2_Sample_43_merged            NaN  \n",
       "G2_Sample_48_merged            NaN  \n",
       "G2_Sample_33_merged            NaN  \n",
       "G2_Sample_42_merged            NaN  \n",
       "G2_Sample_44_merged            NaN  \n",
       "G2_Sample_5_merged             NaN  \n",
       "G2_Sample_39_merged            NaN  \n",
       "G2_Sample_41_merged            NaN  \n",
       "G2_Sample_49_merged            NaN  \n",
       "G2_Sample_51_merged            NaN  \n",
       "G2_Sample_53_merged            NaN  \n",
       "G2_Sample_40_merged            NaN  \n",
       "G2_Sample_36_merged            NaN  \n",
       "G2_Sample_47_merged            NaN  \n",
       "G2_Sample_38_merged            NaN  \n",
       "G2_Sample_52_merged            NaN  \n",
       "G2_Sample_37_merged            NaN  \n",
       "G2_Sample_11_merged            NaN  \n",
       "G2_Sample_12_merged            NaN  \n",
       "G2_Sample_35_merged            NaN  \n",
       "G2_Sample_34_merged            NaN  \n",
       "G2_Sample_46_merged            1/1  \n",
       "\n",
       "[23 rows x 169532 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "## Dataframe containing all Genotypes of All SNPs in all samples\n",
    "\n",
    "G2_raw_genotype = pd.concat(G2_list,axis=0)\n",
    "G2_raw_genotype"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ad893997",
   "metadata": {},
   "source": [
    "# Group_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "6e0471d3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "22e2ad74abc9438fad860623a4fdb4b0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/9 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G3_Sample_4_merged (368446, 40)\n",
      "GT: \n",
      "1/1    187366\n",
      "0/1    118478\n",
      "1|1     32394\n",
      "0|1     27925\n",
      "1/2      2066\n",
      "1|2       217\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "1/1    67264\n",
      "0/1    44190\n",
      "Name: GT, dtype: int64\n",
      "G3_Sample_4_merged (111454, 41)\n",
      "G3_Sample_4_merged (1, 111454)\n",
      "*********************************************************\n",
      "G3_Sample_3_merged (317801, 40)\n",
      "GT: \n",
      "1/1    147959\n",
      "0/1    116208\n",
      "0|1     25777\n",
      "1|1     25727\n",
      "1/2      1928\n",
      "1|2       202\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "1/1    53202\n",
      "0/1    42496\n",
      "Name: GT, dtype: int64\n",
      "G3_Sample_3_merged (95698, 41)\n",
      "G3_Sample_3_merged (1, 95698)\n",
      "*********************************************************\n",
      "G3_Sample_2_merged (312706, 40)\n",
      "GT: \n",
      "1/1    148475\n",
      "0/1    111150\n",
      "1|1     25849\n",
      "0|1     25409\n",
      "1/2      1648\n",
      "1|2       175\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "1/1    52635\n",
      "0/1    39412\n",
      "Name: GT, dtype: int64\n",
      "G3_Sample_2_merged (92047, 41)\n",
      "G3_Sample_2_merged (1, 92047)\n",
      "*********************************************************\n",
      "G3_Sample_1_merged (321886, 40)\n",
      "GT: \n",
      "1/1    147594\n",
      "0/1    116350\n",
      "1|1     28690\n",
      "0|1     27160\n",
      "1/2      1884\n",
      "1|2       208\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "1/1    52807\n",
      "0/1    43358\n",
      "Name: GT, dtype: int64\n",
      "G3_Sample_1_merged (96165, 41)\n",
      "G3_Sample_1_merged (1, 96165)\n",
      "*********************************************************\n",
      "G3_Sample_5_merged (444125, 40)\n",
      "GT: \n",
      "1/1    245746\n",
      "0/1    124198\n",
      "1|1     40799\n",
      "0|1     31134\n",
      "1/2      2025\n",
      "1|2       223\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "1/1    86945\n",
      "0/1    42961\n",
      "Name: GT, dtype: int64\n",
      "G3_Sample_5_merged (129906, 41)\n",
      "G3_Sample_5_merged (1, 129906)\n",
      "*********************************************************\n",
      "G3_Sample_7_merged (375350, 40)\n",
      "GT: \n",
      "1/1    196626\n",
      "0/1    112746\n",
      "1|1     36295\n",
      "0|1     27888\n",
      "1/2      1611\n",
      "1|2       184\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "1/1    71472\n",
      "0/1    41848\n",
      "Name: GT, dtype: int64\n",
      "G3_Sample_7_merged (113320, 41)\n",
      "G3_Sample_7_merged (1, 113320)\n",
      "*********************************************************\n",
      "G3_Sample_9_merged (299608, 40)\n",
      "GT: \n",
      "1/1    154668\n",
      "0/1     92996\n",
      "1|1     27814\n",
      "0|1     22536\n",
      "1/2      1417\n",
      "1|2       177\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "1/1    55499\n",
      "0/1    32882\n",
      "Name: GT, dtype: int64\n",
      "G3_Sample_9_merged (88381, 41)\n",
      "G3_Sample_9_merged (1, 88381)\n",
      "*********************************************************\n",
      "G3_Sample_8_merged (389115, 40)\n",
      "GT: \n",
      "1/1    214697\n",
      "0/1    109543\n",
      "1|1     35706\n",
      "0|1     27146\n",
      "1/2      1850\n",
      "1|2       173\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "1/1    75535\n",
      "0/1    41767\n",
      "Name: GT, dtype: int64\n",
      "G3_Sample_8_merged (117302, 41)\n",
      "G3_Sample_8_merged (1, 117302)\n",
      "*********************************************************\n",
      "G3_Sample_6_merged (283335, 40)\n",
      "GT: \n",
      "1/1    130497\n",
      "0/1     90429\n",
      "1|1     33218\n",
      "0|1     27750\n",
      "1/2      1288\n",
      "1|2       153\n",
      "Name: GT, dtype: int64\n",
      "----------------\n",
      "GT: \n",
      "1/1    49160\n",
      "0/1    33751\n",
      "Name: GT, dtype: int64\n",
      "G3_Sample_6_merged (82911, 41)\n",
      "G3_Sample_6_merged (1, 82911)\n",
      "*********************************************************\n"
     ]
    }
   ],
   "source": [
    "### Extract Genotype Information for each and every single SNP in every single patient\n",
    "### and store these Informations from each patient in a list to then concatenate them in a single Dataframe \n",
    "\n",
    "\n",
    "G3_list= []\n",
    "for file in tqdm(os.listdir(Gp3_merged)):\n",
    "    file_path = os.path.join(Gp3_merged,file)\n",
    "    Sample_ID = os.path.basename(file).split('.')[0]\n",
    "\n",
    "## Read every single file in a separate df\n",
    "## print df.shape to get idea about ur df dimensions.\n",
    "    df =pd.read_csv(file_path) \n",
    "    print(Sample_ID, df.shape)                         \n",
    "    \n",
    "    ## Before apply ordinal encoding which is additive encoding counting number of minor alleles i.e. \n",
    "    ## genotype \"0/0\" will be coded as '0' caz it contains zero minor alelles\n",
    "    ## genotype \"0/1\" will be coded as '1' caz it contains 1 minor alelles\n",
    "    ## genotype \"1/1\" will be coded as '2' caz it contains 2 minor alelles\n",
    "    ## first replace different genotypes as following:\n",
    "    print ('GT: ')\n",
    "    print(df['GT'].value_counts())\n",
    "    df['GT'] = df['GT'].str.replace('0/2','0/1')\n",
    "    df['GT'] = df['GT'].str.replace('1/2','1/1') \n",
    "    df['GT'] = df['GT'].str.replace('0|1','0/1', regex=False) \n",
    "    df['GT'] = df['GT'].str.replace('1|1','1/1', regex=False)\n",
    "    df['GT'] = df['GT'].str.replace('1|2','1/1', regex=False) \n",
    "    \n",
    "## combine CHROM + POS in one variable\n",
    "## Create new column called SNP_ID and insert it into df\n",
    "    \n",
    "    Head = df[\"CHROM\"] + '-' + df[\"POS\"].astype(str)   \n",
    "    df.insert(loc=1, column= 'SNP_ID', value= Head) \n",
    "    \n",
    "## Now use multiple filters on ur df to include only SNPs that has     \n",
    "## 'FILTER' == PASS and that has \n",
    "## minor allele frequency 'AF' >= 0.01  .....???\n",
    "## 'QUAL' >= 30.0\n",
    "## Also exclude SNPs with missing genotypes like '1/.' or '0/.' 0r './.' \n",
    "## i.e include onle those SNPs with genotype '0/1' or '1/1'\n",
    "## include SNPs related to specific chromosomes that is known to be associated with our disease of interest \n",
    "##'CHROM' == ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "\n",
    "    df = df[(df['FILTER']=='PASS')&(df['QUAL']>=30)& ((df['GT'] == '0/1') | (df['GT'] == '1/1'))] \n",
    "   \n",
    "    chromosome = ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "    df = df[df[\"CHROM\"].isin(chromosome)]\n",
    "    \n",
    "    print ('----------------')\n",
    "    print ('GT: ')\n",
    "    print(df['GT'].value_counts())\n",
    "    \n",
    "    print(Sample_ID, df.shape)          ## Note that the df.shape now has fewer rows for SNPs that was filtered out\n",
    "    df = df[['SNP_ID','GT']]\n",
    "    df = df.rename(columns = {'GT': Sample_ID})\n",
    "    df = df.T  \n",
    "    df.columns= df.iloc[0]\n",
    "    df = df.drop(['SNP_ID'])\n",
    "    print(Sample_ID, df.shape)\n",
    "    print('*********************************************************')\n",
    "    G3_list.append(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0e80c7a1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 30s, sys: 107 ms, total: 1min 30s\n",
      "Wall time: 1min 30s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>SNP_ID</th>\n",
       "      <th>chr1-12783</th>\n",
       "      <th>chr1-13116</th>\n",
       "      <th>chr1-13118</th>\n",
       "      <th>chr1-14522</th>\n",
       "      <th>chr1-14542</th>\n",
       "      <th>chr1-14653</th>\n",
       "      <th>chr1-14775</th>\n",
       "      <th>chr1-14815</th>\n",
       "      <th>chr1-14907</th>\n",
       "      <th>chr1-14930</th>\n",
       "      <th>...</th>\n",
       "      <th>chr22-50117010</th>\n",
       "      <th>chr22-50438148</th>\n",
       "      <th>chr22-50451702</th>\n",
       "      <th>chr22-50516420</th>\n",
       "      <th>chr22-50694007</th>\n",
       "      <th>chr22-50751547</th>\n",
       "      <th>chr22-50775822</th>\n",
       "      <th>chr22-50804107</th>\n",
       "      <th>chr22-51066632</th>\n",
       "      <th>chr22-51097011</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G3_Sample_1_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_2_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_3_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_4_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_5_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_6_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_7_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_8_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_9_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9 rows × 500716 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "SNP_ID             chr1-12783 chr1-13116 chr1-13118 chr1-14522 chr1-14542  \\\n",
       "G3_Sample_1_merged        1/1        0/1        0/1        0/1        0/1   \n",
       "G3_Sample_2_merged        0/1        0/1        0/1        0/1        NaN   \n",
       "G3_Sample_3_merged        NaN        0/1        0/1        0/1        0/1   \n",
       "G3_Sample_4_merged        0/1        0/1        0/1        0/1        0/1   \n",
       "G3_Sample_5_merged        NaN        0/1        0/1        NaN        NaN   \n",
       "G3_Sample_6_merged        NaN        1/1        1/1        NaN        NaN   \n",
       "G3_Sample_7_merged        NaN        NaN        NaN        NaN        NaN   \n",
       "G3_Sample_8_merged        NaN        0/1        0/1        NaN        NaN   \n",
       "G3_Sample_9_merged        NaN        NaN        NaN        NaN        NaN   \n",
       "\n",
       "SNP_ID             chr1-14653 chr1-14775 chr1-14815 chr1-14907 chr1-14930  \\\n",
       "G3_Sample_1_merged        0/1        NaN        NaN        0/1        0/1   \n",
       "G3_Sample_2_merged        0/1        NaN        NaN        0/1        0/1   \n",
       "G3_Sample_3_merged        0/1        NaN        NaN        0/1        0/1   \n",
       "G3_Sample_4_merged        0/1        0/1        0/1        0/1        0/1   \n",
       "G3_Sample_5_merged        0/1        NaN        NaN        0/1        0/1   \n",
       "G3_Sample_6_merged        0/1        NaN        NaN        0/1        0/1   \n",
       "G3_Sample_7_merged        0/1        NaN        NaN        0/1        0/1   \n",
       "G3_Sample_8_merged        0/1        NaN        NaN        0/1        0/1   \n",
       "G3_Sample_9_merged        1/1        NaN        0/1        0/1        0/1   \n",
       "\n",
       "SNP_ID              ... chr22-50117010 chr22-50438148 chr22-50451702  \\\n",
       "G3_Sample_1_merged  ...            NaN            NaN            NaN   \n",
       "G3_Sample_2_merged  ...            NaN            NaN            NaN   \n",
       "G3_Sample_3_merged  ...            NaN            NaN            NaN   \n",
       "G3_Sample_4_merged  ...            NaN            NaN            NaN   \n",
       "G3_Sample_5_merged  ...            NaN            NaN            NaN   \n",
       "G3_Sample_6_merged  ...            1/1            1/1            0/1   \n",
       "G3_Sample_7_merged  ...            NaN            NaN            NaN   \n",
       "G3_Sample_8_merged  ...            NaN            NaN            NaN   \n",
       "G3_Sample_9_merged  ...            NaN            NaN            NaN   \n",
       "\n",
       "SNP_ID             chr22-50516420 chr22-50694007 chr22-50751547  \\\n",
       "G3_Sample_1_merged            NaN            NaN            NaN   \n",
       "G3_Sample_2_merged            NaN            NaN            NaN   \n",
       "G3_Sample_3_merged            NaN            NaN            NaN   \n",
       "G3_Sample_4_merged            NaN            NaN            NaN   \n",
       "G3_Sample_5_merged            NaN            NaN            NaN   \n",
       "G3_Sample_6_merged            1/1            0/1            0/1   \n",
       "G3_Sample_7_merged            NaN            NaN            NaN   \n",
       "G3_Sample_8_merged            NaN            NaN            NaN   \n",
       "G3_Sample_9_merged            NaN            NaN            NaN   \n",
       "\n",
       "SNP_ID             chr22-50775822 chr22-50804107 chr22-51066632 chr22-51097011  \n",
       "G3_Sample_1_merged            NaN            NaN            NaN            NaN  \n",
       "G3_Sample_2_merged            NaN            NaN            NaN            NaN  \n",
       "G3_Sample_3_merged            NaN            NaN            NaN            NaN  \n",
       "G3_Sample_4_merged            NaN            NaN            NaN            NaN  \n",
       "G3_Sample_5_merged            NaN            NaN            NaN            NaN  \n",
       "G3_Sample_6_merged            1/1            1/1            0/1            1/1  \n",
       "G3_Sample_7_merged            NaN            NaN            NaN            NaN  \n",
       "G3_Sample_8_merged            NaN            NaN            NaN            NaN  \n",
       "G3_Sample_9_merged            NaN            NaN            NaN            NaN  \n",
       "\n",
       "[9 rows x 500716 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "## Dataframe containing all Genotypes of All SNPs in all samples\n",
    "\n",
    "G3_raw_genotype = pd.concat(G3_list,axis=0)\n",
    "## Sort df by row index label\n",
    "G3_raw_genotype = G3_raw_genotype.sort_index(axis=0)\n",
    "G3_raw_genotype"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "836d1045-1f14-4603-975a-267167eac68b",
   "metadata": {},
   "source": [
    "## save Group3 separately as .csv file to be used as test group for ML external testing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "9b6aebe4-1e08-44e2-bd9f-265a3de20a5e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 15s, sys: 2.62 s, total: 1min 18s\n",
      "Wall time: 1min 18s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "Group_3_TestGp = '/home/phoebe/phd/ML_csv/'\n",
    "file_name = 'Group_3_TestGp.csv'\n",
    "file_out_path = os.path.join(Group_3_TestGp , file_name)\n",
    "G3_raw_genotype.to_csv(file_out_path, index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b7470354",
   "metadata": {},
   "source": [
    " # Concatenate ALL groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b73ae3c8",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 15s, sys: 251 ms, total: 1min 15s\n",
      "Wall time: 1min 15s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>SNP_ID</th>\n",
       "      <th>chr1-156785617</th>\n",
       "      <th>chr1-156846233</th>\n",
       "      <th>chr1-156848995</th>\n",
       "      <th>chr1-201285759</th>\n",
       "      <th>chr1-201285909</th>\n",
       "      <th>chr1-201286678</th>\n",
       "      <th>chr1-201286771</th>\n",
       "      <th>chr1-201288007</th>\n",
       "      <th>chr1-201291066</th>\n",
       "      <th>chr1-201293617</th>\n",
       "      <th>...</th>\n",
       "      <th>chr22-49828879</th>\n",
       "      <th>chr22-49844086</th>\n",
       "      <th>chr22-50117010</th>\n",
       "      <th>chr22-50438148</th>\n",
       "      <th>chr22-50451702</th>\n",
       "      <th>chr22-50516420</th>\n",
       "      <th>chr22-50694007</th>\n",
       "      <th>chr22-50775822</th>\n",
       "      <th>chr22-50804107</th>\n",
       "      <th>chr22-51097011</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G1_Sample_006_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_007_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_017_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_013_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_022_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_024_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_021_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_015_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_014_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_50_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_43_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_48_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_33_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_42_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_44_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_5_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_39_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_41_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_49_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_51_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_53_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_40_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_36_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_47_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_38_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_52_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_37_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_11_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_12_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_35_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_34_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_46_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_1_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_2_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_3_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_4_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_5_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_6_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_7_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_8_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_9_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41 rows × 569650 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "SNP_ID               chr1-156785617 chr1-156846233 chr1-156848995  \\\n",
       "G1_Sample_006_merged            0/1            0/1            0/1   \n",
       "G1_Sample_007_merged            1/1            NaN            0/1   \n",
       "G1_Sample_017_merged            1/1            0/1            NaN   \n",
       "G1_Sample_013_merged            1/1            NaN            0/1   \n",
       "G1_Sample_022_merged            1/1            NaN            1/1   \n",
       "G1_Sample_024_merged            0/1            NaN            1/1   \n",
       "G1_Sample_021_merged            1/1            0/1            0/1   \n",
       "G1_Sample_015_merged            1/1            NaN            1/1   \n",
       "G1_Sample_014_merged            0/1            NaN            0/1   \n",
       "G2_Sample_50_merged             1/1            NaN            1/1   \n",
       "G2_Sample_43_merged             1/1            NaN            1/1   \n",
       "G2_Sample_48_merged             0/1            NaN            1/1   \n",
       "G2_Sample_33_merged             1/1            NaN            1/1   \n",
       "G2_Sample_42_merged             NaN            0/1            1/1   \n",
       "G2_Sample_44_merged             0/1            NaN            NaN   \n",
       "G2_Sample_5_merged              1/1            1/1            NaN   \n",
       "G2_Sample_39_merged             1/1            NaN            0/1   \n",
       "G2_Sample_41_merged             1/1            NaN            0/1   \n",
       "G2_Sample_49_merged             1/1            NaN            0/1   \n",
       "G2_Sample_51_merged             0/1            NaN            NaN   \n",
       "G2_Sample_53_merged             NaN            0/1            0/1   \n",
       "G2_Sample_40_merged             1/1            0/1            0/1   \n",
       "G2_Sample_36_merged             0/1            NaN            0/1   \n",
       "G2_Sample_47_merged             NaN            0/1            0/1   \n",
       "G2_Sample_38_merged             0/1            NaN            1/1   \n",
       "G2_Sample_52_merged             0/1            0/1            0/1   \n",
       "G2_Sample_37_merged             1/1            NaN            0/1   \n",
       "G2_Sample_11_merged             1/1            1/1            NaN   \n",
       "G2_Sample_12_merged             1/1            1/1            NaN   \n",
       "G2_Sample_35_merged             NaN            0/1            0/1   \n",
       "G2_Sample_34_merged             1/1            NaN            1/1   \n",
       "G2_Sample_46_merged             0/1            0/1            0/1   \n",
       "G3_Sample_1_merged              0/1            NaN            1/1   \n",
       "G3_Sample_2_merged              0/1            NaN            1/1   \n",
       "G3_Sample_3_merged              0/1            0/1            0/1   \n",
       "G3_Sample_4_merged              0/1            NaN            0/1   \n",
       "G3_Sample_5_merged              NaN            NaN            NaN   \n",
       "G3_Sample_6_merged              1/1            NaN            0/1   \n",
       "G3_Sample_7_merged              0/1            0/1            NaN   \n",
       "G3_Sample_8_merged              NaN            NaN            NaN   \n",
       "G3_Sample_9_merged              1/1            NaN            1/1   \n",
       "\n",
       "SNP_ID               chr1-201285759 chr1-201285909 chr1-201286678  \\\n",
       "G1_Sample_006_merged            1/1            1/1            0/1   \n",
       "G1_Sample_007_merged            0/1            0/1            NaN   \n",
       "G1_Sample_017_merged            1/1            1/1            NaN   \n",
       "G1_Sample_013_merged            1/1            1/1            NaN   \n",
       "G1_Sample_022_merged            0/1            0/1            NaN   \n",
       "G1_Sample_024_merged            1/1            1/1            NaN   \n",
       "G1_Sample_021_merged            1/1            0/1            0/1   \n",
       "G1_Sample_015_merged            1/1            0/1            0/1   \n",
       "G1_Sample_014_merged            1/1            1/1            NaN   \n",
       "G2_Sample_50_merged             1/1            1/1            NaN   \n",
       "G2_Sample_43_merged             1/1            0/1            0/1   \n",
       "G2_Sample_48_merged             1/1            0/1            0/1   \n",
       "G2_Sample_33_merged             1/1            0/1            0/1   \n",
       "G2_Sample_42_merged             1/1            0/1            0/1   \n",
       "G2_Sample_44_merged             1/1            1/1            NaN   \n",
       "G2_Sample_5_merged              1/1            1/1            NaN   \n",
       "G2_Sample_39_merged             1/1            1/1            NaN   \n",
       "G2_Sample_41_merged             1/1            NaN            1/1   \n",
       "G2_Sample_49_merged             1/1            1/1            NaN   \n",
       "G2_Sample_51_merged             1/1            1/1            NaN   \n",
       "G2_Sample_53_merged             1/1            1/1            NaN   \n",
       "G2_Sample_40_merged             1/1            1/1            NaN   \n",
       "G2_Sample_36_merged             1/1            1/1            NaN   \n",
       "G2_Sample_47_merged             1/1            1/1            NaN   \n",
       "G2_Sample_38_merged             1/1            1/1            0/1   \n",
       "G2_Sample_52_merged             1/1            1/1            NaN   \n",
       "G2_Sample_37_merged             1/1            1/1            NaN   \n",
       "G2_Sample_11_merged             1/1            0/1            0/1   \n",
       "G2_Sample_12_merged             1/1            0/1            0/1   \n",
       "G2_Sample_35_merged             1/1            1/1            NaN   \n",
       "G2_Sample_34_merged             1/1            1/1            0/1   \n",
       "G2_Sample_46_merged             1/1            NaN            1/1   \n",
       "G3_Sample_1_merged              1/1            1/1            NaN   \n",
       "G3_Sample_2_merged              1/1            1/1            NaN   \n",
       "G3_Sample_3_merged              1/1            1/1            NaN   \n",
       "G3_Sample_4_merged              NaN            NaN            NaN   \n",
       "G3_Sample_5_merged              1/1            1/1            NaN   \n",
       "G3_Sample_6_merged              1/1            1/1            NaN   \n",
       "G3_Sample_7_merged              1/1            0/1            0/1   \n",
       "G3_Sample_8_merged              0/1            0/1            NaN   \n",
       "G3_Sample_9_merged              1/1            0/1            0/1   \n",
       "\n",
       "SNP_ID               chr1-201286771 chr1-201288007 chr1-201291066  \\\n",
       "G1_Sample_006_merged            0/1            1/1            0/1   \n",
       "G1_Sample_007_merged            NaN            0/1            0/1   \n",
       "G1_Sample_017_merged            NaN            1/1            1/1   \n",
       "G1_Sample_013_merged            NaN            0/1            0/1   \n",
       "G1_Sample_022_merged            NaN            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN            1/1            1/1   \n",
       "G1_Sample_021_merged            0/1            1/1            0/1   \n",
       "G1_Sample_015_merged            0/1            0/1            NaN   \n",
       "G1_Sample_014_merged            NaN            0/1            0/1   \n",
       "G2_Sample_50_merged             NaN            NaN            NaN   \n",
       "G2_Sample_43_merged             0/1            1/1            0/1   \n",
       "G2_Sample_48_merged             0/1            1/1            0/1   \n",
       "G2_Sample_33_merged             0/1            1/1            0/1   \n",
       "G2_Sample_42_merged             0/1            1/1            0/1   \n",
       "G2_Sample_44_merged             NaN            1/1            1/1   \n",
       "G2_Sample_5_merged              NaN            1/1            1/1   \n",
       "G2_Sample_39_merged             NaN            1/1            1/1   \n",
       "G2_Sample_41_merged             1/1            1/1            NaN   \n",
       "G2_Sample_49_merged             NaN            1/1            1/1   \n",
       "G2_Sample_51_merged             NaN            NaN            1/1   \n",
       "G2_Sample_53_merged             NaN            1/1            1/1   \n",
       "G2_Sample_40_merged             NaN            0/1            0/1   \n",
       "G2_Sample_36_merged             NaN            1/1            1/1   \n",
       "G2_Sample_47_merged             NaN            1/1            1/1   \n",
       "G2_Sample_38_merged             0/1            1/1            0/1   \n",
       "G2_Sample_52_merged             NaN            0/1            0/1   \n",
       "G2_Sample_37_merged             NaN            1/1            1/1   \n",
       "G2_Sample_11_merged             0/1            1/1            0/1   \n",
       "G2_Sample_12_merged             0/1            1/1            0/1   \n",
       "G2_Sample_35_merged             NaN            1/1            1/1   \n",
       "G2_Sample_34_merged             0/1            NaN            NaN   \n",
       "G2_Sample_46_merged             1/1            1/1            NaN   \n",
       "G3_Sample_1_merged              NaN            0/1            0/1   \n",
       "G3_Sample_2_merged              NaN            NaN            NaN   \n",
       "G3_Sample_3_merged              NaN            1/1            1/1   \n",
       "G3_Sample_4_merged              NaN            NaN            NaN   \n",
       "G3_Sample_5_merged              NaN            0/1            0/1   \n",
       "G3_Sample_6_merged              NaN            1/1            1/1   \n",
       "G3_Sample_7_merged              0/1            1/1            0/1   \n",
       "G3_Sample_8_merged              NaN            0/1            0/1   \n",
       "G3_Sample_9_merged              0/1            0/1            NaN   \n",
       "\n",
       "SNP_ID               chr1-201293617  ... chr22-49828879 chr22-49844086  \\\n",
       "G1_Sample_006_merged            0/1  ...            NaN            NaN   \n",
       "G1_Sample_007_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_017_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_013_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_022_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_021_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_015_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_014_merged            NaN  ...            NaN            NaN   \n",
       "G2_Sample_50_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_43_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_48_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_33_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_42_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_44_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_5_merged              NaN  ...            NaN            NaN   \n",
       "G2_Sample_39_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_41_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_49_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_51_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_53_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_40_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_36_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_47_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_38_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_52_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_37_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_11_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_12_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_35_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_34_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_46_merged             NaN  ...            NaN            NaN   \n",
       "G3_Sample_1_merged              NaN  ...            NaN            NaN   \n",
       "G3_Sample_2_merged              NaN  ...            NaN            NaN   \n",
       "G3_Sample_3_merged              NaN  ...            NaN            NaN   \n",
       "G3_Sample_4_merged              NaN  ...            NaN            NaN   \n",
       "G3_Sample_5_merged              NaN  ...            NaN            NaN   \n",
       "G3_Sample_6_merged              NaN  ...            1/1            1/1   \n",
       "G3_Sample_7_merged              NaN  ...            NaN            NaN   \n",
       "G3_Sample_8_merged              NaN  ...            NaN            NaN   \n",
       "G3_Sample_9_merged              NaN  ...            NaN            NaN   \n",
       "\n",
       "SNP_ID               chr22-50117010 chr22-50438148 chr22-50451702  \\\n",
       "G1_Sample_006_merged            NaN            NaN            NaN   \n",
       "G1_Sample_007_merged            NaN            NaN            NaN   \n",
       "G1_Sample_017_merged            NaN            NaN            NaN   \n",
       "G1_Sample_013_merged            NaN            NaN            NaN   \n",
       "G1_Sample_022_merged            NaN            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN            NaN            NaN   \n",
       "G1_Sample_021_merged            NaN            NaN            NaN   \n",
       "G1_Sample_015_merged            NaN            NaN            NaN   \n",
       "G1_Sample_014_merged            NaN            NaN            NaN   \n",
       "G2_Sample_50_merged             NaN            NaN            NaN   \n",
       "G2_Sample_43_merged             NaN            NaN            NaN   \n",
       "G2_Sample_48_merged             NaN            NaN            NaN   \n",
       "G2_Sample_33_merged             NaN            NaN            NaN   \n",
       "G2_Sample_42_merged             NaN            NaN            NaN   \n",
       "G2_Sample_44_merged             NaN            NaN            NaN   \n",
       "G2_Sample_5_merged              NaN            NaN            NaN   \n",
       "G2_Sample_39_merged             NaN            NaN            NaN   \n",
       "G2_Sample_41_merged             NaN            NaN            NaN   \n",
       "G2_Sample_49_merged             NaN            NaN            NaN   \n",
       "G2_Sample_51_merged             NaN            NaN            NaN   \n",
       "G2_Sample_53_merged             NaN            NaN            NaN   \n",
       "G2_Sample_40_merged             NaN            NaN            NaN   \n",
       "G2_Sample_36_merged             NaN            NaN            NaN   \n",
       "G2_Sample_47_merged             NaN            NaN            NaN   \n",
       "G2_Sample_38_merged             NaN            NaN            NaN   \n",
       "G2_Sample_52_merged             NaN            NaN            NaN   \n",
       "G2_Sample_37_merged             NaN            NaN            NaN   \n",
       "G2_Sample_11_merged             NaN            NaN            NaN   \n",
       "G2_Sample_12_merged             NaN            NaN            NaN   \n",
       "G2_Sample_35_merged             NaN            NaN            NaN   \n",
       "G2_Sample_34_merged             NaN            NaN            NaN   \n",
       "G2_Sample_46_merged             NaN            NaN            NaN   \n",
       "G3_Sample_1_merged              NaN            NaN            NaN   \n",
       "G3_Sample_2_merged              NaN            NaN            NaN   \n",
       "G3_Sample_3_merged              NaN            NaN            NaN   \n",
       "G3_Sample_4_merged              NaN            NaN            NaN   \n",
       "G3_Sample_5_merged              NaN            NaN            NaN   \n",
       "G3_Sample_6_merged              1/1            1/1            0/1   \n",
       "G3_Sample_7_merged              NaN            NaN            NaN   \n",
       "G3_Sample_8_merged              NaN            NaN            NaN   \n",
       "G3_Sample_9_merged              NaN            NaN            NaN   \n",
       "\n",
       "SNP_ID               chr22-50516420 chr22-50694007 chr22-50775822  \\\n",
       "G1_Sample_006_merged            NaN            NaN            NaN   \n",
       "G1_Sample_007_merged            NaN            NaN            NaN   \n",
       "G1_Sample_017_merged            NaN            NaN            NaN   \n",
       "G1_Sample_013_merged            NaN            NaN            NaN   \n",
       "G1_Sample_022_merged            NaN            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN            NaN            NaN   \n",
       "G1_Sample_021_merged            NaN            NaN            NaN   \n",
       "G1_Sample_015_merged            NaN            NaN            NaN   \n",
       "G1_Sample_014_merged            NaN            NaN            NaN   \n",
       "G2_Sample_50_merged             NaN            NaN            NaN   \n",
       "G2_Sample_43_merged             NaN            NaN            NaN   \n",
       "G2_Sample_48_merged             NaN            NaN            NaN   \n",
       "G2_Sample_33_merged             NaN            NaN            NaN   \n",
       "G2_Sample_42_merged             NaN            NaN            NaN   \n",
       "G2_Sample_44_merged             NaN            NaN            NaN   \n",
       "G2_Sample_5_merged              NaN            NaN            NaN   \n",
       "G2_Sample_39_merged             NaN            NaN            NaN   \n",
       "G2_Sample_41_merged             NaN            NaN            NaN   \n",
       "G2_Sample_49_merged             NaN            NaN            NaN   \n",
       "G2_Sample_51_merged             NaN            NaN            NaN   \n",
       "G2_Sample_53_merged             NaN            NaN            NaN   \n",
       "G2_Sample_40_merged             NaN            NaN            NaN   \n",
       "G2_Sample_36_merged             NaN            NaN            NaN   \n",
       "G2_Sample_47_merged             NaN            NaN            NaN   \n",
       "G2_Sample_38_merged             NaN            NaN            NaN   \n",
       "G2_Sample_52_merged             NaN            NaN            NaN   \n",
       "G2_Sample_37_merged             NaN            NaN            NaN   \n",
       "G2_Sample_11_merged             NaN            NaN            NaN   \n",
       "G2_Sample_12_merged             NaN            NaN            NaN   \n",
       "G2_Sample_35_merged             NaN            NaN            NaN   \n",
       "G2_Sample_34_merged             NaN            NaN            NaN   \n",
       "G2_Sample_46_merged             NaN            NaN            NaN   \n",
       "G3_Sample_1_merged              NaN            NaN            NaN   \n",
       "G3_Sample_2_merged              NaN            NaN            NaN   \n",
       "G3_Sample_3_merged              NaN            NaN            NaN   \n",
       "G3_Sample_4_merged              NaN            NaN            NaN   \n",
       "G3_Sample_5_merged              NaN            NaN            NaN   \n",
       "G3_Sample_6_merged              1/1            0/1            1/1   \n",
       "G3_Sample_7_merged              NaN            NaN            NaN   \n",
       "G3_Sample_8_merged              NaN            NaN            NaN   \n",
       "G3_Sample_9_merged              NaN            NaN            NaN   \n",
       "\n",
       "SNP_ID               chr22-50804107 chr22-51097011  \n",
       "G1_Sample_006_merged            NaN            NaN  \n",
       "G1_Sample_007_merged            NaN            NaN  \n",
       "G1_Sample_017_merged            NaN            NaN  \n",
       "G1_Sample_013_merged            NaN            NaN  \n",
       "G1_Sample_022_merged            NaN            NaN  \n",
       "G1_Sample_024_merged            NaN            NaN  \n",
       "G1_Sample_021_merged            NaN            NaN  \n",
       "G1_Sample_015_merged            NaN            NaN  \n",
       "G1_Sample_014_merged            NaN            NaN  \n",
       "G2_Sample_50_merged             NaN            NaN  \n",
       "G2_Sample_43_merged             NaN            NaN  \n",
       "G2_Sample_48_merged             NaN            NaN  \n",
       "G2_Sample_33_merged             NaN            NaN  \n",
       "G2_Sample_42_merged             NaN            NaN  \n",
       "G2_Sample_44_merged             NaN            NaN  \n",
       "G2_Sample_5_merged              NaN            NaN  \n",
       "G2_Sample_39_merged             NaN            NaN  \n",
       "G2_Sample_41_merged             NaN            NaN  \n",
       "G2_Sample_49_merged             NaN            NaN  \n",
       "G2_Sample_51_merged             NaN            NaN  \n",
       "G2_Sample_53_merged             NaN            NaN  \n",
       "G2_Sample_40_merged             NaN            NaN  \n",
       "G2_Sample_36_merged             NaN            NaN  \n",
       "G2_Sample_47_merged             NaN            NaN  \n",
       "G2_Sample_38_merged             NaN            NaN  \n",
       "G2_Sample_52_merged             NaN            NaN  \n",
       "G2_Sample_37_merged             NaN            NaN  \n",
       "G2_Sample_11_merged             NaN            NaN  \n",
       "G2_Sample_12_merged             NaN            NaN  \n",
       "G2_Sample_35_merged             NaN            NaN  \n",
       "G2_Sample_34_merged             NaN            NaN  \n",
       "G2_Sample_46_merged             NaN            NaN  \n",
       "G3_Sample_1_merged              NaN            NaN  \n",
       "G3_Sample_2_merged              NaN            NaN  \n",
       "G3_Sample_3_merged              NaN            NaN  \n",
       "G3_Sample_4_merged              NaN            NaN  \n",
       "G3_Sample_5_merged              NaN            NaN  \n",
       "G3_Sample_6_merged              1/1            1/1  \n",
       "G3_Sample_7_merged              NaN            NaN  \n",
       "G3_Sample_8_merged              NaN            NaN  \n",
       "G3_Sample_9_merged              NaN            NaN  \n",
       "\n",
       "[41 rows x 569650 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "## Dataframe containing all Genotypes of All SNPs in all samples in G1 + G2\n",
    "\n",
    "G1_G2_G3 = pd.concat([G1_raw_genotype, G2_raw_genotype, G3_raw_genotype],axis=0)\n",
    "G1_G2_G3"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd8dc2ba",
   "metadata": {},
   "source": [
    "> ## Sort by rows and by columns "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c7542994",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>SNP_ID</th>\n",
       "      <th>chr1-100000012</th>\n",
       "      <th>chr1-100007888</th>\n",
       "      <th>chr1-100011243</th>\n",
       "      <th>chr1-100011253</th>\n",
       "      <th>chr1-1000156</th>\n",
       "      <th>chr1-100015837</th>\n",
       "      <th>chr1-100015925</th>\n",
       "      <th>chr1-100019269</th>\n",
       "      <th>chr1-100022918</th>\n",
       "      <th>chr1-10002546</th>\n",
       "      <th>...</th>\n",
       "      <th>chrX-99961072</th>\n",
       "      <th>chrX-99969558</th>\n",
       "      <th>chrX-99972474</th>\n",
       "      <th>chrX-9997369</th>\n",
       "      <th>chrX-99974005</th>\n",
       "      <th>chrX-99979187</th>\n",
       "      <th>chrX-99990696</th>\n",
       "      <th>chrX-99995215</th>\n",
       "      <th>chrX-99997342</th>\n",
       "      <th>chrX-99997406</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G1_Sample_006_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_007_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_013_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_014_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_015_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_017_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_021_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_022_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_024_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_11_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_12_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_33_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_34_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_35_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_36_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_37_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_38_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_39_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_40_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_41_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_42_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_43_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_44_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_46_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_47_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_48_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_49_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_50_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_51_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_52_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_53_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_5_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_1_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_2_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_3_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_4_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_5_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_6_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_7_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_8_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_9_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41 rows × 569650 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "SNP_ID               chr1-100000012 chr1-100007888 chr1-100011243  \\\n",
       "G1_Sample_006_merged            NaN            NaN            NaN   \n",
       "G1_Sample_007_merged            NaN            NaN            NaN   \n",
       "G1_Sample_013_merged            NaN            NaN            NaN   \n",
       "G1_Sample_014_merged            NaN            NaN            NaN   \n",
       "G1_Sample_015_merged            NaN            NaN            NaN   \n",
       "G1_Sample_017_merged            NaN            NaN            NaN   \n",
       "G1_Sample_021_merged            NaN            NaN            NaN   \n",
       "G1_Sample_022_merged            NaN            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN            NaN            NaN   \n",
       "G2_Sample_11_merged             NaN            NaN            NaN   \n",
       "G2_Sample_12_merged             NaN            NaN            NaN   \n",
       "G2_Sample_33_merged             NaN            NaN            NaN   \n",
       "G2_Sample_34_merged             NaN            NaN            NaN   \n",
       "G2_Sample_35_merged             NaN            NaN            NaN   \n",
       "G2_Sample_36_merged             NaN            NaN            NaN   \n",
       "G2_Sample_37_merged             NaN            NaN            NaN   \n",
       "G2_Sample_38_merged             NaN            NaN            NaN   \n",
       "G2_Sample_39_merged             NaN            NaN            NaN   \n",
       "G2_Sample_40_merged             NaN            NaN            NaN   \n",
       "G2_Sample_41_merged             NaN            NaN            NaN   \n",
       "G2_Sample_42_merged             NaN            NaN            NaN   \n",
       "G2_Sample_43_merged             NaN            NaN            NaN   \n",
       "G2_Sample_44_merged             NaN            NaN            NaN   \n",
       "G2_Sample_46_merged             NaN            NaN            NaN   \n",
       "G2_Sample_47_merged             NaN            NaN            NaN   \n",
       "G2_Sample_48_merged             NaN            NaN            NaN   \n",
       "G2_Sample_49_merged             NaN            NaN            NaN   \n",
       "G2_Sample_50_merged             NaN            NaN            NaN   \n",
       "G2_Sample_51_merged             NaN            NaN            NaN   \n",
       "G2_Sample_52_merged             NaN            NaN            NaN   \n",
       "G2_Sample_53_merged             NaN            NaN            NaN   \n",
       "G2_Sample_5_merged              NaN            NaN            NaN   \n",
       "G3_Sample_1_merged              NaN            NaN            NaN   \n",
       "G3_Sample_2_merged              NaN            1/1            NaN   \n",
       "G3_Sample_3_merged              NaN            NaN            NaN   \n",
       "G3_Sample_4_merged              NaN            NaN            NaN   \n",
       "G3_Sample_5_merged              1/1            NaN            NaN   \n",
       "G3_Sample_6_merged              NaN            NaN            NaN   \n",
       "G3_Sample_7_merged              NaN            NaN            NaN   \n",
       "G3_Sample_8_merged              NaN            NaN            0/1   \n",
       "G3_Sample_9_merged              NaN            NaN            NaN   \n",
       "\n",
       "SNP_ID               chr1-100011253 chr1-1000156 chr1-100015837  \\\n",
       "G1_Sample_006_merged            NaN          NaN            NaN   \n",
       "G1_Sample_007_merged            NaN          NaN            NaN   \n",
       "G1_Sample_013_merged            NaN          NaN            NaN   \n",
       "G1_Sample_014_merged            NaN          NaN            NaN   \n",
       "G1_Sample_015_merged            NaN          NaN            NaN   \n",
       "G1_Sample_017_merged            NaN          NaN            NaN   \n",
       "G1_Sample_021_merged            NaN          NaN            NaN   \n",
       "G1_Sample_022_merged            NaN          NaN            NaN   \n",
       "G1_Sample_024_merged            NaN          NaN            NaN   \n",
       "G2_Sample_11_merged             NaN          NaN            NaN   \n",
       "G2_Sample_12_merged             NaN          NaN            NaN   \n",
       "G2_Sample_33_merged             NaN          NaN            NaN   \n",
       "G2_Sample_34_merged             NaN          NaN            NaN   \n",
       "G2_Sample_35_merged             NaN          NaN            NaN   \n",
       "G2_Sample_36_merged             NaN          NaN            NaN   \n",
       "G2_Sample_37_merged             NaN          NaN            NaN   \n",
       "G2_Sample_38_merged             NaN          NaN            NaN   \n",
       "G2_Sample_39_merged             NaN          NaN            NaN   \n",
       "G2_Sample_40_merged             NaN          NaN            NaN   \n",
       "G2_Sample_41_merged             NaN          NaN            NaN   \n",
       "G2_Sample_42_merged             NaN          NaN            NaN   \n",
       "G2_Sample_43_merged             NaN          NaN            NaN   \n",
       "G2_Sample_44_merged             NaN          NaN            NaN   \n",
       "G2_Sample_46_merged             NaN          NaN            NaN   \n",
       "G2_Sample_47_merged             NaN          NaN            NaN   \n",
       "G2_Sample_48_merged             NaN          NaN            NaN   \n",
       "G2_Sample_49_merged             NaN          NaN            NaN   \n",
       "G2_Sample_50_merged             NaN          NaN            NaN   \n",
       "G2_Sample_51_merged             NaN          NaN            NaN   \n",
       "G2_Sample_52_merged             NaN          NaN            NaN   \n",
       "G2_Sample_53_merged             NaN          NaN            NaN   \n",
       "G2_Sample_5_merged              NaN          NaN            NaN   \n",
       "G3_Sample_1_merged              NaN          NaN            NaN   \n",
       "G3_Sample_2_merged              NaN          NaN            NaN   \n",
       "G3_Sample_3_merged              NaN          NaN            NaN   \n",
       "G3_Sample_4_merged              NaN          NaN            NaN   \n",
       "G3_Sample_5_merged              NaN          NaN            1/1   \n",
       "G3_Sample_6_merged              NaN          1/1            NaN   \n",
       "G3_Sample_7_merged              NaN          NaN            NaN   \n",
       "G3_Sample_8_merged              0/1          NaN            NaN   \n",
       "G3_Sample_9_merged              NaN          NaN            NaN   \n",
       "\n",
       "SNP_ID               chr1-100015925 chr1-100019269 chr1-100022918  \\\n",
       "G1_Sample_006_merged            NaN            NaN            NaN   \n",
       "G1_Sample_007_merged            NaN            NaN            NaN   \n",
       "G1_Sample_013_merged            NaN            NaN            NaN   \n",
       "G1_Sample_014_merged            NaN            NaN            NaN   \n",
       "G1_Sample_015_merged            NaN            NaN            NaN   \n",
       "G1_Sample_017_merged            NaN            NaN            NaN   \n",
       "G1_Sample_021_merged            NaN            NaN            NaN   \n",
       "G1_Sample_022_merged            NaN            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN            NaN            NaN   \n",
       "G2_Sample_11_merged             NaN            NaN            NaN   \n",
       "G2_Sample_12_merged             NaN            NaN            NaN   \n",
       "G2_Sample_33_merged             NaN            NaN            NaN   \n",
       "G2_Sample_34_merged             NaN            NaN            NaN   \n",
       "G2_Sample_35_merged             NaN            NaN            NaN   \n",
       "G2_Sample_36_merged             NaN            NaN            NaN   \n",
       "G2_Sample_37_merged             NaN            NaN            NaN   \n",
       "G2_Sample_38_merged             NaN            NaN            NaN   \n",
       "G2_Sample_39_merged             NaN            NaN            NaN   \n",
       "G2_Sample_40_merged             NaN            NaN            NaN   \n",
       "G2_Sample_41_merged             NaN            NaN            NaN   \n",
       "G2_Sample_42_merged             NaN            NaN            NaN   \n",
       "G2_Sample_43_merged             NaN            NaN            NaN   \n",
       "G2_Sample_44_merged             NaN            NaN            NaN   \n",
       "G2_Sample_46_merged             NaN            NaN            NaN   \n",
       "G2_Sample_47_merged             NaN            NaN            NaN   \n",
       "G2_Sample_48_merged             NaN            NaN            NaN   \n",
       "G2_Sample_49_merged             NaN            NaN            NaN   \n",
       "G2_Sample_50_merged             NaN            NaN            NaN   \n",
       "G2_Sample_51_merged             NaN            NaN            NaN   \n",
       "G2_Sample_52_merged             NaN            NaN            NaN   \n",
       "G2_Sample_53_merged             NaN            NaN            NaN   \n",
       "G2_Sample_5_merged              NaN            NaN            NaN   \n",
       "G3_Sample_1_merged              NaN            NaN            NaN   \n",
       "G3_Sample_2_merged              NaN            NaN            NaN   \n",
       "G3_Sample_3_merged              NaN            NaN            NaN   \n",
       "G3_Sample_4_merged              1/1            1/1            NaN   \n",
       "G3_Sample_5_merged              NaN            NaN            NaN   \n",
       "G3_Sample_6_merged              NaN            NaN            NaN   \n",
       "G3_Sample_7_merged              NaN            NaN            NaN   \n",
       "G3_Sample_8_merged              NaN            NaN            NaN   \n",
       "G3_Sample_9_merged              NaN            NaN            1/1   \n",
       "\n",
       "SNP_ID               chr1-10002546  ... chrX-99961072 chrX-99969558  \\\n",
       "G1_Sample_006_merged           NaN  ...           NaN           NaN   \n",
       "G1_Sample_007_merged           NaN  ...           NaN           NaN   \n",
       "G1_Sample_013_merged           NaN  ...           NaN           NaN   \n",
       "G1_Sample_014_merged           NaN  ...           NaN           NaN   \n",
       "G1_Sample_015_merged           NaN  ...           NaN           NaN   \n",
       "G1_Sample_017_merged           NaN  ...           NaN           NaN   \n",
       "G1_Sample_021_merged           NaN  ...           NaN           NaN   \n",
       "G1_Sample_022_merged           NaN  ...           NaN           NaN   \n",
       "G1_Sample_024_merged           NaN  ...           NaN           NaN   \n",
       "G2_Sample_11_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_12_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_33_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_34_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_35_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_36_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_37_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_38_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_39_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_40_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_41_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_42_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_43_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_44_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_46_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_47_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_48_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_49_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_50_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_51_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_52_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_53_merged            NaN  ...           NaN           NaN   \n",
       "G2_Sample_5_merged             NaN  ...           NaN           NaN   \n",
       "G3_Sample_1_merged             NaN  ...           NaN           1/1   \n",
       "G3_Sample_2_merged             NaN  ...           NaN           NaN   \n",
       "G3_Sample_3_merged             NaN  ...           NaN           NaN   \n",
       "G3_Sample_4_merged             NaN  ...           NaN           NaN   \n",
       "G3_Sample_5_merged             NaN  ...           1/1           NaN   \n",
       "G3_Sample_6_merged             NaN  ...           NaN           NaN   \n",
       "G3_Sample_7_merged             0/1  ...           NaN           NaN   \n",
       "G3_Sample_8_merged             NaN  ...           NaN           NaN   \n",
       "G3_Sample_9_merged             NaN  ...           NaN           NaN   \n",
       "\n",
       "SNP_ID               chrX-99972474 chrX-9997369 chrX-99974005 chrX-99979187  \\\n",
       "G1_Sample_006_merged           NaN          NaN           NaN           NaN   \n",
       "G1_Sample_007_merged           NaN          NaN           NaN           NaN   \n",
       "G1_Sample_013_merged           NaN          NaN           NaN           NaN   \n",
       "G1_Sample_014_merged           NaN          NaN           NaN           NaN   \n",
       "G1_Sample_015_merged           NaN          NaN           NaN           NaN   \n",
       "G1_Sample_017_merged           NaN          NaN           NaN           NaN   \n",
       "G1_Sample_021_merged           NaN          NaN           NaN           NaN   \n",
       "G1_Sample_022_merged           NaN          NaN           NaN           NaN   \n",
       "G1_Sample_024_merged           NaN          NaN           NaN           NaN   \n",
       "G2_Sample_11_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_12_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_33_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_34_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_35_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_36_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_37_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_38_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_39_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_40_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_41_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_42_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_43_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_44_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_46_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_47_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_48_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_49_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_50_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_51_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_52_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_53_merged            NaN          NaN           NaN           NaN   \n",
       "G2_Sample_5_merged             NaN          NaN           NaN           NaN   \n",
       "G3_Sample_1_merged             NaN          NaN           NaN           NaN   \n",
       "G3_Sample_2_merged             NaN          0/1           NaN           NaN   \n",
       "G3_Sample_3_merged             NaN          NaN           1/1           NaN   \n",
       "G3_Sample_4_merged             NaN          NaN           NaN           NaN   \n",
       "G3_Sample_5_merged             1/1          NaN           NaN           0/1   \n",
       "G3_Sample_6_merged             NaN          NaN           NaN           NaN   \n",
       "G3_Sample_7_merged             NaN          NaN           NaN           NaN   \n",
       "G3_Sample_8_merged             NaN          NaN           NaN           NaN   \n",
       "G3_Sample_9_merged             NaN          NaN           NaN           NaN   \n",
       "\n",
       "SNP_ID               chrX-99990696 chrX-99995215 chrX-99997342 chrX-99997406  \n",
       "G1_Sample_006_merged           NaN           NaN           NaN           NaN  \n",
       "G1_Sample_007_merged           NaN           NaN           NaN           NaN  \n",
       "G1_Sample_013_merged           NaN           NaN           NaN           NaN  \n",
       "G1_Sample_014_merged           NaN           NaN           NaN           NaN  \n",
       "G1_Sample_015_merged           NaN           NaN           NaN           NaN  \n",
       "G1_Sample_017_merged           NaN           NaN           NaN           NaN  \n",
       "G1_Sample_021_merged           NaN           NaN           NaN           NaN  \n",
       "G1_Sample_022_merged           NaN           NaN           NaN           NaN  \n",
       "G1_Sample_024_merged           NaN           NaN           NaN           NaN  \n",
       "G2_Sample_11_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_12_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_33_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_34_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_35_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_36_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_37_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_38_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_39_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_40_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_41_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_42_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_43_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_44_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_46_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_47_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_48_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_49_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_50_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_51_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_52_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_53_merged            NaN           NaN           NaN           NaN  \n",
       "G2_Sample_5_merged             NaN           NaN           NaN           NaN  \n",
       "G3_Sample_1_merged             NaN           NaN           NaN           NaN  \n",
       "G3_Sample_2_merged             NaN           NaN           NaN           NaN  \n",
       "G3_Sample_3_merged             NaN           0/1           NaN           1/1  \n",
       "G3_Sample_4_merged             NaN           NaN           NaN           NaN  \n",
       "G3_Sample_5_merged             NaN           NaN           NaN           NaN  \n",
       "G3_Sample_6_merged             NaN           NaN           NaN           NaN  \n",
       "G3_Sample_7_merged             NaN           NaN           NaN           NaN  \n",
       "G3_Sample_8_merged             1/1           NaN           1/1           1/1  \n",
       "G3_Sample_9_merged             NaN           NaN           NaN           NaN  \n",
       "\n",
       "[41 rows x 569650 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Sort df by row index label\n",
    "G1_G2_G3_Sorted = G1_G2_G3.sort_index(axis=0)\n",
    "\n",
    "## Sort df by column index label\n",
    "G1_G2_G3_Sorted = G1_G2_G3_Sorted.sort_index(axis=1)\n",
    "G1_G2_G3_Sorted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fe90b2b9",
   "metadata": {},
   "source": [
    "> ## Replace nan values in Genotype DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "226dacb8",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>SNP_ID</th>\n",
       "      <th>chr1-100000012</th>\n",
       "      <th>chr1-100007888</th>\n",
       "      <th>chr1-100011243</th>\n",
       "      <th>chr1-100011253</th>\n",
       "      <th>chr1-1000156</th>\n",
       "      <th>chr1-100015837</th>\n",
       "      <th>chr1-100015925</th>\n",
       "      <th>chr1-100019269</th>\n",
       "      <th>chr1-100022918</th>\n",
       "      <th>chr1-10002546</th>\n",
       "      <th>...</th>\n",
       "      <th>chrX-99961072</th>\n",
       "      <th>chrX-99969558</th>\n",
       "      <th>chrX-99972474</th>\n",
       "      <th>chrX-9997369</th>\n",
       "      <th>chrX-99974005</th>\n",
       "      <th>chrX-99979187</th>\n",
       "      <th>chrX-99990696</th>\n",
       "      <th>chrX-99995215</th>\n",
       "      <th>chrX-99997342</th>\n",
       "      <th>chrX-99997406</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G1_Sample_006_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_007_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_013_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_014_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_015_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_017_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_021_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_022_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_024_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_11_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_12_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_33_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_34_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_35_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_36_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_37_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_38_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_39_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_40_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_41_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_42_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_43_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_44_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_46_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_47_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_48_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_49_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_50_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_51_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_52_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_53_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_5_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_1_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_2_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_3_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_4_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_5_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_6_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_7_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_8_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_9_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41 rows × 569650 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "SNP_ID               chr1-100000012 chr1-100007888 chr1-100011243  \\\n",
       "G1_Sample_006_merged            0/0            0/0            0/0   \n",
       "G1_Sample_007_merged            0/0            0/0            0/0   \n",
       "G1_Sample_013_merged            0/0            0/0            0/0   \n",
       "G1_Sample_014_merged            0/0            0/0            0/0   \n",
       "G1_Sample_015_merged            0/0            0/0            0/0   \n",
       "G1_Sample_017_merged            0/0            0/0            0/0   \n",
       "G1_Sample_021_merged            0/0            0/0            0/0   \n",
       "G1_Sample_022_merged            0/0            0/0            0/0   \n",
       "G1_Sample_024_merged            0/0            0/0            0/0   \n",
       "G2_Sample_11_merged             0/0            0/0            0/0   \n",
       "G2_Sample_12_merged             0/0            0/0            0/0   \n",
       "G2_Sample_33_merged             0/0            0/0            0/0   \n",
       "G2_Sample_34_merged             0/0            0/0            0/0   \n",
       "G2_Sample_35_merged             0/0            0/0            0/0   \n",
       "G2_Sample_36_merged             0/0            0/0            0/0   \n",
       "G2_Sample_37_merged             0/0            0/0            0/0   \n",
       "G2_Sample_38_merged             0/0            0/0            0/0   \n",
       "G2_Sample_39_merged             0/0            0/0            0/0   \n",
       "G2_Sample_40_merged             0/0            0/0            0/0   \n",
       "G2_Sample_41_merged             0/0            0/0            0/0   \n",
       "G2_Sample_42_merged             0/0            0/0            0/0   \n",
       "G2_Sample_43_merged             0/0            0/0            0/0   \n",
       "G2_Sample_44_merged             0/0            0/0            0/0   \n",
       "G2_Sample_46_merged             0/0            0/0            0/0   \n",
       "G2_Sample_47_merged             0/0            0/0            0/0   \n",
       "G2_Sample_48_merged             0/0            0/0            0/0   \n",
       "G2_Sample_49_merged             0/0            0/0            0/0   \n",
       "G2_Sample_50_merged             0/0            0/0            0/0   \n",
       "G2_Sample_51_merged             0/0            0/0            0/0   \n",
       "G2_Sample_52_merged             0/0            0/0            0/0   \n",
       "G2_Sample_53_merged             0/0            0/0            0/0   \n",
       "G2_Sample_5_merged              0/0            0/0            0/0   \n",
       "G3_Sample_1_merged              0/0            0/0            0/0   \n",
       "G3_Sample_2_merged              0/0            1/1            0/0   \n",
       "G3_Sample_3_merged              0/0            0/0            0/0   \n",
       "G3_Sample_4_merged              0/0            0/0            0/0   \n",
       "G3_Sample_5_merged              1/1            0/0            0/0   \n",
       "G3_Sample_6_merged              0/0            0/0            0/0   \n",
       "G3_Sample_7_merged              0/0            0/0            0/0   \n",
       "G3_Sample_8_merged              0/0            0/0            0/1   \n",
       "G3_Sample_9_merged              0/0            0/0            0/0   \n",
       "\n",
       "SNP_ID               chr1-100011253 chr1-1000156 chr1-100015837  \\\n",
       "G1_Sample_006_merged            0/0          0/0            0/0   \n",
       "G1_Sample_007_merged            0/0          0/0            0/0   \n",
       "G1_Sample_013_merged            0/0          0/0            0/0   \n",
       "G1_Sample_014_merged            0/0          0/0            0/0   \n",
       "G1_Sample_015_merged            0/0          0/0            0/0   \n",
       "G1_Sample_017_merged            0/0          0/0            0/0   \n",
       "G1_Sample_021_merged            0/0          0/0            0/0   \n",
       "G1_Sample_022_merged            0/0          0/0            0/0   \n",
       "G1_Sample_024_merged            0/0          0/0            0/0   \n",
       "G2_Sample_11_merged             0/0          0/0            0/0   \n",
       "G2_Sample_12_merged             0/0          0/0            0/0   \n",
       "G2_Sample_33_merged             0/0          0/0            0/0   \n",
       "G2_Sample_34_merged             0/0          0/0            0/0   \n",
       "G2_Sample_35_merged             0/0          0/0            0/0   \n",
       "G2_Sample_36_merged             0/0          0/0            0/0   \n",
       "G2_Sample_37_merged             0/0          0/0            0/0   \n",
       "G2_Sample_38_merged             0/0          0/0            0/0   \n",
       "G2_Sample_39_merged             0/0          0/0            0/0   \n",
       "G2_Sample_40_merged             0/0          0/0            0/0   \n",
       "G2_Sample_41_merged             0/0          0/0            0/0   \n",
       "G2_Sample_42_merged             0/0          0/0            0/0   \n",
       "G2_Sample_43_merged             0/0          0/0            0/0   \n",
       "G2_Sample_44_merged             0/0          0/0            0/0   \n",
       "G2_Sample_46_merged             0/0          0/0            0/0   \n",
       "G2_Sample_47_merged             0/0          0/0            0/0   \n",
       "G2_Sample_48_merged             0/0          0/0            0/0   \n",
       "G2_Sample_49_merged             0/0          0/0            0/0   \n",
       "G2_Sample_50_merged             0/0          0/0            0/0   \n",
       "G2_Sample_51_merged             0/0          0/0            0/0   \n",
       "G2_Sample_52_merged             0/0          0/0            0/0   \n",
       "G2_Sample_53_merged             0/0          0/0            0/0   \n",
       "G2_Sample_5_merged              0/0          0/0            0/0   \n",
       "G3_Sample_1_merged              0/0          0/0            0/0   \n",
       "G3_Sample_2_merged              0/0          0/0            0/0   \n",
       "G3_Sample_3_merged              0/0          0/0            0/0   \n",
       "G3_Sample_4_merged              0/0          0/0            0/0   \n",
       "G3_Sample_5_merged              0/0          0/0            1/1   \n",
       "G3_Sample_6_merged              0/0          1/1            0/0   \n",
       "G3_Sample_7_merged              0/0          0/0            0/0   \n",
       "G3_Sample_8_merged              0/1          0/0            0/0   \n",
       "G3_Sample_9_merged              0/0          0/0            0/0   \n",
       "\n",
       "SNP_ID               chr1-100015925 chr1-100019269 chr1-100022918  \\\n",
       "G1_Sample_006_merged            0/0            0/0            0/0   \n",
       "G1_Sample_007_merged            0/0            0/0            0/0   \n",
       "G1_Sample_013_merged            0/0            0/0            0/0   \n",
       "G1_Sample_014_merged            0/0            0/0            0/0   \n",
       "G1_Sample_015_merged            0/0            0/0            0/0   \n",
       "G1_Sample_017_merged            0/0            0/0            0/0   \n",
       "G1_Sample_021_merged            0/0            0/0            0/0   \n",
       "G1_Sample_022_merged            0/0            0/0            0/0   \n",
       "G1_Sample_024_merged            0/0            0/0            0/0   \n",
       "G2_Sample_11_merged             0/0            0/0            0/0   \n",
       "G2_Sample_12_merged             0/0            0/0            0/0   \n",
       "G2_Sample_33_merged             0/0            0/0            0/0   \n",
       "G2_Sample_34_merged             0/0            0/0            0/0   \n",
       "G2_Sample_35_merged             0/0            0/0            0/0   \n",
       "G2_Sample_36_merged             0/0            0/0            0/0   \n",
       "G2_Sample_37_merged             0/0            0/0            0/0   \n",
       "G2_Sample_38_merged             0/0            0/0            0/0   \n",
       "G2_Sample_39_merged             0/0            0/0            0/0   \n",
       "G2_Sample_40_merged             0/0            0/0            0/0   \n",
       "G2_Sample_41_merged             0/0            0/0            0/0   \n",
       "G2_Sample_42_merged             0/0            0/0            0/0   \n",
       "G2_Sample_43_merged             0/0            0/0            0/0   \n",
       "G2_Sample_44_merged             0/0            0/0            0/0   \n",
       "G2_Sample_46_merged             0/0            0/0            0/0   \n",
       "G2_Sample_47_merged             0/0            0/0            0/0   \n",
       "G2_Sample_48_merged             0/0            0/0            0/0   \n",
       "G2_Sample_49_merged             0/0            0/0            0/0   \n",
       "G2_Sample_50_merged             0/0            0/0            0/0   \n",
       "G2_Sample_51_merged             0/0            0/0            0/0   \n",
       "G2_Sample_52_merged             0/0            0/0            0/0   \n",
       "G2_Sample_53_merged             0/0            0/0            0/0   \n",
       "G2_Sample_5_merged              0/0            0/0            0/0   \n",
       "G3_Sample_1_merged              0/0            0/0            0/0   \n",
       "G3_Sample_2_merged              0/0            0/0            0/0   \n",
       "G3_Sample_3_merged              0/0            0/0            0/0   \n",
       "G3_Sample_4_merged              1/1            1/1            0/0   \n",
       "G3_Sample_5_merged              0/0            0/0            0/0   \n",
       "G3_Sample_6_merged              0/0            0/0            0/0   \n",
       "G3_Sample_7_merged              0/0            0/0            0/0   \n",
       "G3_Sample_8_merged              0/0            0/0            0/0   \n",
       "G3_Sample_9_merged              0/0            0/0            1/1   \n",
       "\n",
       "SNP_ID               chr1-10002546  ... chrX-99961072 chrX-99969558  \\\n",
       "G1_Sample_006_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_007_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_013_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_014_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_015_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_017_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_021_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_022_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_024_merged           0/0  ...           0/0           0/0   \n",
       "G2_Sample_11_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_12_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_33_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_34_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_35_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_36_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_37_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_38_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_39_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_40_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_41_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_42_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_43_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_44_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_46_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_47_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_48_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_49_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_50_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_51_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_52_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_53_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_5_merged             0/0  ...           0/0           0/0   \n",
       "G3_Sample_1_merged             0/0  ...           0/0           1/1   \n",
       "G3_Sample_2_merged             0/0  ...           0/0           0/0   \n",
       "G3_Sample_3_merged             0/0  ...           0/0           0/0   \n",
       "G3_Sample_4_merged             0/0  ...           0/0           0/0   \n",
       "G3_Sample_5_merged             0/0  ...           1/1           0/0   \n",
       "G3_Sample_6_merged             0/0  ...           0/0           0/0   \n",
       "G3_Sample_7_merged             0/1  ...           0/0           0/0   \n",
       "G3_Sample_8_merged             0/0  ...           0/0           0/0   \n",
       "G3_Sample_9_merged             0/0  ...           0/0           0/0   \n",
       "\n",
       "SNP_ID               chrX-99972474 chrX-9997369 chrX-99974005 chrX-99979187  \\\n",
       "G1_Sample_006_merged           0/0          0/0           0/0           0/0   \n",
       "G1_Sample_007_merged           0/0          0/0           0/0           0/0   \n",
       "G1_Sample_013_merged           0/0          0/0           0/0           0/0   \n",
       "G1_Sample_014_merged           0/0          0/0           0/0           0/0   \n",
       "G1_Sample_015_merged           0/0          0/0           0/0           0/0   \n",
       "G1_Sample_017_merged           0/0          0/0           0/0           0/0   \n",
       "G1_Sample_021_merged           0/0          0/0           0/0           0/0   \n",
       "G1_Sample_022_merged           0/0          0/0           0/0           0/0   \n",
       "G1_Sample_024_merged           0/0          0/0           0/0           0/0   \n",
       "G2_Sample_11_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_12_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_33_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_34_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_35_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_36_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_37_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_38_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_39_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_40_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_41_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_42_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_43_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_44_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_46_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_47_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_48_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_49_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_50_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_51_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_52_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_53_merged            0/0          0/0           0/0           0/0   \n",
       "G2_Sample_5_merged             0/0          0/0           0/0           0/0   \n",
       "G3_Sample_1_merged             0/0          0/0           0/0           0/0   \n",
       "G3_Sample_2_merged             0/0          0/1           0/0           0/0   \n",
       "G3_Sample_3_merged             0/0          0/0           1/1           0/0   \n",
       "G3_Sample_4_merged             0/0          0/0           0/0           0/0   \n",
       "G3_Sample_5_merged             1/1          0/0           0/0           0/1   \n",
       "G3_Sample_6_merged             0/0          0/0           0/0           0/0   \n",
       "G3_Sample_7_merged             0/0          0/0           0/0           0/0   \n",
       "G3_Sample_8_merged             0/0          0/0           0/0           0/0   \n",
       "G3_Sample_9_merged             0/0          0/0           0/0           0/0   \n",
       "\n",
       "SNP_ID               chrX-99990696 chrX-99995215 chrX-99997342 chrX-99997406  \n",
       "G1_Sample_006_merged           0/0           0/0           0/0           0/0  \n",
       "G1_Sample_007_merged           0/0           0/0           0/0           0/0  \n",
       "G1_Sample_013_merged           0/0           0/0           0/0           0/0  \n",
       "G1_Sample_014_merged           0/0           0/0           0/0           0/0  \n",
       "G1_Sample_015_merged           0/0           0/0           0/0           0/0  \n",
       "G1_Sample_017_merged           0/0           0/0           0/0           0/0  \n",
       "G1_Sample_021_merged           0/0           0/0           0/0           0/0  \n",
       "G1_Sample_022_merged           0/0           0/0           0/0           0/0  \n",
       "G1_Sample_024_merged           0/0           0/0           0/0           0/0  \n",
       "G2_Sample_11_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_12_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_33_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_34_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_35_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_36_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_37_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_38_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_39_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_40_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_41_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_42_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_43_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_44_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_46_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_47_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_48_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_49_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_50_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_51_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_52_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_53_merged            0/0           0/0           0/0           0/0  \n",
       "G2_Sample_5_merged             0/0           0/0           0/0           0/0  \n",
       "G3_Sample_1_merged             0/0           0/0           0/0           0/0  \n",
       "G3_Sample_2_merged             0/0           0/0           0/0           0/0  \n",
       "G3_Sample_3_merged             0/0           0/1           0/0           1/1  \n",
       "G3_Sample_4_merged             0/0           0/0           0/0           0/0  \n",
       "G3_Sample_5_merged             0/0           0/0           0/0           0/0  \n",
       "G3_Sample_6_merged             0/0           0/0           0/0           0/0  \n",
       "G3_Sample_7_merged             0/0           0/0           0/0           0/0  \n",
       "G3_Sample_8_merged             1/1           0/0           1/1           1/1  \n",
       "G3_Sample_9_merged             0/0           0/0           0/0           0/0  \n",
       "\n",
       "[41 rows x 569650 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## from my point of view the null values could be replaced by the genotype '0/0'\n",
    "## as an example to explain the SNP = chr1-10002921\texists only in sample 'G2_Sample_42_SNPs' and NAN in all others\n",
    "## since it does not appear in others this means that their genotype at thie exact SNP are homozygous wildtype '0/0'\n",
    "## and thus were not called by the variant caller in those samples so we replase them by '0/0'\n",
    "\n",
    "G1_G2_G3_NoNull = G1_G2_G3_Sorted.fillna('0/0')\n",
    "G1_G2_G3_NoNull"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "38b2e397",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([dtype('O')], dtype=object)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "G1_G2_G3_NoNull.dtypes.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "1c2681a6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(41, 569650)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "G1_G2_G3_NoNull.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "255302a1",
   "metadata": {},
   "source": [
    "> ## Save the resulted All_groups_RAW Genotype DataFrame in a .csv format "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae5d57a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "All_groups_RAW_DataFrames = '/home/phoebe/phd/ML_csv/'\n",
    "file_name = 'All_groups_RAW_df.csv'\n",
    "file_out_path = os.path.join(All_groups_RAW_DataFrames , file_name)\n",
    "G1_G2_G3_NoNull.to_csv(file_out_path, index=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aaae9200",
   "metadata": {},
   "outputs": [],
   "source": [
    "All_groups_RAW_df = pd.read_csv('/home/phoebe/phd/ML_csv/All_groups_RAW_df.csv')\n",
    "All_groups_RAW_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a4a830e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "All_groups_RAW_df.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "036be5e4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "006f970c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "ee080bac",
   "metadata": {},
   "source": [
    "# Concatenate G1_raw_genotype & G2_raw_genotype "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "0d607eca",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 13 s, sys: 0 ns, total: 13 s\n",
      "Wall time: 13 s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>SNP_ID</th>\n",
       "      <th>chr1-156785617</th>\n",
       "      <th>chr1-156846233</th>\n",
       "      <th>chr1-156848995</th>\n",
       "      <th>chr1-201285759</th>\n",
       "      <th>chr1-201285909</th>\n",
       "      <th>chr1-201286678</th>\n",
       "      <th>chr1-201286771</th>\n",
       "      <th>chr1-201288007</th>\n",
       "      <th>chr1-201291066</th>\n",
       "      <th>chr1-201293617</th>\n",
       "      <th>...</th>\n",
       "      <th>chrX-114658525</th>\n",
       "      <th>chrX-119419850</th>\n",
       "      <th>chrX-138698651</th>\n",
       "      <th>chrX-140802529</th>\n",
       "      <th>chrX-151290013</th>\n",
       "      <th>chrX-152928065</th>\n",
       "      <th>chrX-152939900</th>\n",
       "      <th>chrX-152980909</th>\n",
       "      <th>chrX-153551757</th>\n",
       "      <th>chrX-153664247</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G1_Sample_006_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_007_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_017_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_013_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_022_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_024_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_021_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_015_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_014_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_50_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_43_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_48_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_33_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_42_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_44_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_5_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_39_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_41_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_49_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_51_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_53_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_40_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_36_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_47_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_38_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_52_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_37_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_11_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_12_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_35_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_34_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_46_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32 rows × 169579 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "SNP_ID               chr1-156785617 chr1-156846233 chr1-156848995  \\\n",
       "G1_Sample_006_merged            0/1            0/1            0/1   \n",
       "G1_Sample_007_merged            1/1            NaN            0/1   \n",
       "G1_Sample_017_merged            1/1            0/1            NaN   \n",
       "G1_Sample_013_merged            1/1            NaN            0/1   \n",
       "G1_Sample_022_merged            1/1            NaN            1/1   \n",
       "G1_Sample_024_merged            0/1            NaN            1/1   \n",
       "G1_Sample_021_merged            1/1            0/1            0/1   \n",
       "G1_Sample_015_merged            1/1            NaN            1/1   \n",
       "G1_Sample_014_merged            0/1            NaN            0/1   \n",
       "G2_Sample_50_merged             1/1            NaN            1/1   \n",
       "G2_Sample_43_merged             1/1            NaN            1/1   \n",
       "G2_Sample_48_merged             0/1            NaN            1/1   \n",
       "G2_Sample_33_merged             1/1            NaN            1/1   \n",
       "G2_Sample_42_merged             NaN            0/1            1/1   \n",
       "G2_Sample_44_merged             0/1            NaN            NaN   \n",
       "G2_Sample_5_merged              1/1            1/1            NaN   \n",
       "G2_Sample_39_merged             1/1            NaN            0/1   \n",
       "G2_Sample_41_merged             1/1            NaN            0/1   \n",
       "G2_Sample_49_merged             1/1            NaN            0/1   \n",
       "G2_Sample_51_merged             0/1            NaN            NaN   \n",
       "G2_Sample_53_merged             NaN            0/1            0/1   \n",
       "G2_Sample_40_merged             1/1            0/1            0/1   \n",
       "G2_Sample_36_merged             0/1            NaN            0/1   \n",
       "G2_Sample_47_merged             NaN            0/1            0/1   \n",
       "G2_Sample_38_merged             0/1            NaN            1/1   \n",
       "G2_Sample_52_merged             0/1            0/1            0/1   \n",
       "G2_Sample_37_merged             1/1            NaN            0/1   \n",
       "G2_Sample_11_merged             1/1            1/1            NaN   \n",
       "G2_Sample_12_merged             1/1            1/1            NaN   \n",
       "G2_Sample_35_merged             NaN            0/1            0/1   \n",
       "G2_Sample_34_merged             1/1            NaN            1/1   \n",
       "G2_Sample_46_merged             0/1            0/1            0/1   \n",
       "\n",
       "SNP_ID               chr1-201285759 chr1-201285909 chr1-201286678  \\\n",
       "G1_Sample_006_merged            1/1            1/1            0/1   \n",
       "G1_Sample_007_merged            0/1            0/1            NaN   \n",
       "G1_Sample_017_merged            1/1            1/1            NaN   \n",
       "G1_Sample_013_merged            1/1            1/1            NaN   \n",
       "G1_Sample_022_merged            0/1            0/1            NaN   \n",
       "G1_Sample_024_merged            1/1            1/1            NaN   \n",
       "G1_Sample_021_merged            1/1            0/1            0/1   \n",
       "G1_Sample_015_merged            1/1            0/1            0/1   \n",
       "G1_Sample_014_merged            1/1            1/1            NaN   \n",
       "G2_Sample_50_merged             1/1            1/1            NaN   \n",
       "G2_Sample_43_merged             1/1            0/1            0/1   \n",
       "G2_Sample_48_merged             1/1            0/1            0/1   \n",
       "G2_Sample_33_merged             1/1            0/1            0/1   \n",
       "G2_Sample_42_merged             1/1            0/1            0/1   \n",
       "G2_Sample_44_merged             1/1            1/1            NaN   \n",
       "G2_Sample_5_merged              1/1            1/1            NaN   \n",
       "G2_Sample_39_merged             1/1            1/1            NaN   \n",
       "G2_Sample_41_merged             1/1            NaN            1/1   \n",
       "G2_Sample_49_merged             1/1            1/1            NaN   \n",
       "G2_Sample_51_merged             1/1            1/1            NaN   \n",
       "G2_Sample_53_merged             1/1            1/1            NaN   \n",
       "G2_Sample_40_merged             1/1            1/1            NaN   \n",
       "G2_Sample_36_merged             1/1            1/1            NaN   \n",
       "G2_Sample_47_merged             1/1            1/1            NaN   \n",
       "G2_Sample_38_merged             1/1            1/1            0/1   \n",
       "G2_Sample_52_merged             1/1            1/1            NaN   \n",
       "G2_Sample_37_merged             1/1            1/1            NaN   \n",
       "G2_Sample_11_merged             1/1            0/1            0/1   \n",
       "G2_Sample_12_merged             1/1            0/1            0/1   \n",
       "G2_Sample_35_merged             1/1            1/1            NaN   \n",
       "G2_Sample_34_merged             1/1            1/1            0/1   \n",
       "G2_Sample_46_merged             1/1            NaN            1/1   \n",
       "\n",
       "SNP_ID               chr1-201286771 chr1-201288007 chr1-201291066  \\\n",
       "G1_Sample_006_merged            0/1            1/1            0/1   \n",
       "G1_Sample_007_merged            NaN            0/1            0/1   \n",
       "G1_Sample_017_merged            NaN            1/1            1/1   \n",
       "G1_Sample_013_merged            NaN            0/1            0/1   \n",
       "G1_Sample_022_merged            NaN            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN            1/1            1/1   \n",
       "G1_Sample_021_merged            0/1            1/1            0/1   \n",
       "G1_Sample_015_merged            0/1            0/1            NaN   \n",
       "G1_Sample_014_merged            NaN            0/1            0/1   \n",
       "G2_Sample_50_merged             NaN            NaN            NaN   \n",
       "G2_Sample_43_merged             0/1            1/1            0/1   \n",
       "G2_Sample_48_merged             0/1            1/1            0/1   \n",
       "G2_Sample_33_merged             0/1            1/1            0/1   \n",
       "G2_Sample_42_merged             0/1            1/1            0/1   \n",
       "G2_Sample_44_merged             NaN            1/1            1/1   \n",
       "G2_Sample_5_merged              NaN            1/1            1/1   \n",
       "G2_Sample_39_merged             NaN            1/1            1/1   \n",
       "G2_Sample_41_merged             1/1            1/1            NaN   \n",
       "G2_Sample_49_merged             NaN            1/1            1/1   \n",
       "G2_Sample_51_merged             NaN            NaN            1/1   \n",
       "G2_Sample_53_merged             NaN            1/1            1/1   \n",
       "G2_Sample_40_merged             NaN            0/1            0/1   \n",
       "G2_Sample_36_merged             NaN            1/1            1/1   \n",
       "G2_Sample_47_merged             NaN            1/1            1/1   \n",
       "G2_Sample_38_merged             0/1            1/1            0/1   \n",
       "G2_Sample_52_merged             NaN            0/1            0/1   \n",
       "G2_Sample_37_merged             NaN            1/1            1/1   \n",
       "G2_Sample_11_merged             0/1            1/1            0/1   \n",
       "G2_Sample_12_merged             0/1            1/1            0/1   \n",
       "G2_Sample_35_merged             NaN            1/1            1/1   \n",
       "G2_Sample_34_merged             0/1            NaN            NaN   \n",
       "G2_Sample_46_merged             1/1            1/1            NaN   \n",
       "\n",
       "SNP_ID               chr1-201293617  ... chrX-114658525 chrX-119419850  \\\n",
       "G1_Sample_006_merged            0/1  ...            NaN            NaN   \n",
       "G1_Sample_007_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_017_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_013_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_022_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_021_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_015_merged            NaN  ...            NaN            NaN   \n",
       "G1_Sample_014_merged            NaN  ...            NaN            NaN   \n",
       "G2_Sample_50_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_43_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_48_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_33_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_42_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_44_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_5_merged              NaN  ...            NaN            NaN   \n",
       "G2_Sample_39_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_41_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_49_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_51_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_53_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_40_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_36_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_47_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_38_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_52_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_37_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_11_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_12_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_35_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_34_merged             NaN  ...            NaN            NaN   \n",
       "G2_Sample_46_merged             NaN  ...            1/1            1/1   \n",
       "\n",
       "SNP_ID               chrX-138698651 chrX-140802529 chrX-151290013  \\\n",
       "G1_Sample_006_merged            NaN            NaN            NaN   \n",
       "G1_Sample_007_merged            NaN            NaN            NaN   \n",
       "G1_Sample_017_merged            NaN            NaN            NaN   \n",
       "G1_Sample_013_merged            NaN            NaN            NaN   \n",
       "G1_Sample_022_merged            NaN            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN            NaN            NaN   \n",
       "G1_Sample_021_merged            NaN            NaN            NaN   \n",
       "G1_Sample_015_merged            NaN            NaN            NaN   \n",
       "G1_Sample_014_merged            NaN            NaN            NaN   \n",
       "G2_Sample_50_merged             NaN            NaN            NaN   \n",
       "G2_Sample_43_merged             NaN            NaN            NaN   \n",
       "G2_Sample_48_merged             NaN            NaN            NaN   \n",
       "G2_Sample_33_merged             NaN            NaN            NaN   \n",
       "G2_Sample_42_merged             NaN            NaN            NaN   \n",
       "G2_Sample_44_merged             NaN            NaN            NaN   \n",
       "G2_Sample_5_merged              NaN            NaN            NaN   \n",
       "G2_Sample_39_merged             NaN            NaN            NaN   \n",
       "G2_Sample_41_merged             NaN            NaN            NaN   \n",
       "G2_Sample_49_merged             NaN            NaN            NaN   \n",
       "G2_Sample_51_merged             NaN            NaN            NaN   \n",
       "G2_Sample_53_merged             NaN            NaN            NaN   \n",
       "G2_Sample_40_merged             NaN            NaN            NaN   \n",
       "G2_Sample_36_merged             NaN            NaN            NaN   \n",
       "G2_Sample_47_merged             NaN            NaN            NaN   \n",
       "G2_Sample_38_merged             NaN            NaN            NaN   \n",
       "G2_Sample_52_merged             NaN            NaN            NaN   \n",
       "G2_Sample_37_merged             NaN            NaN            NaN   \n",
       "G2_Sample_11_merged             NaN            NaN            NaN   \n",
       "G2_Sample_12_merged             NaN            NaN            NaN   \n",
       "G2_Sample_35_merged             NaN            NaN            NaN   \n",
       "G2_Sample_34_merged             NaN            NaN            NaN   \n",
       "G2_Sample_46_merged             1/1            1/1            1/1   \n",
       "\n",
       "SNP_ID               chrX-152928065 chrX-152939900 chrX-152980909  \\\n",
       "G1_Sample_006_merged            NaN            NaN            NaN   \n",
       "G1_Sample_007_merged            NaN            NaN            NaN   \n",
       "G1_Sample_017_merged            NaN            NaN            NaN   \n",
       "G1_Sample_013_merged            NaN            NaN            NaN   \n",
       "G1_Sample_022_merged            NaN            NaN            NaN   \n",
       "G1_Sample_024_merged            NaN            NaN            NaN   \n",
       "G1_Sample_021_merged            NaN            NaN            NaN   \n",
       "G1_Sample_015_merged            NaN            NaN            NaN   \n",
       "G1_Sample_014_merged            NaN            NaN            NaN   \n",
       "G2_Sample_50_merged             NaN            NaN            NaN   \n",
       "G2_Sample_43_merged             NaN            NaN            NaN   \n",
       "G2_Sample_48_merged             NaN            NaN            NaN   \n",
       "G2_Sample_33_merged             NaN            NaN            NaN   \n",
       "G2_Sample_42_merged             NaN            NaN            NaN   \n",
       "G2_Sample_44_merged             NaN            NaN            NaN   \n",
       "G2_Sample_5_merged              NaN            NaN            NaN   \n",
       "G2_Sample_39_merged             NaN            NaN            NaN   \n",
       "G2_Sample_41_merged             NaN            NaN            NaN   \n",
       "G2_Sample_49_merged             NaN            NaN            NaN   \n",
       "G2_Sample_51_merged             NaN            NaN            NaN   \n",
       "G2_Sample_53_merged             NaN            NaN            NaN   \n",
       "G2_Sample_40_merged             NaN            NaN            NaN   \n",
       "G2_Sample_36_merged             NaN            NaN            NaN   \n",
       "G2_Sample_47_merged             NaN            NaN            NaN   \n",
       "G2_Sample_38_merged             NaN            NaN            NaN   \n",
       "G2_Sample_52_merged             NaN            NaN            NaN   \n",
       "G2_Sample_37_merged             NaN            NaN            NaN   \n",
       "G2_Sample_11_merged             NaN            NaN            NaN   \n",
       "G2_Sample_12_merged             NaN            NaN            NaN   \n",
       "G2_Sample_35_merged             NaN            NaN            NaN   \n",
       "G2_Sample_34_merged             NaN            NaN            NaN   \n",
       "G2_Sample_46_merged             1/1            1/1            1/1   \n",
       "\n",
       "SNP_ID               chrX-153551757 chrX-153664247  \n",
       "G1_Sample_006_merged            NaN            NaN  \n",
       "G1_Sample_007_merged            NaN            NaN  \n",
       "G1_Sample_017_merged            NaN            NaN  \n",
       "G1_Sample_013_merged            NaN            NaN  \n",
       "G1_Sample_022_merged            NaN            NaN  \n",
       "G1_Sample_024_merged            NaN            NaN  \n",
       "G1_Sample_021_merged            NaN            NaN  \n",
       "G1_Sample_015_merged            NaN            NaN  \n",
       "G1_Sample_014_merged            NaN            NaN  \n",
       "G2_Sample_50_merged             NaN            NaN  \n",
       "G2_Sample_43_merged             NaN            NaN  \n",
       "G2_Sample_48_merged             NaN            NaN  \n",
       "G2_Sample_33_merged             NaN            NaN  \n",
       "G2_Sample_42_merged             NaN            NaN  \n",
       "G2_Sample_44_merged             NaN            NaN  \n",
       "G2_Sample_5_merged              NaN            NaN  \n",
       "G2_Sample_39_merged             NaN            NaN  \n",
       "G2_Sample_41_merged             NaN            NaN  \n",
       "G2_Sample_49_merged             NaN            NaN  \n",
       "G2_Sample_51_merged             NaN            NaN  \n",
       "G2_Sample_53_merged             NaN            NaN  \n",
       "G2_Sample_40_merged             NaN            NaN  \n",
       "G2_Sample_36_merged             NaN            NaN  \n",
       "G2_Sample_47_merged             NaN            NaN  \n",
       "G2_Sample_38_merged             NaN            NaN  \n",
       "G2_Sample_52_merged             NaN            NaN  \n",
       "G2_Sample_37_merged             NaN            NaN  \n",
       "G2_Sample_11_merged             NaN            NaN  \n",
       "G2_Sample_12_merged             NaN            NaN  \n",
       "G2_Sample_35_merged             NaN            NaN  \n",
       "G2_Sample_34_merged             NaN            NaN  \n",
       "G2_Sample_46_merged             1/1            1/1  \n",
       "\n",
       "[32 rows x 169579 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "## Dataframe containing all Genotypes of All SNPs in all samples in G1 + G2\n",
    "\n",
    "G1_G2 = pd.concat([G1_raw_genotype, G2_raw_genotype],axis=0)\n",
    "G1_G2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "006d2530",
   "metadata": {},
   "source": [
    "> ## Sort by rows and by columns "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "c777b9e3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>SNP_ID</th>\n",
       "      <th>chr1-10002921</th>\n",
       "      <th>chr1-100058793</th>\n",
       "      <th>chr1-10007418</th>\n",
       "      <th>chr1-1000966</th>\n",
       "      <th>chr1-1001028</th>\n",
       "      <th>chr1-100111956</th>\n",
       "      <th>chr1-1001177</th>\n",
       "      <th>chr1-1001233</th>\n",
       "      <th>chr1-1001270</th>\n",
       "      <th>chr1-1001290</th>\n",
       "      <th>...</th>\n",
       "      <th>chrX-99943560</th>\n",
       "      <th>chrX-99945176</th>\n",
       "      <th>chrX-99945292</th>\n",
       "      <th>chrX-99945494</th>\n",
       "      <th>chrX-99945545</th>\n",
       "      <th>chrX-99946267</th>\n",
       "      <th>chrX-99946368</th>\n",
       "      <th>chrX-99955925</th>\n",
       "      <th>chrX-99956157</th>\n",
       "      <th>chrX-99957136</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G1_Sample_006_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_007_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_013_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_014_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_015_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_017_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_021_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_022_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_024_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_11_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_12_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_33_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_34_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_35_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_36_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_37_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_38_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_39_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_40_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_41_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_42_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_43_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_44_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_46_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_47_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_48_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_49_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_50_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_51_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_52_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_53_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_5_merged</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32 rows × 169579 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "SNP_ID               chr1-10002921 chr1-100058793 chr1-10007418 chr1-1000966  \\\n",
       "G1_Sample_006_merged           NaN            NaN           NaN          NaN   \n",
       "G1_Sample_007_merged           NaN            NaN           NaN          NaN   \n",
       "G1_Sample_013_merged           NaN            NaN           NaN          NaN   \n",
       "G1_Sample_014_merged           NaN            NaN           NaN          NaN   \n",
       "G1_Sample_015_merged           NaN            NaN           NaN          NaN   \n",
       "G1_Sample_017_merged           NaN            NaN           NaN          NaN   \n",
       "G1_Sample_021_merged           NaN            NaN           NaN          NaN   \n",
       "G1_Sample_022_merged           NaN            NaN           NaN          NaN   \n",
       "G1_Sample_024_merged           NaN            NaN           NaN          NaN   \n",
       "G2_Sample_11_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_12_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_33_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_34_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_35_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_36_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_37_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_38_merged            NaN            1/1           NaN          NaN   \n",
       "G2_Sample_39_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_40_merged            NaN            NaN           0/1          NaN   \n",
       "G2_Sample_41_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_42_merged            0/1            NaN           NaN          NaN   \n",
       "G2_Sample_43_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_44_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_46_merged            NaN            NaN           NaN          0/1   \n",
       "G2_Sample_47_merged            NaN            0/1           NaN          NaN   \n",
       "G2_Sample_48_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_49_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_50_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_51_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_52_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_53_merged            NaN            NaN           NaN          NaN   \n",
       "G2_Sample_5_merged             NaN            NaN           NaN          NaN   \n",
       "\n",
       "SNP_ID               chr1-1001028 chr1-100111956 chr1-1001177 chr1-1001233  \\\n",
       "G1_Sample_006_merged          NaN            NaN          NaN          NaN   \n",
       "G1_Sample_007_merged          NaN            NaN          NaN          NaN   \n",
       "G1_Sample_013_merged          NaN            NaN          NaN          NaN   \n",
       "G1_Sample_014_merged          NaN            NaN          NaN          NaN   \n",
       "G1_Sample_015_merged          NaN            NaN          NaN          NaN   \n",
       "G1_Sample_017_merged          NaN            NaN          NaN          NaN   \n",
       "G1_Sample_021_merged          NaN            NaN          NaN          NaN   \n",
       "G1_Sample_022_merged          NaN            NaN          NaN          NaN   \n",
       "G1_Sample_024_merged          NaN            NaN          NaN          NaN   \n",
       "G2_Sample_11_merged           NaN            NaN          1/1          0/1   \n",
       "G2_Sample_12_merged           NaN            NaN          0/1          NaN   \n",
       "G2_Sample_33_merged           NaN            0/1          1/1          NaN   \n",
       "G2_Sample_34_merged           NaN            NaN          NaN          NaN   \n",
       "G2_Sample_35_merged           NaN            NaN          NaN          NaN   \n",
       "G2_Sample_36_merged           NaN            NaN          0/1          NaN   \n",
       "G2_Sample_37_merged           NaN            NaN          1/1          NaN   \n",
       "G2_Sample_38_merged           NaN            0/1          1/1          NaN   \n",
       "G2_Sample_39_merged           NaN            1/1          0/1          NaN   \n",
       "G2_Sample_40_merged           NaN            NaN          0/1          NaN   \n",
       "G2_Sample_41_merged           NaN            0/1          1/1          NaN   \n",
       "G2_Sample_42_merged           NaN            1/1          1/1          NaN   \n",
       "G2_Sample_43_merged           NaN            0/1          1/1          NaN   \n",
       "G2_Sample_44_merged           NaN            NaN          NaN          NaN   \n",
       "G2_Sample_46_merged           NaN            0/1          0/1          NaN   \n",
       "G2_Sample_47_merged           NaN            NaN          0/1          NaN   \n",
       "G2_Sample_48_merged           NaN            0/1          1/1          NaN   \n",
       "G2_Sample_49_merged           NaN            NaN          NaN          NaN   \n",
       "G2_Sample_50_merged           NaN            NaN          1/1          NaN   \n",
       "G2_Sample_51_merged           NaN            NaN          0/1          NaN   \n",
       "G2_Sample_52_merged           NaN            0/1          0/1          NaN   \n",
       "G2_Sample_53_merged           NaN            0/1          0/1          NaN   \n",
       "G2_Sample_5_merged            0/1            NaN          NaN          NaN   \n",
       "\n",
       "SNP_ID               chr1-1001270 chr1-1001290  ... chrX-99943560  \\\n",
       "G1_Sample_006_merged          NaN          NaN  ...           NaN   \n",
       "G1_Sample_007_merged          NaN          NaN  ...           NaN   \n",
       "G1_Sample_013_merged          NaN          NaN  ...           NaN   \n",
       "G1_Sample_014_merged          NaN          NaN  ...           NaN   \n",
       "G1_Sample_015_merged          NaN          NaN  ...           NaN   \n",
       "G1_Sample_017_merged          NaN          NaN  ...           NaN   \n",
       "G1_Sample_021_merged          NaN          NaN  ...           NaN   \n",
       "G1_Sample_022_merged          NaN          NaN  ...           NaN   \n",
       "G1_Sample_024_merged          NaN          NaN  ...           NaN   \n",
       "G2_Sample_11_merged           NaN          NaN  ...           0/1   \n",
       "G2_Sample_12_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_33_merged           NaN          0/1  ...           NaN   \n",
       "G2_Sample_34_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_35_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_36_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_37_merged           NaN          0/1  ...           NaN   \n",
       "G2_Sample_38_merged           NaN          0/1  ...           NaN   \n",
       "G2_Sample_39_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_40_merged           NaN          0/1  ...           NaN   \n",
       "G2_Sample_41_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_42_merged           NaN          0/1  ...           NaN   \n",
       "G2_Sample_43_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_44_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_46_merged           NaN          0/1  ...           NaN   \n",
       "G2_Sample_47_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_48_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_49_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_50_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_51_merged           0/1          NaN  ...           NaN   \n",
       "G2_Sample_52_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_53_merged           NaN          NaN  ...           NaN   \n",
       "G2_Sample_5_merged            NaN          NaN  ...           NaN   \n",
       "\n",
       "SNP_ID               chrX-99945176 chrX-99945292 chrX-99945494 chrX-99945545  \\\n",
       "G1_Sample_006_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_007_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_013_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_014_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_015_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_017_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_021_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_022_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_024_merged           NaN           NaN           NaN           NaN   \n",
       "G2_Sample_11_merged            0/1           NaN           0/1           NaN   \n",
       "G2_Sample_12_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_33_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_34_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_35_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_36_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_37_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_38_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_39_merged            1/1           NaN           NaN           NaN   \n",
       "G2_Sample_40_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_41_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_42_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_43_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_44_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_46_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_47_merged            0/1           NaN           NaN           0/1   \n",
       "G2_Sample_48_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_49_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_50_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_51_merged            0/1           0/1           0/1           NaN   \n",
       "G2_Sample_52_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_53_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_5_merged             NaN           NaN           NaN           NaN   \n",
       "\n",
       "SNP_ID               chrX-99946267 chrX-99946368 chrX-99955925 chrX-99956157  \\\n",
       "G1_Sample_006_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_007_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_013_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_014_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_015_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_017_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_021_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_022_merged           NaN           NaN           NaN           NaN   \n",
       "G1_Sample_024_merged           NaN           NaN           NaN           NaN   \n",
       "G2_Sample_11_merged            NaN           NaN           0/1           0/1   \n",
       "G2_Sample_12_merged            NaN           NaN           NaN           0/1   \n",
       "G2_Sample_33_merged            NaN           NaN           NaN           1/1   \n",
       "G2_Sample_34_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_35_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_36_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_37_merged            NaN           NaN           NaN           1/1   \n",
       "G2_Sample_38_merged            NaN           NaN           NaN           1/1   \n",
       "G2_Sample_39_merged            NaN           NaN           NaN           1/1   \n",
       "G2_Sample_40_merged            NaN           NaN           NaN           1/1   \n",
       "G2_Sample_41_merged            NaN           NaN           NaN           0/1   \n",
       "G2_Sample_42_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_43_merged            NaN           NaN           NaN           1/1   \n",
       "G2_Sample_44_merged            NaN           NaN           NaN           1/1   \n",
       "G2_Sample_46_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_47_merged            0/1           0/1           0/1           NaN   \n",
       "G2_Sample_48_merged            NaN           NaN           NaN           0/1   \n",
       "G2_Sample_49_merged            NaN           NaN           NaN           0/1   \n",
       "G2_Sample_50_merged            NaN           NaN           NaN           1/1   \n",
       "G2_Sample_51_merged            NaN           NaN           0/1           NaN   \n",
       "G2_Sample_52_merged            NaN           NaN           NaN           NaN   \n",
       "G2_Sample_53_merged            NaN           NaN           NaN           1/1   \n",
       "G2_Sample_5_merged             NaN           NaN           NaN           0/1   \n",
       "\n",
       "SNP_ID               chrX-99957136  \n",
       "G1_Sample_006_merged           NaN  \n",
       "G1_Sample_007_merged           NaN  \n",
       "G1_Sample_013_merged           NaN  \n",
       "G1_Sample_014_merged           NaN  \n",
       "G1_Sample_015_merged           NaN  \n",
       "G1_Sample_017_merged           NaN  \n",
       "G1_Sample_021_merged           NaN  \n",
       "G1_Sample_022_merged           NaN  \n",
       "G1_Sample_024_merged           NaN  \n",
       "G2_Sample_11_merged            NaN  \n",
       "G2_Sample_12_merged            0/1  \n",
       "G2_Sample_33_merged            NaN  \n",
       "G2_Sample_34_merged            NaN  \n",
       "G2_Sample_35_merged            NaN  \n",
       "G2_Sample_36_merged            NaN  \n",
       "G2_Sample_37_merged            1/1  \n",
       "G2_Sample_38_merged            1/1  \n",
       "G2_Sample_39_merged            NaN  \n",
       "G2_Sample_40_merged            1/1  \n",
       "G2_Sample_41_merged            0/1  \n",
       "G2_Sample_42_merged            NaN  \n",
       "G2_Sample_43_merged            1/1  \n",
       "G2_Sample_44_merged            1/1  \n",
       "G2_Sample_46_merged            NaN  \n",
       "G2_Sample_47_merged            NaN  \n",
       "G2_Sample_48_merged            0/1  \n",
       "G2_Sample_49_merged            0/1  \n",
       "G2_Sample_50_merged            1/1  \n",
       "G2_Sample_51_merged            NaN  \n",
       "G2_Sample_52_merged            NaN  \n",
       "G2_Sample_53_merged            NaN  \n",
       "G2_Sample_5_merged             0/1  \n",
       "\n",
       "[32 rows x 169579 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Sort df by row index label\n",
    "G1_G2_Sorted = G1_G2.sort_index(axis=0)\n",
    "\n",
    "## Sort df by column index label\n",
    "G1_G2_Sorted = G1_G2_Sorted.sort_index(axis=1)\n",
    "G1_G2_Sorted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "624f08b0",
   "metadata": {},
   "source": [
    "> ## Replace nan values in Genotype DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "a3888213",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>SNP_ID</th>\n",
       "      <th>chr1-10002921</th>\n",
       "      <th>chr1-100058793</th>\n",
       "      <th>chr1-10007418</th>\n",
       "      <th>chr1-1000966</th>\n",
       "      <th>chr1-1001028</th>\n",
       "      <th>chr1-100111956</th>\n",
       "      <th>chr1-1001177</th>\n",
       "      <th>chr1-1001233</th>\n",
       "      <th>chr1-1001270</th>\n",
       "      <th>chr1-1001290</th>\n",
       "      <th>...</th>\n",
       "      <th>chrX-99943560</th>\n",
       "      <th>chrX-99945176</th>\n",
       "      <th>chrX-99945292</th>\n",
       "      <th>chrX-99945494</th>\n",
       "      <th>chrX-99945545</th>\n",
       "      <th>chrX-99946267</th>\n",
       "      <th>chrX-99946368</th>\n",
       "      <th>chrX-99955925</th>\n",
       "      <th>chrX-99956157</th>\n",
       "      <th>chrX-99957136</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G1_Sample_006_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_007_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_013_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_014_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_015_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_017_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_021_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_022_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_024_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_11_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_12_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_33_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_34_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_35_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_36_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_37_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_38_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_39_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_40_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_41_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_42_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_43_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_44_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_46_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_47_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_48_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_49_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_50_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_51_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_52_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_53_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_5_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32 rows × 169579 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "SNP_ID               chr1-10002921 chr1-100058793 chr1-10007418 chr1-1000966  \\\n",
       "G1_Sample_006_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_007_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_013_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_014_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_015_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_017_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_021_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_022_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_024_merged           0/0            0/0           0/0          0/0   \n",
       "G2_Sample_11_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_12_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_33_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_34_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_35_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_36_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_37_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_38_merged            0/0            1/1           0/0          0/0   \n",
       "G2_Sample_39_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_40_merged            0/0            0/0           0/1          0/0   \n",
       "G2_Sample_41_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_42_merged            0/1            0/0           0/0          0/0   \n",
       "G2_Sample_43_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_44_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_46_merged            0/0            0/0           0/0          0/1   \n",
       "G2_Sample_47_merged            0/0            0/1           0/0          0/0   \n",
       "G2_Sample_48_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_49_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_50_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_51_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_52_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_53_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_5_merged             0/0            0/0           0/0          0/0   \n",
       "\n",
       "SNP_ID               chr1-1001028 chr1-100111956 chr1-1001177 chr1-1001233  \\\n",
       "G1_Sample_006_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_007_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_013_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_014_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_015_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_017_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_021_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_022_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_024_merged          0/0            0/0          0/0          0/0   \n",
       "G2_Sample_11_merged           0/0            0/0          1/1          0/1   \n",
       "G2_Sample_12_merged           0/0            0/0          0/1          0/0   \n",
       "G2_Sample_33_merged           0/0            0/1          1/1          0/0   \n",
       "G2_Sample_34_merged           0/0            0/0          0/0          0/0   \n",
       "G2_Sample_35_merged           0/0            0/0          0/0          0/0   \n",
       "G2_Sample_36_merged           0/0            0/0          0/1          0/0   \n",
       "G2_Sample_37_merged           0/0            0/0          1/1          0/0   \n",
       "G2_Sample_38_merged           0/0            0/1          1/1          0/0   \n",
       "G2_Sample_39_merged           0/0            1/1          0/1          0/0   \n",
       "G2_Sample_40_merged           0/0            0/0          0/1          0/0   \n",
       "G2_Sample_41_merged           0/0            0/1          1/1          0/0   \n",
       "G2_Sample_42_merged           0/0            1/1          1/1          0/0   \n",
       "G2_Sample_43_merged           0/0            0/1          1/1          0/0   \n",
       "G2_Sample_44_merged           0/0            0/0          0/0          0/0   \n",
       "G2_Sample_46_merged           0/0            0/1          0/1          0/0   \n",
       "G2_Sample_47_merged           0/0            0/0          0/1          0/0   \n",
       "G2_Sample_48_merged           0/0            0/1          1/1          0/0   \n",
       "G2_Sample_49_merged           0/0            0/0          0/0          0/0   \n",
       "G2_Sample_50_merged           0/0            0/0          1/1          0/0   \n",
       "G2_Sample_51_merged           0/0            0/0          0/1          0/0   \n",
       "G2_Sample_52_merged           0/0            0/1          0/1          0/0   \n",
       "G2_Sample_53_merged           0/0            0/1          0/1          0/0   \n",
       "G2_Sample_5_merged            0/1            0/0          0/0          0/0   \n",
       "\n",
       "SNP_ID               chr1-1001270 chr1-1001290  ... chrX-99943560  \\\n",
       "G1_Sample_006_merged          0/0          0/0  ...           0/0   \n",
       "G1_Sample_007_merged          0/0          0/0  ...           0/0   \n",
       "G1_Sample_013_merged          0/0          0/0  ...           0/0   \n",
       "G1_Sample_014_merged          0/0          0/0  ...           0/0   \n",
       "G1_Sample_015_merged          0/0          0/0  ...           0/0   \n",
       "G1_Sample_017_merged          0/0          0/0  ...           0/0   \n",
       "G1_Sample_021_merged          0/0          0/0  ...           0/0   \n",
       "G1_Sample_022_merged          0/0          0/0  ...           0/0   \n",
       "G1_Sample_024_merged          0/0          0/0  ...           0/0   \n",
       "G2_Sample_11_merged           0/0          0/0  ...           0/1   \n",
       "G2_Sample_12_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_33_merged           0/0          0/1  ...           0/0   \n",
       "G2_Sample_34_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_35_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_36_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_37_merged           0/0          0/1  ...           0/0   \n",
       "G2_Sample_38_merged           0/0          0/1  ...           0/0   \n",
       "G2_Sample_39_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_40_merged           0/0          0/1  ...           0/0   \n",
       "G2_Sample_41_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_42_merged           0/0          0/1  ...           0/0   \n",
       "G2_Sample_43_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_44_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_46_merged           0/0          0/1  ...           0/0   \n",
       "G2_Sample_47_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_48_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_49_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_50_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_51_merged           0/1          0/0  ...           0/0   \n",
       "G2_Sample_52_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_53_merged           0/0          0/0  ...           0/0   \n",
       "G2_Sample_5_merged            0/0          0/0  ...           0/0   \n",
       "\n",
       "SNP_ID               chrX-99945176 chrX-99945292 chrX-99945494 chrX-99945545  \\\n",
       "G1_Sample_006_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_007_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_013_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_014_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_015_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_017_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_021_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_022_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_024_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_11_merged            0/1           0/0           0/1           0/0   \n",
       "G2_Sample_12_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_33_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_34_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_35_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_36_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_37_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_38_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_39_merged            1/1           0/0           0/0           0/0   \n",
       "G2_Sample_40_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_41_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_42_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_43_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_44_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_46_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_47_merged            0/1           0/0           0/0           0/1   \n",
       "G2_Sample_48_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_49_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_50_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_51_merged            0/1           0/1           0/1           0/0   \n",
       "G2_Sample_52_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_53_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_5_merged             0/0           0/0           0/0           0/0   \n",
       "\n",
       "SNP_ID               chrX-99946267 chrX-99946368 chrX-99955925 chrX-99956157  \\\n",
       "G1_Sample_006_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_007_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_013_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_014_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_015_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_017_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_021_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_022_merged           0/0           0/0           0/0           0/0   \n",
       "G1_Sample_024_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_11_merged            0/0           0/0           0/1           0/1   \n",
       "G2_Sample_12_merged            0/0           0/0           0/0           0/1   \n",
       "G2_Sample_33_merged            0/0           0/0           0/0           1/1   \n",
       "G2_Sample_34_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_35_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_36_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_37_merged            0/0           0/0           0/0           1/1   \n",
       "G2_Sample_38_merged            0/0           0/0           0/0           1/1   \n",
       "G2_Sample_39_merged            0/0           0/0           0/0           1/1   \n",
       "G2_Sample_40_merged            0/0           0/0           0/0           1/1   \n",
       "G2_Sample_41_merged            0/0           0/0           0/0           0/1   \n",
       "G2_Sample_42_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_43_merged            0/0           0/0           0/0           1/1   \n",
       "G2_Sample_44_merged            0/0           0/0           0/0           1/1   \n",
       "G2_Sample_46_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_47_merged            0/1           0/1           0/1           0/0   \n",
       "G2_Sample_48_merged            0/0           0/0           0/0           0/1   \n",
       "G2_Sample_49_merged            0/0           0/0           0/0           0/1   \n",
       "G2_Sample_50_merged            0/0           0/0           0/0           1/1   \n",
       "G2_Sample_51_merged            0/0           0/0           0/1           0/0   \n",
       "G2_Sample_52_merged            0/0           0/0           0/0           0/0   \n",
       "G2_Sample_53_merged            0/0           0/0           0/0           1/1   \n",
       "G2_Sample_5_merged             0/0           0/0           0/0           0/1   \n",
       "\n",
       "SNP_ID               chrX-99957136  \n",
       "G1_Sample_006_merged           0/0  \n",
       "G1_Sample_007_merged           0/0  \n",
       "G1_Sample_013_merged           0/0  \n",
       "G1_Sample_014_merged           0/0  \n",
       "G1_Sample_015_merged           0/0  \n",
       "G1_Sample_017_merged           0/0  \n",
       "G1_Sample_021_merged           0/0  \n",
       "G1_Sample_022_merged           0/0  \n",
       "G1_Sample_024_merged           0/0  \n",
       "G2_Sample_11_merged            0/0  \n",
       "G2_Sample_12_merged            0/1  \n",
       "G2_Sample_33_merged            0/0  \n",
       "G2_Sample_34_merged            0/0  \n",
       "G2_Sample_35_merged            0/0  \n",
       "G2_Sample_36_merged            0/0  \n",
       "G2_Sample_37_merged            1/1  \n",
       "G2_Sample_38_merged            1/1  \n",
       "G2_Sample_39_merged            0/0  \n",
       "G2_Sample_40_merged            1/1  \n",
       "G2_Sample_41_merged            0/1  \n",
       "G2_Sample_42_merged            0/0  \n",
       "G2_Sample_43_merged            1/1  \n",
       "G2_Sample_44_merged            1/1  \n",
       "G2_Sample_46_merged            0/0  \n",
       "G2_Sample_47_merged            0/0  \n",
       "G2_Sample_48_merged            0/1  \n",
       "G2_Sample_49_merged            0/1  \n",
       "G2_Sample_50_merged            1/1  \n",
       "G2_Sample_51_merged            0/0  \n",
       "G2_Sample_52_merged            0/0  \n",
       "G2_Sample_53_merged            0/0  \n",
       "G2_Sample_5_merged             0/1  \n",
       "\n",
       "[32 rows x 169579 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## from my point of view the null values could be replaced by the genotype '0/0'\n",
    "## as an example to explain the SNP = chr1-10002921\texists only in sample 'G2_Sample_42_SNPs' and NAN in all others\n",
    "## since it does not appear in others this means that their genotype at thie exact SNP are homozygous wildtype '0/0'\n",
    "## and thus were not called by the variant caller in those samples so we replase them by '0/0'\n",
    "\n",
    "G1_G2_NoNull = G1_G2_Sorted.fillna('0/0')\n",
    "G1_G2_NoNull"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "44425857",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([dtype('O')], dtype=object)"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "G1_G2_NoNull.dtypes.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "bcbb9d14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(32, 169579)"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "G1_G2_NoNull.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc81973f",
   "metadata": {},
   "source": [
    "> ## Save the resulted G1_G2_RAW Genotype DataFrame in a .csv format "
   ]
  },
  {
   "cell_type": "raw",
   "id": "e240e463",
   "metadata": {},
   "source": [
    "%%time\n",
    "G1_G2_DataFrame = '/home/phoebe/phd/ML_csv/'\n",
    "file_name = 'G1_G2_df.csv'\n",
    "file_out_path = os.path.join(G1_G2_DataFrame , file_name)\n",
    "G1_G2_NoNull.to_csv(file_out_path, index=True)\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "d3657625",
   "metadata": {},
   "source": [
    "G1_G2_df = pd.read_csv('/home/phoebe/phd/ML_csv/G1_G2_df.csv')\n",
    "G1_G2_df\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "1bac8221",
   "metadata": {},
   "source": [
    "G1_G2_df.set_index('Unnamed: 0')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2bd6e155",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "ef9937f2",
   "metadata": {},
   "source": [
    "> ## Clinical Dataframe For all 3 Groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "861b906d",
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical_df = pd.read_csv('/home/phoebe/phd/ML_csv/Clinical_Data_G1_2_3_Two_outputs.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "54940af6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Sample_ID', 'Patient_ID', 'Unnamed: 2', 'Patient Name',\n",
       "       'Consanguinity', 'Sex', 'Age/Year', 'sweating', 'Scalp Hair', 'eyebrow',\n",
       "       ' receding anterior hair line', 'skin', 'Nails',\n",
       "       'no of missing deciduous teeth', 'no of missing permenant teeth',\n",
       "       'Missing teeth', 'Peg shape teeth', 'Y', 'Molecular'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e7d5ffad",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Patient Name</th>\n",
       "      <th>Consanguinity</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Age/Year</th>\n",
       "      <th>sweating</th>\n",
       "      <th>Scalp Hair</th>\n",
       "      <th>eyebrow</th>\n",
       "      <th>receding anterior hair line</th>\n",
       "      <th>skin</th>\n",
       "      <th>Nails</th>\n",
       "      <th>no of missing deciduous teeth</th>\n",
       "      <th>no of missing permenant teeth</th>\n",
       "      <th>Missing teeth</th>\n",
       "      <th>Peg shape teeth</th>\n",
       "      <th>Y</th>\n",
       "      <th>Molecular</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>fastq_006</td>\n",
       "      <td>G1_Sample_006_SNPs</td>\n",
       "      <td>G1_Sample_006</td>\n",
       "      <td>Donia Ahmed Mahmoud Ali</td>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>-</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>EDA\\nE7\\nc.871G&gt;A\\np.G291R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>fastq_007</td>\n",
       "      <td>G1_Sample_007_SNPs</td>\n",
       "      <td>G1_Sample_007</td>\n",
       "      <td>Malak Shadi</td>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>3.50</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>14.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>EDA\\nE2\\nc.492delA\\np.G165Efs*115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>fastq_013</td>\n",
       "      <td>G1_Sample_013_SNPs</td>\n",
       "      <td>G1_Sample_013</td>\n",
       "      <td>Yusuf M Taha</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>13.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>EDA\\nE4\\nc.659_676delCAGGTCCTCCTGGTCCTC\\np.P22...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>fastq_014</td>\n",
       "      <td>G1_Sample_014_SNPs</td>\n",
       "      <td>G1_Sample_014</td>\n",
       "      <td>Yusuf Mohamed Saeed Yusuf</td>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>2.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>-</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>EDA\\nE7\\nc.865C&gt;T\\np.R289C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>fastq_015</td>\n",
       "      <td>G1_Sample_015_SNPs</td>\n",
       "      <td>G1_Sample_015</td>\n",
       "      <td>Badr Yaser(Amniotic fluid)*</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>5.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>EDARADD\\nE2\\nc.85G&gt;A\\np.E29K</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>fastq_017</td>\n",
       "      <td>G1_Sample_017_SNPs</td>\n",
       "      <td>G1_Sample_017</td>\n",
       "      <td>Ibrahim Khattab</td>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>2.25</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "      <td>EDA\\nE4\\nc.620G&gt;A\\np.G207E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>fastq_021</td>\n",
       "      <td>G1_Sample_021_SNPs</td>\n",
       "      <td>G1_Sample_021</td>\n",
       "      <td>Sohila Abbas</td>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>9.42</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>fastq_022</td>\n",
       "      <td>G1_Sample_022_SNPs</td>\n",
       "      <td>G1_Sample_022</td>\n",
       "      <td>Mariam Ahmad Elward</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>3.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "      <td>EDAR\\nE4\\nc.204delC\\np.Y69Tfs*34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>fastq_024</td>\n",
       "      <td>G1_Sample_024_SNPs</td>\n",
       "      <td>G1_Sample_024</td>\n",
       "      <td>AbdelFattah Hussein</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>3.83</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "      <td>EDARADD\\nE6\\nc.570C&gt;A\\np.D190E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>S4174Nr5</td>\n",
       "      <td>G2_Sample_5_SNPs</td>\n",
       "      <td>Sample_5_SNPs</td>\n",
       "      <td>Shaimaa Naser</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>25.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28.0</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>moderate</td>\n",
       "      <td>TSPEAR Exon11:c.1788_1790del:p.E596del y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>S4174Nr11</td>\n",
       "      <td>G2_Sample_11_SNPs</td>\n",
       "      <td>Sample_11_SNPs</td>\n",
       "      <td>Farida AbdelRahman</td>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>1.75</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "      <td>VUS:GREM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>S4174Nr12</td>\n",
       "      <td>G2_Sample_12_SNPs</td>\n",
       "      <td>Sample_12_SNPs</td>\n",
       "      <td>Renad AbdelTawab</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>1.67</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>VUS:KRT14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>S4174Nr33</td>\n",
       "      <td>G2_Sample_33_SNPs</td>\n",
       "      <td>Sample_33_SNPs</td>\n",
       "      <td>Adam Ashraf</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>1.50</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "      <td>EDAR:\\nExon4:c.G313A:p.G105R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>S4174Nr34</td>\n",
       "      <td>G2_Sample_34_SNPs</td>\n",
       "      <td>Sample_34_SNPs</td>\n",
       "      <td>Jana Taher</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>6.50</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>10.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>moderate</td>\n",
       "      <td>y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>S4174Nr35</td>\n",
       "      <td>G2_Sample_35_SNPs</td>\n",
       "      <td>Sample_35_SNPs</td>\n",
       "      <td>Yusuf Medhat AbdelHameed</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>16.25</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>TSPEAR:\\nExon1:c.44delC:p.G17Afs*34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>S4174Nr36</td>\n",
       "      <td>G2_Sample_36_SNPs</td>\n",
       "      <td>Sample_36_SNPs</td>\n",
       "      <td>Aya Tamer</td>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>14.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.0</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>moderate</td>\n",
       "      <td>EDA:\\nExon8:c.1045G&gt;A p.A349T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>S4174Nr37</td>\n",
       "      <td>G2_Sample_37_SNPs</td>\n",
       "      <td>Sample_37_SNPs</td>\n",
       "      <td>Moh. Saad Gaber</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>4.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>S4174Nr38</td>\n",
       "      <td>G2_Sample_38_SNPs</td>\n",
       "      <td>Sample_38_SNPs</td>\n",
       "      <td>Ali Ismail</td>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>20.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>S4174Nr39</td>\n",
       "      <td>G2_Sample_39_SNPs</td>\n",
       "      <td>Sample_39_SNPs</td>\n",
       "      <td>Yusuf Saber</td>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>8.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>-</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>-ve</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>S4174Nr40</td>\n",
       "      <td>G2_Sample_40_SNPs</td>\n",
       "      <td>Sample_40_SNPs</td>\n",
       "      <td>Hamza Abbas</td>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>5.58</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>TSPEAR Exon11:c.1788_1790del:p.E596del y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>S4174Nr41</td>\n",
       "      <td>G2_Sample_41_SNPs</td>\n",
       "      <td>Sample_41_SNPs</td>\n",
       "      <td>Talia Sharawy</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>2.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>Severe</td>\n",
       "      <td>VUS:BTD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>S4174Nr42</td>\n",
       "      <td>G2_Sample_42_SNPs</td>\n",
       "      <td>Sample_42_SNPs</td>\n",
       "      <td>Omar Karam</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>3.67</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>5.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>TSPEAR Exon11:c.1788_1790del:p.E596del y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>S4174Nr43</td>\n",
       "      <td>G2_Sample_43_SNPs</td>\n",
       "      <td>Sample_43_SNPs</td>\n",
       "      <td>Yusuf Mohamed Eid</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>6.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>20.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "      <td>-ve</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>S4174Nr44</td>\n",
       "      <td>G2_Sample_44_SNPs</td>\n",
       "      <td>Sample_44_SNPs</td>\n",
       "      <td>Sherouk Emad</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>8.17</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>-</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>TSPEAR Exon11:c.1788_1790del:p.E596del y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>S4174Nr46</td>\n",
       "      <td>G2_Sample_46_SNPs</td>\n",
       "      <td>Sample_46_SNPs</td>\n",
       "      <td>Shabaan Hani</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>6.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>-ve</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>S4174Nr47</td>\n",
       "      <td>G2_Sample_47_SNPs</td>\n",
       "      <td>Sample_47_SNPs</td>\n",
       "      <td>Aya Gamal</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>S4174Nr48</td>\n",
       "      <td>G2_Sample_48_SNPs</td>\n",
       "      <td>Sample_48_SNPs</td>\n",
       "      <td>Roaa Hesham</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>3.42</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>-ve</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>S4174Nr49</td>\n",
       "      <td>G2_Sample_49_SNPs</td>\n",
       "      <td>Sample_49_SNPs</td>\n",
       "      <td>Farida M.Shawky</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>3.50</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>TSPEAR Exon11:c.1788_1790del:p.E596del y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>S4174Nr50</td>\n",
       "      <td>G2_Sample_50_SNPs</td>\n",
       "      <td>Sample_50_SNPs</td>\n",
       "      <td>Malek Moh Shabaan</td>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>1.75</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "      <td>VUS;NFKB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>S4174Nr51</td>\n",
       "      <td>G2_Sample_51_SNPs</td>\n",
       "      <td>Sample_51_SNPs</td>\n",
       "      <td>Mariam Rashwan</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>12.67</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>TSPEAR Exon11:c.1788_1790del:p.E596del y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>S4174Nr52</td>\n",
       "      <td>G2_Sample_52_SNPs</td>\n",
       "      <td>Sample_52_SNPs</td>\n",
       "      <td>Moh Mesbah</td>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>1.92</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>S4174Nr53</td>\n",
       "      <td>G2_Sample_53_SNPs</td>\n",
       "      <td>Sample_53_SNPs</td>\n",
       "      <td>Moh. AbdelRahman Farghaly</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>TSPEAR\\nOne variant! (c.668C&gt;T:p.Ser223Leu)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>S5424Nr1</td>\n",
       "      <td>G3_Sample_1_SNPs</td>\n",
       "      <td>Sample_1</td>\n",
       "      <td>Sandy Hussien Mohamed</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>2.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>-</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>12.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>S5424Nr2</td>\n",
       "      <td>G3_Sample_2_SNPs</td>\n",
       "      <td>Sample_2</td>\n",
       "      <td>Sayed Sarhan</td>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>10.50</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>moderate</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>S5424Nr3</td>\n",
       "      <td>G3_Sample_3_SNPs</td>\n",
       "      <td>Sample_3</td>\n",
       "      <td>Farida AbdelKhalek Ibrahim</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>1.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>S5424Nr4</td>\n",
       "      <td>G3_Sample_4_SNPs</td>\n",
       "      <td>Sample_4</td>\n",
       "      <td>Sajda Ahmed Saloma</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>2.42</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>S5424Nr5</td>\n",
       "      <td>G3_Sample_5_SNPs</td>\n",
       "      <td>Sample_5</td>\n",
       "      <td>Mohamed Kareem</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>S5424Nr6</td>\n",
       "      <td>G3_Sample_6_SNPs</td>\n",
       "      <td>Sample_6</td>\n",
       "      <td>Mohamed Ahmed Shabaan 1686</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>3.00</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>S5424Nr7</td>\n",
       "      <td>G3_Sample_7_SNPs</td>\n",
       "      <td>Sample_7</td>\n",
       "      <td>Jana AbdelRahman   J 63</td>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>12.33</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>S5424Nr8</td>\n",
       "      <td>G3_Sample_8_SNPs</td>\n",
       "      <td>Sample_8</td>\n",
       "      <td>Doaa Shaaban</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>6.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>+</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>4.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>S5424Nr9</td>\n",
       "      <td>G3_Sample_9_SNPs</td>\n",
       "      <td>Sample_9</td>\n",
       "      <td>Rashad Emad</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>15.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11.0</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Sample_ID          Patient_ID      Unnamed: 2  \\\n",
       "0   fastq_006  G1_Sample_006_SNPs   G1_Sample_006   \n",
       "1   fastq_007  G1_Sample_007_SNPs   G1_Sample_007   \n",
       "2   fastq_013  G1_Sample_013_SNPs   G1_Sample_013   \n",
       "3   fastq_014  G1_Sample_014_SNPs   G1_Sample_014   \n",
       "4   fastq_015  G1_Sample_015_SNPs   G1_Sample_015   \n",
       "5   fastq_017  G1_Sample_017_SNPs   G1_Sample_017   \n",
       "6   fastq_021  G1_Sample_021_SNPs   G1_Sample_021   \n",
       "7   fastq_022  G1_Sample_022_SNPs   G1_Sample_022   \n",
       "8   fastq_024  G1_Sample_024_SNPs   G1_Sample_024   \n",
       "9    S4174Nr5    G2_Sample_5_SNPs   Sample_5_SNPs   \n",
       "10  S4174Nr11   G2_Sample_11_SNPs  Sample_11_SNPs   \n",
       "11  S4174Nr12   G2_Sample_12_SNPs  Sample_12_SNPs   \n",
       "12  S4174Nr33   G2_Sample_33_SNPs  Sample_33_SNPs   \n",
       "13  S4174Nr34   G2_Sample_34_SNPs  Sample_34_SNPs   \n",
       "14  S4174Nr35   G2_Sample_35_SNPs  Sample_35_SNPs   \n",
       "15  S4174Nr36   G2_Sample_36_SNPs  Sample_36_SNPs   \n",
       "16  S4174Nr37   G2_Sample_37_SNPs  Sample_37_SNPs   \n",
       "17  S4174Nr38   G2_Sample_38_SNPs  Sample_38_SNPs   \n",
       "18  S4174Nr39   G2_Sample_39_SNPs  Sample_39_SNPs   \n",
       "19  S4174Nr40   G2_Sample_40_SNPs  Sample_40_SNPs   \n",
       "20  S4174Nr41   G2_Sample_41_SNPs  Sample_41_SNPs   \n",
       "21  S4174Nr42   G2_Sample_42_SNPs  Sample_42_SNPs   \n",
       "22  S4174Nr43   G2_Sample_43_SNPs  Sample_43_SNPs   \n",
       "23  S4174Nr44   G2_Sample_44_SNPs  Sample_44_SNPs   \n",
       "24  S4174Nr46   G2_Sample_46_SNPs  Sample_46_SNPs   \n",
       "25  S4174Nr47   G2_Sample_47_SNPs  Sample_47_SNPs   \n",
       "26  S4174Nr48   G2_Sample_48_SNPs  Sample_48_SNPs   \n",
       "27  S4174Nr49   G2_Sample_49_SNPs  Sample_49_SNPs   \n",
       "28  S4174Nr50   G2_Sample_50_SNPs  Sample_50_SNPs   \n",
       "29  S4174Nr51   G2_Sample_51_SNPs  Sample_51_SNPs   \n",
       "30  S4174Nr52   G2_Sample_52_SNPs  Sample_52_SNPs   \n",
       "31  S4174Nr53   G2_Sample_53_SNPs  Sample_53_SNPs   \n",
       "32   S5424Nr1    G3_Sample_1_SNPs        Sample_1   \n",
       "33   S5424Nr2    G3_Sample_2_SNPs        Sample_2   \n",
       "34   S5424Nr3    G3_Sample_3_SNPs        Sample_3   \n",
       "35   S5424Nr4    G3_Sample_4_SNPs        Sample_4   \n",
       "36   S5424Nr5    G3_Sample_5_SNPs        Sample_5   \n",
       "37   S5424Nr6    G3_Sample_6_SNPs        Sample_6   \n",
       "38   S5424Nr7    G3_Sample_7_SNPs        Sample_7   \n",
       "39   S5424Nr8    G3_Sample_8_SNPs        Sample_8   \n",
       "40   S5424Nr9    G3_Sample_9_SNPs        Sample_9   \n",
       "\n",
       "                   Patient Name  Consanguinity Sex  Age/Year      sweating  \\\n",
       "0       Donia Ahmed Mahmoud Ali              0   F     10.00  Hypohidrotic   \n",
       "1                   Malak Shadi              0   F      3.50  Hypohidrotic   \n",
       "2                  Yusuf M Taha              1   M     13.00    Anhidrotic   \n",
       "3     Yusuf Mohamed Saeed Yusuf              0   M      2.00        Normal   \n",
       "4   Badr Yaser(Amniotic fluid)*              1   M      5.00    Anhidrotic   \n",
       "5               Ibrahim Khattab              0   M      2.25  Hypohidrotic   \n",
       "6                  Sohila Abbas              0   F      9.42        Normal   \n",
       "7           Mariam Ahmad Elward              1   F      3.00    Anhidrotic   \n",
       "8           AbdelFattah Hussein              1   M      3.83    Anhidrotic   \n",
       "9                 Shaimaa Naser              1   F     25.00        Normal   \n",
       "10           Farida AbdelRahman              0   F      1.75  Hypohidrotic   \n",
       "11             Renad AbdelTawab              1   F      1.67  Hypohidrotic   \n",
       "12                  Adam Ashraf              1   M      1.50  Hypohidrotic   \n",
       "13                   Jana Taher              1   F      6.50        Normal   \n",
       "14     Yusuf Medhat AbdelHameed              1   M     16.25        Normal   \n",
       "15                    Aya Tamer              0   F     14.00        Normal   \n",
       "16              Moh. Saad Gaber              1   M      4.00    Anhidrotic   \n",
       "17                   Ali Ismail              0   M     20.00           NaN   \n",
       "18                  Yusuf Saber              0   M      8.00        Normal   \n",
       "19                  Hamza Abbas              0   M      5.58        Normal   \n",
       "20               Talia Sharawy               1   F      2.50    Anhidrotic   \n",
       "21                   Omar Karam              1   M      3.67        Normal   \n",
       "22            Yusuf Mohamed Eid              1   M      6.50    Anhidrotic   \n",
       "23                 Sherouk Emad              1   F      8.17        Normal   \n",
       "24                 Shabaan Hani              1   M      6.50    Anhidrotic   \n",
       "25                    Aya Gamal              1   F     10.00    Anhidrotic   \n",
       "26                  Roaa Hesham              1   F      3.42        Normal   \n",
       "27             Farida M.Shawky               1   F      3.50        Normal   \n",
       "28            Malek Moh Shabaan              0   M      1.75    Anhidrotic   \n",
       "29               Mariam Rashwan              1   F     12.67  Hypohidrotic   \n",
       "30                   Moh Mesbah              0   M      1.92    Anhidrotic   \n",
       "31    Moh. AbdelRahman Farghaly              1   M     10.00        Normal   \n",
       "32        Sandy Hussien Mohamed              1   F      2.50    Anhidrotic   \n",
       "33                 Sayed Sarhan              0   M     10.50        Normal   \n",
       "34   Farida AbdelKhalek Ibrahim              1   F      1.50    Anhidrotic   \n",
       "35           Sajda Ahmed Saloma              1   F      2.42  Hypohidrotic   \n",
       "36              Mohamed Kareem               1   M     10.00        Normal   \n",
       "37   Mohamed Ahmed Shabaan 1686              1   M      3.00  Hypohidrotic   \n",
       "38      Jana AbdelRahman   J 63              0   F     12.33        Normal   \n",
       "39                 Doaa Shaaban              1   F      6.00        Normal   \n",
       "40                  Rashad Emad              1   M     15.00        Normal   \n",
       "\n",
       "   Scalp Hair  eyebrow  receding anterior hair line      skin       Nails  \\\n",
       "0      Normal   Sparse                            -    Normal      Normal   \n",
       "1      Sparse   Sparse                          NaN  Affected  Dysplastic   \n",
       "2      Sparse   Sparse                            +  Affected  Dysplastic   \n",
       "3      Sparse   Sparse                            -    Normal      Normal   \n",
       "4      Sparse   Sparse                          NaN  Affected  Dysplastic   \n",
       "5      Sparse   Sparse                          NaN  Affected      Normal   \n",
       "6      Normal   Normal                            -    Normal      Normal   \n",
       "7      Sparse   Sparse                            +  Affected      Normal   \n",
       "8      Sparse   Sparse                          NaN  Affected      Normal   \n",
       "9     Sparse   Sparse                             +    Normal         NaN   \n",
       "10    Sparse   Sparse                             +  Affected      Normal   \n",
       "11    Sparse   Sparse                             +  Affected      Normal   \n",
       "12    Sparse   Sparse                             +  Affected      Normal   \n",
       "13    Sparse         -                            -  Affected      Normal   \n",
       "14     Normal   Normal                            +  Affected  Dysplastic   \n",
       "15     Normal  Sparse                             +  Affected      Normal   \n",
       "16    Sparse   Sparse                             +  Affected         NaN   \n",
       "17    Sparse   Sparse                             +  Affected      Normal   \n",
       "18     Normal  Sparse                             -  Affected      Normal   \n",
       "19     Normal   Normal                            +    Normal      Normal   \n",
       "20    Sparse   Sparse                             +       NaN         NaN   \n",
       "21    Sparse   Sparse                             +    Normal      Normal   \n",
       "22    Sparse   Sparse                             +  Affected      Normal   \n",
       "23     Normal  Sparse                             -  Affected      Normal   \n",
       "24    Sparse   Sparse                           NaN  Affected      Normal   \n",
       "25    Sparse   Sparse                             +  Affected      Normal   \n",
       "26     Normal  Sparse                             +  Affected  Dysplastic   \n",
       "27    Sparse   Sparse                             +  Affected  Dysplastic   \n",
       "28    Sparse   Sparse                             +  Affected      Normal   \n",
       "29     Normal   Normal                            +    Normal      Normal   \n",
       "30    Sparse   Sparse                             +  Affected         NaN   \n",
       "31     Normal   Normal                          NaN  Affected  Dysplastic   \n",
       "32     Sparse   Sparse                            -  Affected  Dysplastic   \n",
       "33     Normal   Normal                            -    Normal      Normal   \n",
       "34     Sparse   Sparse                            +  Affected      Normal   \n",
       "35     Sparse   Sparse                            +  Affected         NaN   \n",
       "36     Normal   Normal                            +    Normal      Normal   \n",
       "37     Sparse   Sparse                            +  Affected         NaN   \n",
       "38     Normal   Sparse                            +    normal      Normal   \n",
       "39     Normal   Sparse                            +  Affected      Normal   \n",
       "40     Normal   Normal                            -  Affected      Normal   \n",
       "\n",
       "    no of missing deciduous teeth  no of missing permenant teeth  \\\n",
       "0                             NaN                           12.0   \n",
       "1                            14.0                            NaN   \n",
       "2                             NaN                            NaN   \n",
       "3                             NaN                            NaN   \n",
       "4                             NaN                            NaN   \n",
       "5                             NaN                            NaN   \n",
       "6                             NaN                           10.0   \n",
       "7                             NaN                            NaN   \n",
       "8                             NaN                            NaN   \n",
       "9                             NaN                           28.0   \n",
       "10                            NaN                            NaN   \n",
       "11                            NaN                            NaN   \n",
       "12                            NaN                            NaN   \n",
       "13                           10.0                           17.0   \n",
       "14                            NaN                           10.0   \n",
       "15                            NaN                           17.0   \n",
       "16                            NaN                            NaN   \n",
       "17                            NaN                            NaN   \n",
       "18                            NaN                           20.0   \n",
       "19                            NaN                            NaN   \n",
       "20                            NaN                            NaN   \n",
       "21                            5.0                           11.0   \n",
       "22                           20.0                           28.0   \n",
       "23                            NaN                           14.0   \n",
       "24                            NaN                            NaN   \n",
       "25                            NaN                            NaN   \n",
       "26                            NaN                            NaN   \n",
       "27                            NaN                            NaN   \n",
       "28                            NaN                            NaN   \n",
       "29                            NaN                           10.0   \n",
       "30                            NaN                            NaN   \n",
       "31                            NaN                           12.0   \n",
       "32                           12.0                           17.0   \n",
       "33                            NaN                           10.0   \n",
       "34                            NaN                            NaN   \n",
       "35                           18.0                            NaN   \n",
       "36                            NaN                           12.0   \n",
       "37                           18.0                            NaN   \n",
       "38                            NaN                            6.0   \n",
       "39                            4.0                           18.0   \n",
       "40                            NaN                           11.0   \n",
       "\n",
       "   Missing teeth Peg shape teeth         Y  \\\n",
       "0              +               +  moderate   \n",
       "1              +               +    Severe   \n",
       "2              +               +    Severe   \n",
       "3              +               +  moderate   \n",
       "4              +               +    Severe   \n",
       "5              +             NaN    Severe   \n",
       "6              +               +  moderate   \n",
       "7              +             NaN    Severe   \n",
       "8              +             NaN    Severe   \n",
       "9              +             NaN  moderate   \n",
       "10             +             NaN    Severe   \n",
       "11             +               +    Severe   \n",
       "12             +             NaN    Severe   \n",
       "13             +               -  moderate   \n",
       "14             +               +  moderate   \n",
       "15             +               -  moderate   \n",
       "16             +               +    Severe   \n",
       "17             +               +  moderate   \n",
       "18             +               +  moderate   \n",
       "19             +               +  moderate   \n",
       "20             +               -    Severe   \n",
       "21             +               +  moderate   \n",
       "22             +             NaN    Severe   \n",
       "23             +               +  moderate   \n",
       "24             +               +    Severe   \n",
       "25             +               +    Severe   \n",
       "26             +               +    Severe   \n",
       "27             +               +    Severe   \n",
       "28             +             NaN    Severe   \n",
       "29             +               +  moderate   \n",
       "30             +             NaN    Severe   \n",
       "31             +               +  moderate   \n",
       "32             +               +    Severe   \n",
       "33             +               -  moderate   \n",
       "34             +               +    Severe   \n",
       "35             +               +    Severe   \n",
       "36             +               +  moderate   \n",
       "37             +               +    Severe   \n",
       "38             +               +  moderate   \n",
       "39             +               +  moderate   \n",
       "40             +               +  moderate   \n",
       "\n",
       "                                            Molecular  \n",
       "0                          EDA\\nE7\\nc.871G>A\\np.G291R  \n",
       "1                   EDA\\nE2\\nc.492delA\\np.G165Efs*115  \n",
       "2   EDA\\nE4\\nc.659_676delCAGGTCCTCCTGGTCCTC\\np.P22...  \n",
       "3                          EDA\\nE7\\nc.865C>T\\np.R289C  \n",
       "4                        EDARADD\\nE2\\nc.85G>A\\np.E29K  \n",
       "5                          EDA\\nE4\\nc.620G>A\\np.G207E  \n",
       "6                                                 NaN  \n",
       "7                    EDAR\\nE4\\nc.204delC\\np.Y69Tfs*34  \n",
       "8                      EDARADD\\nE6\\nc.570C>A\\np.D190E  \n",
       "9            TSPEAR Exon11:c.1788_1790del:p.E596del y  \n",
       "10                                           VUS:GREM  \n",
       "11                                          VUS:KRT14  \n",
       "12                       EDAR:\\nExon4:c.G313A:p.G105R  \n",
       "13                                                  y  \n",
       "14                TSPEAR:\\nExon1:c.44delC:p.G17Afs*34  \n",
       "15                      EDA:\\nExon8:c.1045G>A p.A349T  \n",
       "16                                                NaN  \n",
       "17                                                NaN  \n",
       "18                                                -ve  \n",
       "19           TSPEAR Exon11:c.1788_1790del:p.E596del y  \n",
       "20                                            VUS:BTD  \n",
       "21           TSPEAR Exon11:c.1788_1790del:p.E596del y  \n",
       "22                                                -ve  \n",
       "23           TSPEAR Exon11:c.1788_1790del:p.E596del y  \n",
       "24                                                -ve  \n",
       "25                                                NaN  \n",
       "26                                                -ve  \n",
       "27           TSPEAR Exon11:c.1788_1790del:p.E596del y  \n",
       "28                                          VUS;NFKB2  \n",
       "29           TSPEAR Exon11:c.1788_1790del:p.E596del y  \n",
       "30                                                NaN  \n",
       "31        TSPEAR\\nOne variant! (c.668C>T:p.Ser223Leu)  \n",
       "32                                                NaN  \n",
       "33                                                NaN  \n",
       "34                                                NaN  \n",
       "35                                                NaN  \n",
       "36                                                NaN  \n",
       "37                                                NaN  \n",
       "38                                                NaN  \n",
       "39                                                NaN  \n",
       "40                                                NaN  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "9e4b6b99-18be-4114-a036-e7b455a53195",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(41, 19)"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "622abc41",
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical_data = clinical_df.drop(columns=['Sample_ID', 'Unnamed: 2','Patient Name' , ' receding anterior hair line', 'no of missing deciduous teeth', 'no of missing permenant teeth',\n",
    "        'Molecular'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "934b2b05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(41, 12)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "cff2f3a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical_data.index = clinical_df['Patient_ID']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "0d3ecb82",
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical_data = clinical_data.drop(['Patient_ID'] , axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "cd7f151c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Consanguinity</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Age/Year</th>\n",
       "      <th>sweating</th>\n",
       "      <th>Scalp Hair</th>\n",
       "      <th>eyebrow</th>\n",
       "      <th>skin</th>\n",
       "      <th>Nails</th>\n",
       "      <th>Missing teeth</th>\n",
       "      <th>Peg shape teeth</th>\n",
       "      <th>Y</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G1_Sample_006_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_007_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>3.50</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_013_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>13.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_014_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>2.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_015_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>5.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_017_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>2.25</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_021_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>9.42</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_022_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>3.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_024_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>3.83</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_5_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>25.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_11_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>1.75</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_12_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>1.67</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_33_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>1.50</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_34_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>6.50</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>-</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_35_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>16.25</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_36_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>14.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_37_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>4.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_38_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>20.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_39_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>8.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_40_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>5.58</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_41_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>2.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_42_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>3.67</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_43_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>6.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_44_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>8.17</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_46_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>6.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_47_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_48_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>3.42</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_49_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>3.50</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_50_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>1.75</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_51_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>12.67</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_52_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>1.92</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_53_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_1_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>2.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_2_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>10.50</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_3_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>1.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_4_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>2.42</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_5_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_6_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>3.00</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_7_SNPs</th>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>12.33</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_8_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>6.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_9_SNPs</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>15.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Consanguinity Sex  Age/Year      sweating Scalp Hair  \\\n",
       "Patient_ID                                                                 \n",
       "G1_Sample_006_SNPs              0   F     10.00  Hypohidrotic     Normal   \n",
       "G1_Sample_007_SNPs              0   F      3.50  Hypohidrotic     Sparse   \n",
       "G1_Sample_013_SNPs              1   M     13.00    Anhidrotic     Sparse   \n",
       "G1_Sample_014_SNPs              0   M      2.00        Normal     Sparse   \n",
       "G1_Sample_015_SNPs              1   M      5.00    Anhidrotic     Sparse   \n",
       "G1_Sample_017_SNPs              0   M      2.25  Hypohidrotic     Sparse   \n",
       "G1_Sample_021_SNPs              0   F      9.42        Normal     Normal   \n",
       "G1_Sample_022_SNPs              1   F      3.00    Anhidrotic     Sparse   \n",
       "G1_Sample_024_SNPs              1   M      3.83    Anhidrotic     Sparse   \n",
       "G2_Sample_5_SNPs                1   F     25.00        Normal    Sparse    \n",
       "G2_Sample_11_SNPs               0   F      1.75  Hypohidrotic    Sparse    \n",
       "G2_Sample_12_SNPs               1   F      1.67  Hypohidrotic    Sparse    \n",
       "G2_Sample_33_SNPs               1   M      1.50  Hypohidrotic    Sparse    \n",
       "G2_Sample_34_SNPs               1   F      6.50        Normal    Sparse    \n",
       "G2_Sample_35_SNPs               1   M     16.25        Normal     Normal   \n",
       "G2_Sample_36_SNPs               0   F     14.00        Normal     Normal   \n",
       "G2_Sample_37_SNPs               1   M      4.00    Anhidrotic    Sparse    \n",
       "G2_Sample_38_SNPs               0   M     20.00           NaN    Sparse    \n",
       "G2_Sample_39_SNPs               0   M      8.00        Normal     Normal   \n",
       "G2_Sample_40_SNPs               0   M      5.58        Normal     Normal   \n",
       "G2_Sample_41_SNPs               1   F      2.50    Anhidrotic    Sparse    \n",
       "G2_Sample_42_SNPs               1   M      3.67        Normal    Sparse    \n",
       "G2_Sample_43_SNPs               1   M      6.50    Anhidrotic    Sparse    \n",
       "G2_Sample_44_SNPs               1   F      8.17        Normal     Normal   \n",
       "G2_Sample_46_SNPs               1   M      6.50    Anhidrotic    Sparse    \n",
       "G2_Sample_47_SNPs               1   F     10.00    Anhidrotic    Sparse    \n",
       "G2_Sample_48_SNPs               1   F      3.42        Normal     Normal   \n",
       "G2_Sample_49_SNPs               1   F      3.50        Normal    Sparse    \n",
       "G2_Sample_50_SNPs               0   M      1.75    Anhidrotic    Sparse    \n",
       "G2_Sample_51_SNPs               1   F     12.67  Hypohidrotic     Normal   \n",
       "G2_Sample_52_SNPs               0   M      1.92    Anhidrotic    Sparse    \n",
       "G2_Sample_53_SNPs               1   M     10.00        Normal     Normal   \n",
       "G3_Sample_1_SNPs                1   F      2.50    Anhidrotic     Sparse   \n",
       "G3_Sample_2_SNPs                0   M     10.50        Normal     Normal   \n",
       "G3_Sample_3_SNPs                1   F      1.50    Anhidrotic     Sparse   \n",
       "G3_Sample_4_SNPs                1   F      2.42  Hypohidrotic     Sparse   \n",
       "G3_Sample_5_SNPs                1   M     10.00        Normal     Normal   \n",
       "G3_Sample_6_SNPs                1   M      3.00  Hypohidrotic     Sparse   \n",
       "G3_Sample_7_SNPs                0   F     12.33        Normal     Normal   \n",
       "G3_Sample_8_SNPs                1   F      6.00        Normal     Normal   \n",
       "G3_Sample_9_SNPs                1   M     15.00        Normal     Normal   \n",
       "\n",
       "                    eyebrow      skin       Nails Missing teeth  \\\n",
       "Patient_ID                                                        \n",
       "G1_Sample_006_SNPs   Sparse    Normal      Normal             +   \n",
       "G1_Sample_007_SNPs   Sparse  Affected  Dysplastic             +   \n",
       "G1_Sample_013_SNPs   Sparse  Affected  Dysplastic             +   \n",
       "G1_Sample_014_SNPs   Sparse    Normal      Normal             +   \n",
       "G1_Sample_015_SNPs   Sparse  Affected  Dysplastic             +   \n",
       "G1_Sample_017_SNPs   Sparse  Affected      Normal             +   \n",
       "G1_Sample_021_SNPs   Normal    Normal      Normal             +   \n",
       "G1_Sample_022_SNPs   Sparse  Affected      Normal             +   \n",
       "G1_Sample_024_SNPs   Sparse  Affected      Normal             +   \n",
       "G2_Sample_5_SNPs    Sparse     Normal         NaN             +   \n",
       "G2_Sample_11_SNPs   Sparse   Affected      Normal             +   \n",
       "G2_Sample_12_SNPs   Sparse   Affected      Normal             +   \n",
       "G2_Sample_33_SNPs   Sparse   Affected      Normal             +   \n",
       "G2_Sample_34_SNPs         -  Affected      Normal             +   \n",
       "G2_Sample_35_SNPs    Normal  Affected  Dysplastic             +   \n",
       "G2_Sample_36_SNPs   Sparse   Affected      Normal             +   \n",
       "G2_Sample_37_SNPs   Sparse   Affected         NaN             +   \n",
       "G2_Sample_38_SNPs   Sparse   Affected      Normal             +   \n",
       "G2_Sample_39_SNPs   Sparse   Affected      Normal             +   \n",
       "G2_Sample_40_SNPs    Normal    Normal      Normal             +   \n",
       "G2_Sample_41_SNPs   Sparse        NaN         NaN             +   \n",
       "G2_Sample_42_SNPs   Sparse     Normal      Normal             +   \n",
       "G2_Sample_43_SNPs   Sparse   Affected      Normal             +   \n",
       "G2_Sample_44_SNPs   Sparse   Affected      Normal             +   \n",
       "G2_Sample_46_SNPs   Sparse   Affected      Normal             +   \n",
       "G2_Sample_47_SNPs   Sparse   Affected      Normal             +   \n",
       "G2_Sample_48_SNPs   Sparse   Affected  Dysplastic             +   \n",
       "G2_Sample_49_SNPs   Sparse   Affected  Dysplastic             +   \n",
       "G2_Sample_50_SNPs   Sparse   Affected      Normal             +   \n",
       "G2_Sample_51_SNPs    Normal    Normal      Normal             +   \n",
       "G2_Sample_52_SNPs   Sparse   Affected         NaN             +   \n",
       "G2_Sample_53_SNPs    Normal  Affected  Dysplastic             +   \n",
       "G3_Sample_1_SNPs     Sparse  Affected  Dysplastic             +   \n",
       "G3_Sample_2_SNPs     Normal    Normal      Normal             +   \n",
       "G3_Sample_3_SNPs     Sparse  Affected      Normal             +   \n",
       "G3_Sample_4_SNPs     Sparse  Affected         NaN             +   \n",
       "G3_Sample_5_SNPs     Normal    Normal      Normal             +   \n",
       "G3_Sample_6_SNPs     Sparse  Affected         NaN             +   \n",
       "G3_Sample_7_SNPs     Sparse    normal      Normal             +   \n",
       "G3_Sample_8_SNPs     Sparse  Affected      Normal             +   \n",
       "G3_Sample_9_SNPs     Normal  Affected      Normal             +   \n",
       "\n",
       "                   Peg shape teeth         Y  \n",
       "Patient_ID                                    \n",
       "G1_Sample_006_SNPs               +  moderate  \n",
       "G1_Sample_007_SNPs               +    Severe  \n",
       "G1_Sample_013_SNPs               +    Severe  \n",
       "G1_Sample_014_SNPs               +  moderate  \n",
       "G1_Sample_015_SNPs               +    Severe  \n",
       "G1_Sample_017_SNPs             NaN    Severe  \n",
       "G1_Sample_021_SNPs               +  moderate  \n",
       "G1_Sample_022_SNPs             NaN    Severe  \n",
       "G1_Sample_024_SNPs             NaN    Severe  \n",
       "G2_Sample_5_SNPs               NaN  moderate  \n",
       "G2_Sample_11_SNPs              NaN    Severe  \n",
       "G2_Sample_12_SNPs                +    Severe  \n",
       "G2_Sample_33_SNPs              NaN    Severe  \n",
       "G2_Sample_34_SNPs                -  moderate  \n",
       "G2_Sample_35_SNPs                +  moderate  \n",
       "G2_Sample_36_SNPs                -  moderate  \n",
       "G2_Sample_37_SNPs                +    Severe  \n",
       "G2_Sample_38_SNPs                +  moderate  \n",
       "G2_Sample_39_SNPs                +  moderate  \n",
       "G2_Sample_40_SNPs                +  moderate  \n",
       "G2_Sample_41_SNPs                -    Severe  \n",
       "G2_Sample_42_SNPs                +  moderate  \n",
       "G2_Sample_43_SNPs              NaN    Severe  \n",
       "G2_Sample_44_SNPs                +  moderate  \n",
       "G2_Sample_46_SNPs                +    Severe  \n",
       "G2_Sample_47_SNPs                +    Severe  \n",
       "G2_Sample_48_SNPs                +    Severe  \n",
       "G2_Sample_49_SNPs                +    Severe  \n",
       "G2_Sample_50_SNPs              NaN    Severe  \n",
       "G2_Sample_51_SNPs                +  moderate  \n",
       "G2_Sample_52_SNPs              NaN    Severe  \n",
       "G2_Sample_53_SNPs                +  moderate  \n",
       "G3_Sample_1_SNPs                 +    Severe  \n",
       "G3_Sample_2_SNPs                 -  moderate  \n",
       "G3_Sample_3_SNPs                 +    Severe  \n",
       "G3_Sample_4_SNPs                 +    Severe  \n",
       "G3_Sample_5_SNPs                 +  moderate  \n",
       "G3_Sample_6_SNPs                 +    Severe  \n",
       "G3_Sample_7_SNPs                 +  moderate  \n",
       "G3_Sample_8_SNPs                 +  moderate  \n",
       "G3_Sample_9_SNPs                 +  moderate  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b78ee5d",
   "metadata": {},
   "source": [
    "## Change row indices of clinical data to match row indices of G1_G2_G3_NoNull df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "9f17af0e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['G1_Sample_006_merged', 'G1_Sample_007_merged', 'G1_Sample_013_merged',\n",
       "       'G1_Sample_014_merged', 'G1_Sample_015_merged', 'G1_Sample_017_merged',\n",
       "       'G1_Sample_021_merged', 'G1_Sample_022_merged', 'G1_Sample_024_merged',\n",
       "       'G2_Sample_11_merged', 'G2_Sample_12_merged', 'G2_Sample_33_merged',\n",
       "       'G2_Sample_34_merged', 'G2_Sample_35_merged', 'G2_Sample_36_merged',\n",
       "       'G2_Sample_37_merged', 'G2_Sample_38_merged', 'G2_Sample_39_merged',\n",
       "       'G2_Sample_40_merged', 'G2_Sample_41_merged', 'G2_Sample_42_merged',\n",
       "       'G2_Sample_43_merged', 'G2_Sample_44_merged', 'G2_Sample_46_merged',\n",
       "       'G2_Sample_47_merged', 'G2_Sample_48_merged', 'G2_Sample_49_merged',\n",
       "       'G2_Sample_50_merged', 'G2_Sample_51_merged', 'G2_Sample_52_merged',\n",
       "       'G2_Sample_53_merged', 'G2_Sample_5_merged', 'G3_Sample_1_merged',\n",
       "       'G3_Sample_2_merged', 'G3_Sample_3_merged', 'G3_Sample_4_merged',\n",
       "       'G3_Sample_5_merged', 'G3_Sample_6_merged', 'G3_Sample_7_merged',\n",
       "       'G3_Sample_8_merged', 'G3_Sample_9_merged'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "G1_G2_G3_NoNull.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "456437fa",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['G1_Sample_006_SNPs', 'G1_Sample_007_SNPs', 'G1_Sample_013_SNPs',\n",
       "       'G1_Sample_014_SNPs', 'G1_Sample_015_SNPs', 'G1_Sample_017_SNPs',\n",
       "       'G1_Sample_021_SNPs', 'G1_Sample_022_SNPs', 'G1_Sample_024_SNPs',\n",
       "       'G2_Sample_5_SNPs', 'G2_Sample_11_SNPs', 'G2_Sample_12_SNPs',\n",
       "       'G2_Sample_33_SNPs', 'G2_Sample_34_SNPs', 'G2_Sample_35_SNPs',\n",
       "       'G2_Sample_36_SNPs', 'G2_Sample_37_SNPs', 'G2_Sample_38_SNPs',\n",
       "       'G2_Sample_39_SNPs', 'G2_Sample_40_SNPs', 'G2_Sample_41_SNPs',\n",
       "       'G2_Sample_42_SNPs', 'G2_Sample_43_SNPs', 'G2_Sample_44_SNPs',\n",
       "       'G2_Sample_46_SNPs', 'G2_Sample_47_SNPs', 'G2_Sample_48_SNPs',\n",
       "       'G2_Sample_49_SNPs', 'G2_Sample_50_SNPs', 'G2_Sample_51_SNPs',\n",
       "       'G2_Sample_52_SNPs', 'G2_Sample_53_SNPs', 'G3_Sample_1_SNPs',\n",
       "       'G3_Sample_2_SNPs', 'G3_Sample_3_SNPs', 'G3_Sample_4_SNPs',\n",
       "       'G3_Sample_5_SNPs', 'G3_Sample_6_SNPs', 'G3_Sample_7_SNPs',\n",
       "       'G3_Sample_8_SNPs', 'G3_Sample_9_SNPs'],\n",
       "      dtype='object', name='Patient_ID')"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_data.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "cf7a67ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_index= []\n",
    "for i in clinical_data.index:\n",
    "    i = i.rsplit('_',1)[0] + '_merged'\n",
    "    #i = i.replace('s','S')\n",
    "    new_index.append(i)\n",
    "#new_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "b4314321",
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical_data.index = new_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "91ab269b-e30f-4e9d-97ce-8f8ce07c6bb3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['G1_Sample_006_merged', 'G1_Sample_007_merged', 'G1_Sample_013_merged',\n",
       "       'G1_Sample_014_merged', 'G1_Sample_015_merged', 'G1_Sample_017_merged',\n",
       "       'G1_Sample_021_merged', 'G1_Sample_022_merged', 'G1_Sample_024_merged',\n",
       "       'G2_Sample_5_merged', 'G2_Sample_11_merged', 'G2_Sample_12_merged',\n",
       "       'G2_Sample_33_merged', 'G2_Sample_34_merged', 'G2_Sample_35_merged',\n",
       "       'G2_Sample_36_merged', 'G2_Sample_37_merged', 'G2_Sample_38_merged',\n",
       "       'G2_Sample_39_merged', 'G2_Sample_40_merged', 'G2_Sample_41_merged',\n",
       "       'G2_Sample_42_merged', 'G2_Sample_43_merged', 'G2_Sample_44_merged',\n",
       "       'G2_Sample_46_merged', 'G2_Sample_47_merged', 'G2_Sample_48_merged',\n",
       "       'G2_Sample_49_merged', 'G2_Sample_50_merged', 'G2_Sample_51_merged',\n",
       "       'G2_Sample_52_merged', 'G2_Sample_53_merged', 'G3_Sample_1_merged',\n",
       "       'G3_Sample_2_merged', 'G3_Sample_3_merged', 'G3_Sample_4_merged',\n",
       "       'G3_Sample_5_merged', 'G3_Sample_6_merged', 'G3_Sample_7_merged',\n",
       "       'G3_Sample_8_merged', 'G3_Sample_9_merged'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_data.index"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c8bf97c",
   "metadata": {},
   "source": [
    "# Combine merged for all groups df & Label 'Y' column from clinical data for all groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "2af04672",
   "metadata": {},
   "outputs": [],
   "source": [
    "All_Groups_df = pd.merge(G1_G2_G3_NoNull, clinical_data['Y'], left_index=True, right_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "042fb324",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr1-100000012</th>\n",
       "      <th>chr1-100007888</th>\n",
       "      <th>chr1-100011243</th>\n",
       "      <th>chr1-100011253</th>\n",
       "      <th>chr1-1000156</th>\n",
       "      <th>chr1-100015837</th>\n",
       "      <th>chr1-100015925</th>\n",
       "      <th>chr1-100019269</th>\n",
       "      <th>chr1-100022918</th>\n",
       "      <th>chr1-10002546</th>\n",
       "      <th>...</th>\n",
       "      <th>chrX-99969558</th>\n",
       "      <th>chrX-99972474</th>\n",
       "      <th>chrX-9997369</th>\n",
       "      <th>chrX-99974005</th>\n",
       "      <th>chrX-99979187</th>\n",
       "      <th>chrX-99990696</th>\n",
       "      <th>chrX-99995215</th>\n",
       "      <th>chrX-99997342</th>\n",
       "      <th>chrX-99997406</th>\n",
       "      <th>Y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G1_Sample_006_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_007_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_013_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_014_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_015_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_017_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_021_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_022_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_024_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_11_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_12_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_33_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_34_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_35_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_36_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_37_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_38_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_39_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_40_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_41_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_42_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_43_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_44_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_46_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_47_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_48_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_49_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_50_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_51_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_52_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_53_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_5_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_1_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_2_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_3_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_4_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_5_merged</th>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_6_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_7_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_8_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G3_Sample_9_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41 rows × 569651 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     chr1-100000012 chr1-100007888 chr1-100011243  \\\n",
       "G1_Sample_006_merged            0/0            0/0            0/0   \n",
       "G1_Sample_007_merged            0/0            0/0            0/0   \n",
       "G1_Sample_013_merged            0/0            0/0            0/0   \n",
       "G1_Sample_014_merged            0/0            0/0            0/0   \n",
       "G1_Sample_015_merged            0/0            0/0            0/0   \n",
       "G1_Sample_017_merged            0/0            0/0            0/0   \n",
       "G1_Sample_021_merged            0/0            0/0            0/0   \n",
       "G1_Sample_022_merged            0/0            0/0            0/0   \n",
       "G1_Sample_024_merged            0/0            0/0            0/0   \n",
       "G2_Sample_11_merged             0/0            0/0            0/0   \n",
       "G2_Sample_12_merged             0/0            0/0            0/0   \n",
       "G2_Sample_33_merged             0/0            0/0            0/0   \n",
       "G2_Sample_34_merged             0/0            0/0            0/0   \n",
       "G2_Sample_35_merged             0/0            0/0            0/0   \n",
       "G2_Sample_36_merged             0/0            0/0            0/0   \n",
       "G2_Sample_37_merged             0/0            0/0            0/0   \n",
       "G2_Sample_38_merged             0/0            0/0            0/0   \n",
       "G2_Sample_39_merged             0/0            0/0            0/0   \n",
       "G2_Sample_40_merged             0/0            0/0            0/0   \n",
       "G2_Sample_41_merged             0/0            0/0            0/0   \n",
       "G2_Sample_42_merged             0/0            0/0            0/0   \n",
       "G2_Sample_43_merged             0/0            0/0            0/0   \n",
       "G2_Sample_44_merged             0/0            0/0            0/0   \n",
       "G2_Sample_46_merged             0/0            0/0            0/0   \n",
       "G2_Sample_47_merged             0/0            0/0            0/0   \n",
       "G2_Sample_48_merged             0/0            0/0            0/0   \n",
       "G2_Sample_49_merged             0/0            0/0            0/0   \n",
       "G2_Sample_50_merged             0/0            0/0            0/0   \n",
       "G2_Sample_51_merged             0/0            0/0            0/0   \n",
       "G2_Sample_52_merged             0/0            0/0            0/0   \n",
       "G2_Sample_53_merged             0/0            0/0            0/0   \n",
       "G2_Sample_5_merged              0/0            0/0            0/0   \n",
       "G3_Sample_1_merged              0/0            0/0            0/0   \n",
       "G3_Sample_2_merged              0/0            1/1            0/0   \n",
       "G3_Sample_3_merged              0/0            0/0            0/0   \n",
       "G3_Sample_4_merged              0/0            0/0            0/0   \n",
       "G3_Sample_5_merged              1/1            0/0            0/0   \n",
       "G3_Sample_6_merged              0/0            0/0            0/0   \n",
       "G3_Sample_7_merged              0/0            0/0            0/0   \n",
       "G3_Sample_8_merged              0/0            0/0            0/1   \n",
       "G3_Sample_9_merged              0/0            0/0            0/0   \n",
       "\n",
       "                     chr1-100011253 chr1-1000156 chr1-100015837  \\\n",
       "G1_Sample_006_merged            0/0          0/0            0/0   \n",
       "G1_Sample_007_merged            0/0          0/0            0/0   \n",
       "G1_Sample_013_merged            0/0          0/0            0/0   \n",
       "G1_Sample_014_merged            0/0          0/0            0/0   \n",
       "G1_Sample_015_merged            0/0          0/0            0/0   \n",
       "G1_Sample_017_merged            0/0          0/0            0/0   \n",
       "G1_Sample_021_merged            0/0          0/0            0/0   \n",
       "G1_Sample_022_merged            0/0          0/0            0/0   \n",
       "G1_Sample_024_merged            0/0          0/0            0/0   \n",
       "G2_Sample_11_merged             0/0          0/0            0/0   \n",
       "G2_Sample_12_merged             0/0          0/0            0/0   \n",
       "G2_Sample_33_merged             0/0          0/0            0/0   \n",
       "G2_Sample_34_merged             0/0          0/0            0/0   \n",
       "G2_Sample_35_merged             0/0          0/0            0/0   \n",
       "G2_Sample_36_merged             0/0          0/0            0/0   \n",
       "G2_Sample_37_merged             0/0          0/0            0/0   \n",
       "G2_Sample_38_merged             0/0          0/0            0/0   \n",
       "G2_Sample_39_merged             0/0          0/0            0/0   \n",
       "G2_Sample_40_merged             0/0          0/0            0/0   \n",
       "G2_Sample_41_merged             0/0          0/0            0/0   \n",
       "G2_Sample_42_merged             0/0          0/0            0/0   \n",
       "G2_Sample_43_merged             0/0          0/0            0/0   \n",
       "G2_Sample_44_merged             0/0          0/0            0/0   \n",
       "G2_Sample_46_merged             0/0          0/0            0/0   \n",
       "G2_Sample_47_merged             0/0          0/0            0/0   \n",
       "G2_Sample_48_merged             0/0          0/0            0/0   \n",
       "G2_Sample_49_merged             0/0          0/0            0/0   \n",
       "G2_Sample_50_merged             0/0          0/0            0/0   \n",
       "G2_Sample_51_merged             0/0          0/0            0/0   \n",
       "G2_Sample_52_merged             0/0          0/0            0/0   \n",
       "G2_Sample_53_merged             0/0          0/0            0/0   \n",
       "G2_Sample_5_merged              0/0          0/0            0/0   \n",
       "G3_Sample_1_merged              0/0          0/0            0/0   \n",
       "G3_Sample_2_merged              0/0          0/0            0/0   \n",
       "G3_Sample_3_merged              0/0          0/0            0/0   \n",
       "G3_Sample_4_merged              0/0          0/0            0/0   \n",
       "G3_Sample_5_merged              0/0          0/0            1/1   \n",
       "G3_Sample_6_merged              0/0          1/1            0/0   \n",
       "G3_Sample_7_merged              0/0          0/0            0/0   \n",
       "G3_Sample_8_merged              0/1          0/0            0/0   \n",
       "G3_Sample_9_merged              0/0          0/0            0/0   \n",
       "\n",
       "                     chr1-100015925 chr1-100019269 chr1-100022918  \\\n",
       "G1_Sample_006_merged            0/0            0/0            0/0   \n",
       "G1_Sample_007_merged            0/0            0/0            0/0   \n",
       "G1_Sample_013_merged            0/0            0/0            0/0   \n",
       "G1_Sample_014_merged            0/0            0/0            0/0   \n",
       "G1_Sample_015_merged            0/0            0/0            0/0   \n",
       "G1_Sample_017_merged            0/0            0/0            0/0   \n",
       "G1_Sample_021_merged            0/0            0/0            0/0   \n",
       "G1_Sample_022_merged            0/0            0/0            0/0   \n",
       "G1_Sample_024_merged            0/0            0/0            0/0   \n",
       "G2_Sample_11_merged             0/0            0/0            0/0   \n",
       "G2_Sample_12_merged             0/0            0/0            0/0   \n",
       "G2_Sample_33_merged             0/0            0/0            0/0   \n",
       "G2_Sample_34_merged             0/0            0/0            0/0   \n",
       "G2_Sample_35_merged             0/0            0/0            0/0   \n",
       "G2_Sample_36_merged             0/0            0/0            0/0   \n",
       "G2_Sample_37_merged             0/0            0/0            0/0   \n",
       "G2_Sample_38_merged             0/0            0/0            0/0   \n",
       "G2_Sample_39_merged             0/0            0/0            0/0   \n",
       "G2_Sample_40_merged             0/0            0/0            0/0   \n",
       "G2_Sample_41_merged             0/0            0/0            0/0   \n",
       "G2_Sample_42_merged             0/0            0/0            0/0   \n",
       "G2_Sample_43_merged             0/0            0/0            0/0   \n",
       "G2_Sample_44_merged             0/0            0/0            0/0   \n",
       "G2_Sample_46_merged             0/0            0/0            0/0   \n",
       "G2_Sample_47_merged             0/0            0/0            0/0   \n",
       "G2_Sample_48_merged             0/0            0/0            0/0   \n",
       "G2_Sample_49_merged             0/0            0/0            0/0   \n",
       "G2_Sample_50_merged             0/0            0/0            0/0   \n",
       "G2_Sample_51_merged             0/0            0/0            0/0   \n",
       "G2_Sample_52_merged             0/0            0/0            0/0   \n",
       "G2_Sample_53_merged             0/0            0/0            0/0   \n",
       "G2_Sample_5_merged              0/0            0/0            0/0   \n",
       "G3_Sample_1_merged              0/0            0/0            0/0   \n",
       "G3_Sample_2_merged              0/0            0/0            0/0   \n",
       "G3_Sample_3_merged              0/0            0/0            0/0   \n",
       "G3_Sample_4_merged              1/1            1/1            0/0   \n",
       "G3_Sample_5_merged              0/0            0/0            0/0   \n",
       "G3_Sample_6_merged              0/0            0/0            0/0   \n",
       "G3_Sample_7_merged              0/0            0/0            0/0   \n",
       "G3_Sample_8_merged              0/0            0/0            0/0   \n",
       "G3_Sample_9_merged              0/0            0/0            1/1   \n",
       "\n",
       "                     chr1-10002546  ... chrX-99969558 chrX-99972474  \\\n",
       "G1_Sample_006_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_007_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_013_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_014_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_015_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_017_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_021_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_022_merged           0/0  ...           0/0           0/0   \n",
       "G1_Sample_024_merged           0/0  ...           0/0           0/0   \n",
       "G2_Sample_11_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_12_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_33_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_34_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_35_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_36_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_37_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_38_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_39_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_40_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_41_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_42_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_43_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_44_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_46_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_47_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_48_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_49_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_50_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_51_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_52_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_53_merged            0/0  ...           0/0           0/0   \n",
       "G2_Sample_5_merged             0/0  ...           0/0           0/0   \n",
       "G3_Sample_1_merged             0/0  ...           1/1           0/0   \n",
       "G3_Sample_2_merged             0/0  ...           0/0           0/0   \n",
       "G3_Sample_3_merged             0/0  ...           0/0           0/0   \n",
       "G3_Sample_4_merged             0/0  ...           0/0           0/0   \n",
       "G3_Sample_5_merged             0/0  ...           0/0           1/1   \n",
       "G3_Sample_6_merged             0/0  ...           0/0           0/0   \n",
       "G3_Sample_7_merged             0/1  ...           0/0           0/0   \n",
       "G3_Sample_8_merged             0/0  ...           0/0           0/0   \n",
       "G3_Sample_9_merged             0/0  ...           0/0           0/0   \n",
       "\n",
       "                     chrX-9997369 chrX-99974005 chrX-99979187 chrX-99990696  \\\n",
       "G1_Sample_006_merged          0/0           0/0           0/0           0/0   \n",
       "G1_Sample_007_merged          0/0           0/0           0/0           0/0   \n",
       "G1_Sample_013_merged          0/0           0/0           0/0           0/0   \n",
       "G1_Sample_014_merged          0/0           0/0           0/0           0/0   \n",
       "G1_Sample_015_merged          0/0           0/0           0/0           0/0   \n",
       "G1_Sample_017_merged          0/0           0/0           0/0           0/0   \n",
       "G1_Sample_021_merged          0/0           0/0           0/0           0/0   \n",
       "G1_Sample_022_merged          0/0           0/0           0/0           0/0   \n",
       "G1_Sample_024_merged          0/0           0/0           0/0           0/0   \n",
       "G2_Sample_11_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_12_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_33_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_34_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_35_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_36_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_37_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_38_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_39_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_40_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_41_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_42_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_43_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_44_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_46_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_47_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_48_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_49_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_50_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_51_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_52_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_53_merged           0/0           0/0           0/0           0/0   \n",
       "G2_Sample_5_merged            0/0           0/0           0/0           0/0   \n",
       "G3_Sample_1_merged            0/0           0/0           0/0           0/0   \n",
       "G3_Sample_2_merged            0/1           0/0           0/0           0/0   \n",
       "G3_Sample_3_merged            0/0           1/1           0/0           0/0   \n",
       "G3_Sample_4_merged            0/0           0/0           0/0           0/0   \n",
       "G3_Sample_5_merged            0/0           0/0           0/1           0/0   \n",
       "G3_Sample_6_merged            0/0           0/0           0/0           0/0   \n",
       "G3_Sample_7_merged            0/0           0/0           0/0           0/0   \n",
       "G3_Sample_8_merged            0/0           0/0           0/0           1/1   \n",
       "G3_Sample_9_merged            0/0           0/0           0/0           0/0   \n",
       "\n",
       "                     chrX-99995215 chrX-99997342 chrX-99997406         Y  \n",
       "G1_Sample_006_merged           0/0           0/0           0/0  moderate  \n",
       "G1_Sample_007_merged           0/0           0/0           0/0    Severe  \n",
       "G1_Sample_013_merged           0/0           0/0           0/0    Severe  \n",
       "G1_Sample_014_merged           0/0           0/0           0/0  moderate  \n",
       "G1_Sample_015_merged           0/0           0/0           0/0    Severe  \n",
       "G1_Sample_017_merged           0/0           0/0           0/0    Severe  \n",
       "G1_Sample_021_merged           0/0           0/0           0/0  moderate  \n",
       "G1_Sample_022_merged           0/0           0/0           0/0    Severe  \n",
       "G1_Sample_024_merged           0/0           0/0           0/0    Severe  \n",
       "G2_Sample_11_merged            0/0           0/0           0/0    Severe  \n",
       "G2_Sample_12_merged            0/0           0/0           0/0    Severe  \n",
       "G2_Sample_33_merged            0/0           0/0           0/0    Severe  \n",
       "G2_Sample_34_merged            0/0           0/0           0/0  moderate  \n",
       "G2_Sample_35_merged            0/0           0/0           0/0  moderate  \n",
       "G2_Sample_36_merged            0/0           0/0           0/0  moderate  \n",
       "G2_Sample_37_merged            0/0           0/0           0/0    Severe  \n",
       "G2_Sample_38_merged            0/0           0/0           0/0  moderate  \n",
       "G2_Sample_39_merged            0/0           0/0           0/0  moderate  \n",
       "G2_Sample_40_merged            0/0           0/0           0/0  moderate  \n",
       "G2_Sample_41_merged            0/0           0/0           0/0    Severe  \n",
       "G2_Sample_42_merged            0/0           0/0           0/0  moderate  \n",
       "G2_Sample_43_merged            0/0           0/0           0/0    Severe  \n",
       "G2_Sample_44_merged            0/0           0/0           0/0  moderate  \n",
       "G2_Sample_46_merged            0/0           0/0           0/0    Severe  \n",
       "G2_Sample_47_merged            0/0           0/0           0/0    Severe  \n",
       "G2_Sample_48_merged            0/0           0/0           0/0    Severe  \n",
       "G2_Sample_49_merged            0/0           0/0           0/0    Severe  \n",
       "G2_Sample_50_merged            0/0           0/0           0/0    Severe  \n",
       "G2_Sample_51_merged            0/0           0/0           0/0  moderate  \n",
       "G2_Sample_52_merged            0/0           0/0           0/0    Severe  \n",
       "G2_Sample_53_merged            0/0           0/0           0/0  moderate  \n",
       "G2_Sample_5_merged             0/0           0/0           0/0  moderate  \n",
       "G3_Sample_1_merged             0/0           0/0           0/0    Severe  \n",
       "G3_Sample_2_merged             0/0           0/0           0/0  moderate  \n",
       "G3_Sample_3_merged             0/1           0/0           1/1    Severe  \n",
       "G3_Sample_4_merged             0/0           0/0           0/0    Severe  \n",
       "G3_Sample_5_merged             0/0           0/0           0/0  moderate  \n",
       "G3_Sample_6_merged             0/0           0/0           0/0    Severe  \n",
       "G3_Sample_7_merged             0/0           0/0           0/0  moderate  \n",
       "G3_Sample_8_merged             0/0           1/1           1/1  moderate  \n",
       "G3_Sample_9_merged             0/0           0/0           0/0  moderate  \n",
       "\n",
       "[41 rows x 569651 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "All_Groups_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65cfddf1",
   "metadata": {},
   "source": [
    "# Save clinical df for all groups as csv file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "1b52a819",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 8min 16s, sys: 12.2 s, total: 8min 28s\n",
      "Wall time: 8min 27s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "All_groups_Label = '/home/phoebe/phd/ML_csv/'\n",
    "file_name = 'All_groups_Label.csv'\n",
    "file_out_path = os.path.join(All_groups_Label , file_name)\n",
    "All_Groups_df.to_csv(file_out_path)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71cc16dc",
   "metadata": {},
   "source": [
    "> ## Separate clinical data for the first two groups in a separate Dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "5be2b492",
   "metadata": {},
   "outputs": [],
   "source": [
    "G1_G2_clinical = clinical_data.iloc[:32, :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "384fb10e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Consanguinity</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Age/Year</th>\n",
       "      <th>sweating</th>\n",
       "      <th>Scalp Hair</th>\n",
       "      <th>eyebrow</th>\n",
       "      <th>skin</th>\n",
       "      <th>Nails</th>\n",
       "      <th>Missing teeth</th>\n",
       "      <th>Peg shape teeth</th>\n",
       "      <th>Y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G1_Sample_006_merged</th>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_007_merged</th>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>3.50</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_013_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>13.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_014_merged</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>2.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_015_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>5.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_017_merged</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>2.25</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_021_merged</th>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>9.42</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_022_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>3.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_024_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>3.83</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_5_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>25.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_11_merged</th>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>1.75</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_12_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>1.67</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_33_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>1.50</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_34_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>6.50</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>-</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_35_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>16.25</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_36_merged</th>\n",
       "      <td>0</td>\n",
       "      <td>F</td>\n",
       "      <td>14.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_37_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>4.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_38_merged</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>20.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_39_merged</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>8.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_40_merged</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>5.58</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_41_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>2.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_42_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>3.67</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_43_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>6.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_44_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>8.17</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_46_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>6.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_47_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_48_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>3.42</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_49_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>3.50</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_50_merged</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>1.75</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_51_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>12.67</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_52_merged</th>\n",
       "      <td>0</td>\n",
       "      <td>M</td>\n",
       "      <td>1.92</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_53_merged</th>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Consanguinity Sex  Age/Year      sweating Scalp Hair  \\\n",
       "G1_Sample_006_merged              0   F     10.00  Hypohidrotic     Normal   \n",
       "G1_Sample_007_merged              0   F      3.50  Hypohidrotic     Sparse   \n",
       "G1_Sample_013_merged              1   M     13.00    Anhidrotic     Sparse   \n",
       "G1_Sample_014_merged              0   M      2.00        Normal     Sparse   \n",
       "G1_Sample_015_merged              1   M      5.00    Anhidrotic     Sparse   \n",
       "G1_Sample_017_merged              0   M      2.25  Hypohidrotic     Sparse   \n",
       "G1_Sample_021_merged              0   F      9.42        Normal     Normal   \n",
       "G1_Sample_022_merged              1   F      3.00    Anhidrotic     Sparse   \n",
       "G1_Sample_024_merged              1   M      3.83    Anhidrotic     Sparse   \n",
       "G2_Sample_5_merged                1   F     25.00        Normal    Sparse    \n",
       "G2_Sample_11_merged               0   F      1.75  Hypohidrotic    Sparse    \n",
       "G2_Sample_12_merged               1   F      1.67  Hypohidrotic    Sparse    \n",
       "G2_Sample_33_merged               1   M      1.50  Hypohidrotic    Sparse    \n",
       "G2_Sample_34_merged               1   F      6.50        Normal    Sparse    \n",
       "G2_Sample_35_merged               1   M     16.25        Normal     Normal   \n",
       "G2_Sample_36_merged               0   F     14.00        Normal     Normal   \n",
       "G2_Sample_37_merged               1   M      4.00    Anhidrotic    Sparse    \n",
       "G2_Sample_38_merged               0   M     20.00           NaN    Sparse    \n",
       "G2_Sample_39_merged               0   M      8.00        Normal     Normal   \n",
       "G2_Sample_40_merged               0   M      5.58        Normal     Normal   \n",
       "G2_Sample_41_merged               1   F      2.50    Anhidrotic    Sparse    \n",
       "G2_Sample_42_merged               1   M      3.67        Normal    Sparse    \n",
       "G2_Sample_43_merged               1   M      6.50    Anhidrotic    Sparse    \n",
       "G2_Sample_44_merged               1   F      8.17        Normal     Normal   \n",
       "G2_Sample_46_merged               1   M      6.50    Anhidrotic    Sparse    \n",
       "G2_Sample_47_merged               1   F     10.00    Anhidrotic    Sparse    \n",
       "G2_Sample_48_merged               1   F      3.42        Normal     Normal   \n",
       "G2_Sample_49_merged               1   F      3.50        Normal    Sparse    \n",
       "G2_Sample_50_merged               0   M      1.75    Anhidrotic    Sparse    \n",
       "G2_Sample_51_merged               1   F     12.67  Hypohidrotic     Normal   \n",
       "G2_Sample_52_merged               0   M      1.92    Anhidrotic    Sparse    \n",
       "G2_Sample_53_merged               1   M     10.00        Normal     Normal   \n",
       "\n",
       "                      eyebrow      skin       Nails Missing teeth  \\\n",
       "G1_Sample_006_merged   Sparse    Normal      Normal             +   \n",
       "G1_Sample_007_merged   Sparse  Affected  Dysplastic             +   \n",
       "G1_Sample_013_merged   Sparse  Affected  Dysplastic             +   \n",
       "G1_Sample_014_merged   Sparse    Normal      Normal             +   \n",
       "G1_Sample_015_merged   Sparse  Affected  Dysplastic             +   \n",
       "G1_Sample_017_merged   Sparse  Affected      Normal             +   \n",
       "G1_Sample_021_merged   Normal    Normal      Normal             +   \n",
       "G1_Sample_022_merged   Sparse  Affected      Normal             +   \n",
       "G1_Sample_024_merged   Sparse  Affected      Normal             +   \n",
       "G2_Sample_5_merged    Sparse     Normal         NaN             +   \n",
       "G2_Sample_11_merged   Sparse   Affected      Normal             +   \n",
       "G2_Sample_12_merged   Sparse   Affected      Normal             +   \n",
       "G2_Sample_33_merged   Sparse   Affected      Normal             +   \n",
       "G2_Sample_34_merged         -  Affected      Normal             +   \n",
       "G2_Sample_35_merged    Normal  Affected  Dysplastic             +   \n",
       "G2_Sample_36_merged   Sparse   Affected      Normal             +   \n",
       "G2_Sample_37_merged   Sparse   Affected         NaN             +   \n",
       "G2_Sample_38_merged   Sparse   Affected      Normal             +   \n",
       "G2_Sample_39_merged   Sparse   Affected      Normal             +   \n",
       "G2_Sample_40_merged    Normal    Normal      Normal             +   \n",
       "G2_Sample_41_merged   Sparse        NaN         NaN             +   \n",
       "G2_Sample_42_merged   Sparse     Normal      Normal             +   \n",
       "G2_Sample_43_merged   Sparse   Affected      Normal             +   \n",
       "G2_Sample_44_merged   Sparse   Affected      Normal             +   \n",
       "G2_Sample_46_merged   Sparse   Affected      Normal             +   \n",
       "G2_Sample_47_merged   Sparse   Affected      Normal             +   \n",
       "G2_Sample_48_merged   Sparse   Affected  Dysplastic             +   \n",
       "G2_Sample_49_merged   Sparse   Affected  Dysplastic             +   \n",
       "G2_Sample_50_merged   Sparse   Affected      Normal             +   \n",
       "G2_Sample_51_merged    Normal    Normal      Normal             +   \n",
       "G2_Sample_52_merged   Sparse   Affected         NaN             +   \n",
       "G2_Sample_53_merged    Normal  Affected  Dysplastic             +   \n",
       "\n",
       "                     Peg shape teeth         Y  \n",
       "G1_Sample_006_merged               +  moderate  \n",
       "G1_Sample_007_merged               +    Severe  \n",
       "G1_Sample_013_merged               +    Severe  \n",
       "G1_Sample_014_merged               +  moderate  \n",
       "G1_Sample_015_merged               +    Severe  \n",
       "G1_Sample_017_merged             NaN    Severe  \n",
       "G1_Sample_021_merged               +  moderate  \n",
       "G1_Sample_022_merged             NaN    Severe  \n",
       "G1_Sample_024_merged             NaN    Severe  \n",
       "G2_Sample_5_merged               NaN  moderate  \n",
       "G2_Sample_11_merged              NaN    Severe  \n",
       "G2_Sample_12_merged                +    Severe  \n",
       "G2_Sample_33_merged              NaN    Severe  \n",
       "G2_Sample_34_merged                -  moderate  \n",
       "G2_Sample_35_merged                +  moderate  \n",
       "G2_Sample_36_merged                -  moderate  \n",
       "G2_Sample_37_merged                +    Severe  \n",
       "G2_Sample_38_merged                +  moderate  \n",
       "G2_Sample_39_merged                +  moderate  \n",
       "G2_Sample_40_merged                +  moderate  \n",
       "G2_Sample_41_merged                -    Severe  \n",
       "G2_Sample_42_merged                +  moderate  \n",
       "G2_Sample_43_merged              NaN    Severe  \n",
       "G2_Sample_44_merged                +  moderate  \n",
       "G2_Sample_46_merged                +    Severe  \n",
       "G2_Sample_47_merged                +    Severe  \n",
       "G2_Sample_48_merged                +    Severe  \n",
       "G2_Sample_49_merged                +    Severe  \n",
       "G2_Sample_50_merged              NaN    Severe  \n",
       "G2_Sample_51_merged                +  moderate  \n",
       "G2_Sample_52_merged              NaN    Severe  \n",
       "G2_Sample_53_merged                +  moderate  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "G1_G2_clinical"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "3be5a5fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(32, 11)"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "G1_G2_clinical.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "accd0565-252d-4646-b282-08682a79b36e",
   "metadata": {},
   "source": [
    "## save Group3_OUTPUT separately as .csv file to be used as test group for ML external testing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "5ebb14f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "G3_Sample_1_merged      Severe\n",
       "G3_Sample_2_merged    moderate\n",
       "G3_Sample_3_merged      Severe\n",
       "G3_Sample_4_merged      Severe\n",
       "G3_Sample_5_merged    moderate\n",
       "G3_Sample_6_merged      Severe\n",
       "G3_Sample_7_merged    moderate\n",
       "G3_Sample_8_merged    moderate\n",
       "G3_Sample_9_merged    moderate\n",
       "Name: Y, dtype: object"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "G3_OUTPUT_testing = clinical_data.iloc[32:, -1]\n",
    "G3_OUTPUT_testing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "68cdb355",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 10 ms, sys: 0 ns, total: 10 ms\n",
      "Wall time: 9.23 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "Group_3_OUTput_TestGp = '/home/phoebe/phd/ML_csv/'\n",
    "file_name = 'Group_3_OUTput.csv'\n",
    "file_out_path = os.path.join(Group_3_TestGp , file_name)\n",
    "G3_OUTPUT_testing.to_csv(file_out_path, index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a6e0eb6c",
   "metadata": {},
   "source": [
    "> ## Combined DF containig: G1 & G2 RAW Genotype + Clinical data for both groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "9b3329be",
   "metadata": {},
   "outputs": [],
   "source": [
    "G1_G2_ALL = pd.merge(G1_G2_NoNull, G1_G2_clinical, left_index=True, right_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "ec0339d4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr1-10002921</th>\n",
       "      <th>chr1-100058793</th>\n",
       "      <th>chr1-10007418</th>\n",
       "      <th>chr1-1000966</th>\n",
       "      <th>chr1-1001028</th>\n",
       "      <th>chr1-100111956</th>\n",
       "      <th>chr1-1001177</th>\n",
       "      <th>chr1-1001233</th>\n",
       "      <th>chr1-1001270</th>\n",
       "      <th>chr1-1001290</th>\n",
       "      <th>...</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Age/Year</th>\n",
       "      <th>sweating</th>\n",
       "      <th>Scalp Hair</th>\n",
       "      <th>eyebrow</th>\n",
       "      <th>skin</th>\n",
       "      <th>Nails</th>\n",
       "      <th>Missing teeth</th>\n",
       "      <th>Peg shape teeth</th>\n",
       "      <th>Y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G1_Sample_006_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_007_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>3.50</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_013_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>13.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_014_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>2.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_015_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>5.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_017_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>2.25</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_021_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>9.42</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_022_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>3.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G1_Sample_024_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>3.83</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_11_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>1.75</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_12_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>1.67</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_33_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>1.50</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_34_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>6.50</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>-</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_35_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>16.25</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_36_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>14.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_37_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>4.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_38_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>20.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_39_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>8.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_40_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>5.58</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_41_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>2.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_42_merged</th>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>3.67</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_43_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>6.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_44_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>8.17</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_46_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>6.50</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_47_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_48_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>3.42</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_49_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>3.50</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_50_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>1.75</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_51_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>12.67</td>\n",
       "      <td>Hypohidrotic</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_52_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>1.92</td>\n",
       "      <td>Anhidrotic</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Affected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Severe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_53_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>M</td>\n",
       "      <td>10.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Dysplastic</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2_Sample_5_merged</th>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>...</td>\n",
       "      <td>F</td>\n",
       "      <td>25.00</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Sparse</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>moderate</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32 rows × 169590 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     chr1-10002921 chr1-100058793 chr1-10007418 chr1-1000966  \\\n",
       "G1_Sample_006_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_007_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_013_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_014_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_015_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_017_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_021_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_022_merged           0/0            0/0           0/0          0/0   \n",
       "G1_Sample_024_merged           0/0            0/0           0/0          0/0   \n",
       "G2_Sample_11_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_12_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_33_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_34_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_35_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_36_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_37_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_38_merged            0/0            1/1           0/0          0/0   \n",
       "G2_Sample_39_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_40_merged            0/0            0/0           0/1          0/0   \n",
       "G2_Sample_41_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_42_merged            0/1            0/0           0/0          0/0   \n",
       "G2_Sample_43_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_44_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_46_merged            0/0            0/0           0/0          0/1   \n",
       "G2_Sample_47_merged            0/0            0/1           0/0          0/0   \n",
       "G2_Sample_48_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_49_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_50_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_51_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_52_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_53_merged            0/0            0/0           0/0          0/0   \n",
       "G2_Sample_5_merged             0/0            0/0           0/0          0/0   \n",
       "\n",
       "                     chr1-1001028 chr1-100111956 chr1-1001177 chr1-1001233  \\\n",
       "G1_Sample_006_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_007_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_013_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_014_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_015_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_017_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_021_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_022_merged          0/0            0/0          0/0          0/0   \n",
       "G1_Sample_024_merged          0/0            0/0          0/0          0/0   \n",
       "G2_Sample_11_merged           0/0            0/0          1/1          0/1   \n",
       "G2_Sample_12_merged           0/0            0/0          0/1          0/0   \n",
       "G2_Sample_33_merged           0/0            0/1          1/1          0/0   \n",
       "G2_Sample_34_merged           0/0            0/0          0/0          0/0   \n",
       "G2_Sample_35_merged           0/0            0/0          0/0          0/0   \n",
       "G2_Sample_36_merged           0/0            0/0          0/1          0/0   \n",
       "G2_Sample_37_merged           0/0            0/0          1/1          0/0   \n",
       "G2_Sample_38_merged           0/0            0/1          1/1          0/0   \n",
       "G2_Sample_39_merged           0/0            1/1          0/1          0/0   \n",
       "G2_Sample_40_merged           0/0            0/0          0/1          0/0   \n",
       "G2_Sample_41_merged           0/0            0/1          1/1          0/0   \n",
       "G2_Sample_42_merged           0/0            1/1          1/1          0/0   \n",
       "G2_Sample_43_merged           0/0            0/1          1/1          0/0   \n",
       "G2_Sample_44_merged           0/0            0/0          0/0          0/0   \n",
       "G2_Sample_46_merged           0/0            0/1          0/1          0/0   \n",
       "G2_Sample_47_merged           0/0            0/0          0/1          0/0   \n",
       "G2_Sample_48_merged           0/0            0/1          1/1          0/0   \n",
       "G2_Sample_49_merged           0/0            0/0          0/0          0/0   \n",
       "G2_Sample_50_merged           0/0            0/0          1/1          0/0   \n",
       "G2_Sample_51_merged           0/0            0/0          0/1          0/0   \n",
       "G2_Sample_52_merged           0/0            0/1          0/1          0/0   \n",
       "G2_Sample_53_merged           0/0            0/1          0/1          0/0   \n",
       "G2_Sample_5_merged            0/1            0/0          0/0          0/0   \n",
       "\n",
       "                     chr1-1001270 chr1-1001290  ... Sex Age/Year  \\\n",
       "G1_Sample_006_merged          0/0          0/0  ...   F    10.00   \n",
       "G1_Sample_007_merged          0/0          0/0  ...   F     3.50   \n",
       "G1_Sample_013_merged          0/0          0/0  ...   M    13.00   \n",
       "G1_Sample_014_merged          0/0          0/0  ...   M     2.00   \n",
       "G1_Sample_015_merged          0/0          0/0  ...   M     5.00   \n",
       "G1_Sample_017_merged          0/0          0/0  ...   M     2.25   \n",
       "G1_Sample_021_merged          0/0          0/0  ...   F     9.42   \n",
       "G1_Sample_022_merged          0/0          0/0  ...   F     3.00   \n",
       "G1_Sample_024_merged          0/0          0/0  ...   M     3.83   \n",
       "G2_Sample_11_merged           0/0          0/0  ...   F     1.75   \n",
       "G2_Sample_12_merged           0/0          0/0  ...   F     1.67   \n",
       "G2_Sample_33_merged           0/0          0/1  ...   M     1.50   \n",
       "G2_Sample_34_merged           0/0          0/0  ...   F     6.50   \n",
       "G2_Sample_35_merged           0/0          0/0  ...   M    16.25   \n",
       "G2_Sample_36_merged           0/0          0/0  ...   F    14.00   \n",
       "G2_Sample_37_merged           0/0          0/1  ...   M     4.00   \n",
       "G2_Sample_38_merged           0/0          0/1  ...   M    20.00   \n",
       "G2_Sample_39_merged           0/0          0/0  ...   M     8.00   \n",
       "G2_Sample_40_merged           0/0          0/1  ...   M     5.58   \n",
       "G2_Sample_41_merged           0/0          0/0  ...   F     2.50   \n",
       "G2_Sample_42_merged           0/0          0/1  ...   M     3.67   \n",
       "G2_Sample_43_merged           0/0          0/0  ...   M     6.50   \n",
       "G2_Sample_44_merged           0/0          0/0  ...   F     8.17   \n",
       "G2_Sample_46_merged           0/0          0/1  ...   M     6.50   \n",
       "G2_Sample_47_merged           0/0          0/0  ...   F    10.00   \n",
       "G2_Sample_48_merged           0/0          0/0  ...   F     3.42   \n",
       "G2_Sample_49_merged           0/0          0/0  ...   F     3.50   \n",
       "G2_Sample_50_merged           0/0          0/0  ...   M     1.75   \n",
       "G2_Sample_51_merged           0/1          0/0  ...   F    12.67   \n",
       "G2_Sample_52_merged           0/0          0/0  ...   M     1.92   \n",
       "G2_Sample_53_merged           0/0          0/0  ...   M    10.00   \n",
       "G2_Sample_5_merged            0/0          0/0  ...   F    25.00   \n",
       "\n",
       "                          sweating Scalp Hair  eyebrow      skin       Nails  \\\n",
       "G1_Sample_006_merged  Hypohidrotic     Normal   Sparse    Normal      Normal   \n",
       "G1_Sample_007_merged  Hypohidrotic     Sparse   Sparse  Affected  Dysplastic   \n",
       "G1_Sample_013_merged    Anhidrotic     Sparse   Sparse  Affected  Dysplastic   \n",
       "G1_Sample_014_merged        Normal     Sparse   Sparse    Normal      Normal   \n",
       "G1_Sample_015_merged    Anhidrotic     Sparse   Sparse  Affected  Dysplastic   \n",
       "G1_Sample_017_merged  Hypohidrotic     Sparse   Sparse  Affected      Normal   \n",
       "G1_Sample_021_merged        Normal     Normal   Normal    Normal      Normal   \n",
       "G1_Sample_022_merged    Anhidrotic     Sparse   Sparse  Affected      Normal   \n",
       "G1_Sample_024_merged    Anhidrotic     Sparse   Sparse  Affected      Normal   \n",
       "G2_Sample_11_merged   Hypohidrotic    Sparse   Sparse   Affected      Normal   \n",
       "G2_Sample_12_merged   Hypohidrotic    Sparse   Sparse   Affected      Normal   \n",
       "G2_Sample_33_merged   Hypohidrotic    Sparse   Sparse   Affected      Normal   \n",
       "G2_Sample_34_merged         Normal    Sparse         -  Affected      Normal   \n",
       "G2_Sample_35_merged         Normal     Normal   Normal  Affected  Dysplastic   \n",
       "G2_Sample_36_merged         Normal     Normal  Sparse   Affected      Normal   \n",
       "G2_Sample_37_merged     Anhidrotic    Sparse   Sparse   Affected         NaN   \n",
       "G2_Sample_38_merged            NaN    Sparse   Sparse   Affected      Normal   \n",
       "G2_Sample_39_merged         Normal     Normal  Sparse   Affected      Normal   \n",
       "G2_Sample_40_merged         Normal     Normal   Normal    Normal      Normal   \n",
       "G2_Sample_41_merged     Anhidrotic    Sparse   Sparse        NaN         NaN   \n",
       "G2_Sample_42_merged         Normal    Sparse   Sparse     Normal      Normal   \n",
       "G2_Sample_43_merged     Anhidrotic    Sparse   Sparse   Affected      Normal   \n",
       "G2_Sample_44_merged         Normal     Normal  Sparse   Affected      Normal   \n",
       "G2_Sample_46_merged     Anhidrotic    Sparse   Sparse   Affected      Normal   \n",
       "G2_Sample_47_merged     Anhidrotic    Sparse   Sparse   Affected      Normal   \n",
       "G2_Sample_48_merged         Normal     Normal  Sparse   Affected  Dysplastic   \n",
       "G2_Sample_49_merged         Normal    Sparse   Sparse   Affected  Dysplastic   \n",
       "G2_Sample_50_merged     Anhidrotic    Sparse   Sparse   Affected      Normal   \n",
       "G2_Sample_51_merged   Hypohidrotic     Normal   Normal    Normal      Normal   \n",
       "G2_Sample_52_merged     Anhidrotic    Sparse   Sparse   Affected         NaN   \n",
       "G2_Sample_53_merged         Normal     Normal   Normal  Affected  Dysplastic   \n",
       "G2_Sample_5_merged          Normal    Sparse   Sparse     Normal         NaN   \n",
       "\n",
       "                     Missing teeth Peg shape teeth         Y  \n",
       "G1_Sample_006_merged             +               +  moderate  \n",
       "G1_Sample_007_merged             +               +    Severe  \n",
       "G1_Sample_013_merged             +               +    Severe  \n",
       "G1_Sample_014_merged             +               +  moderate  \n",
       "G1_Sample_015_merged             +               +    Severe  \n",
       "G1_Sample_017_merged             +             NaN    Severe  \n",
       "G1_Sample_021_merged             +               +  moderate  \n",
       "G1_Sample_022_merged             +             NaN    Severe  \n",
       "G1_Sample_024_merged             +             NaN    Severe  \n",
       "G2_Sample_11_merged              +             NaN    Severe  \n",
       "G2_Sample_12_merged              +               +    Severe  \n",
       "G2_Sample_33_merged              +             NaN    Severe  \n",
       "G2_Sample_34_merged              +               -  moderate  \n",
       "G2_Sample_35_merged              +               +  moderate  \n",
       "G2_Sample_36_merged              +               -  moderate  \n",
       "G2_Sample_37_merged              +               +    Severe  \n",
       "G2_Sample_38_merged              +               +  moderate  \n",
       "G2_Sample_39_merged              +               +  moderate  \n",
       "G2_Sample_40_merged              +               +  moderate  \n",
       "G2_Sample_41_merged              +               -    Severe  \n",
       "G2_Sample_42_merged              +               +  moderate  \n",
       "G2_Sample_43_merged              +             NaN    Severe  \n",
       "G2_Sample_44_merged              +               +  moderate  \n",
       "G2_Sample_46_merged              +               +    Severe  \n",
       "G2_Sample_47_merged              +               +    Severe  \n",
       "G2_Sample_48_merged              +               +    Severe  \n",
       "G2_Sample_49_merged              +               +    Severe  \n",
       "G2_Sample_50_merged              +             NaN    Severe  \n",
       "G2_Sample_51_merged              +               +  moderate  \n",
       "G2_Sample_52_merged              +             NaN    Severe  \n",
       "G2_Sample_53_merged              +               +  moderate  \n",
       "G2_Sample_5_merged               +             NaN  moderate  \n",
       "\n",
       "[32 rows x 169590 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "G1_G2_ALL"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bccb5411",
   "metadata": {},
   "source": [
    "## Save the resulted  G1 & G2 RAW Genotype + Clinical data for G1 & G2 in a .csv format "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "5387f662",
   "metadata": {},
   "outputs": [],
   "source": [
    "G1_G2_RAW_DataFrames = '/home/phoebe/phd/ML_csv/'\n",
    "file_name = 'G1_G2_ALL.csv'\n",
    "file_out_path = os.path.join(G1_G2_RAW_DataFrames , file_name)\n",
    "G1_G2_ALL.to_csv(file_out_path)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e6b7e1b4",
   "metadata": {},
   "source": [
    "# ------------------------------------------------END---------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc8c6a5c",
   "metadata": {},
   "source": [
    "> # Genotype Dataframe With Encoding"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "97089fc7",
   "metadata": {},
   "source": [
    "## Group_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ba8a2de",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "### Extract Genotype Information for each and every single SNP in every single patient\n",
    "### and store these Informations from each patient in a list to then concatenate them in a single Dataframe \n",
    "\n",
    "\n",
    "G1_coded_list= []\n",
    "for file in tqdm(os.listdir(Gp1_Ready_SNPs_VCF)):\n",
    "    file_path = os.path.join(Gp1_Ready_SNPs_VCF,file)\n",
    "    Sample_ID = '_'.join(os.path.basename(file).split('_')[0:4]) \n",
    "\n",
    "## Read every single file in a separate df\n",
    "## print df.shape to get idea about ur df dimensions.\n",
    "    df =pd.read_csv(file_path) \n",
    "    print(Sample_ID, df.shape)                         \n",
    "    \n",
    "    ## Before apply ordinal encoding which is additive encoding counting number of minor alleles i.e. \n",
    "    ## genotype \"0/0\" will be coded as '0' caz it contains zero minor alelles\n",
    "    ## genotype \"0/1\" will be coded as '1' caz it contains 1 minor alelles\n",
    "    ## genotype \"1/1\" will be coded as '2' caz it contains 2 minor alelles\n",
    "    ## first replace different genotypes as following:\n",
    "    print ('GT: ')\n",
    "    print(df['GT'].value_counts())\n",
    "    df['GT'] = df['GT'].str.replace('0/2','0/1')\n",
    "    df['GT'] = df['GT'].str.replace('1/2','1/1')\n",
    "\n",
    "## Apply OrdinalEncoder to the Genotype column with the following categories ordered as follow:\n",
    "## ordered categories ['0/0','0/1', '0/2','1/1', '1/2'] will take orders as ['0','1','2','3','4']\n",
    "## And add new column named 'Sample_ID' with the genotype encoded:\n",
    "## this naming will be of use when transposing the table so that each group of SNPs will be indexed with their Sample_ID\n",
    "    \n",
    "    \n",
    "    GT_categ = ['0/0','0/1','1/1']\n",
    "    OE = OrdinalEncoder(categories= [GT_categ], handle_unknown= 'use_encoded_value', unknown_value= -1)\n",
    "    df[Sample_ID] = pd.DataFrame(OE.fit_transform(df[['GT']]) )\n",
    "   \n",
    "    print(Sample_ID, df.shape)\n",
    "\n",
    "## combine CHROM + POS in one variable\n",
    "## Create new column called SNP_ID and insert it into df\n",
    "    \n",
    "    Head = df[\"CHROM\"] + '-' + df[\"POS\"].astype(str)   \n",
    "    df.insert(loc=1, column= 'SNP_ID', value= Head) \n",
    "    \n",
    "## Now use multiple filters on ur df to include only SNPs that has     \n",
    "## 'FILTER' == PASS and that has \n",
    "## 'AF' >= 0.2  .....???\n",
    "## 'QUAL' >= 20.01\n",
    "## 'CHROM' == ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "\n",
    "    df = df[(df['FILTER']=='PASS')  & (df['QUAL']>=20.1)]  \n",
    "    chromosome = ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "    df = df[df[\"CHROM\"].isin(chromosome)]\n",
    "    print(Sample_ID, df.shape)          ## Note that the df.shape now has fewer rows for SNPs that was filtered out\n",
    "    print ('GT: ')\n",
    "    print (df['GT'].value_counts())\n",
    "      \n",
    "    df = df[['SNP_ID','GT',Sample_ID]]\n",
    "    df = df.T  \n",
    "    df.columns= df.iloc[0]\n",
    "    df = df.drop(['SNP_ID','GT'])\n",
    "#     #print(Sample_ID, df.shape)\n",
    "    G1_coded_list.append(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "518fcb8c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%time\n",
    "## Dataframe containing all LABELTRANSFORMED Genotypes of All SNPs in all samples\n",
    "\n",
    "G1_coded = pd.concat(G1_coded_list,axis=0)\n",
    "G1_coded"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c972d8e1",
   "metadata": {},
   "source": [
    "## Group_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8abc47b0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "### Extract Genotype Information for each and every single SNP in every single patient\n",
    "### and store these Informations from each patient in a list to then concatenate them in a single Dataframe \n",
    "\n",
    "\n",
    "G2_coded_list= []\n",
    "for file in tqdm(os.listdir(Gp2_Ready_SNPs_VCF)):\n",
    "    file_path = os.path.join(Gp2_Ready_SNPs_VCF,file)\n",
    "    Sample_ID = 'G2_' + '_'.join(os.path.basename(file).split('_')[0:3])\n",
    "\n",
    "## Read every single file in a separate df\n",
    "## print df.shape to get idea about ur df dimensions.\n",
    "    df =pd.read_csv(file_path, index_col='Unnamed: 0') \n",
    "    print(Sample_ID, df.shape)                         \n",
    "\n",
    "    ## Before apply ordinal encoding which is additive encoding counting number of minor alleles i.e. \n",
    "    ## genotype \"0/0\" will be coded as '0' caz it contains zero minor alelles\n",
    "    ## genotype \"0/1\" will be coded as '1' caz it contains 1 minor alelles\n",
    "    ## genotype \"1/1\" will be coded as '2' caz it contains 2 minor alelles\n",
    "    ## first replace different genotypes as following:\n",
    "    print ('GT: ')\n",
    "    print(df['GT'].value_counts())\n",
    "    df['GT'] = df['GT'].str.replace('0/2','0/1')\n",
    "    df['GT'] = df['GT'].str.replace('1/2','1/1')    \n",
    "    \n",
    "    \n",
    "## Apply OrdinalEncoder to the Genotype column with the following categories ordered as follow:\n",
    "## ordered categories ['0/0','0/1', '0/2','1/1', '1/2'] will take orders as ['0','1','2','3','4']\n",
    "## And add new column named 'Sample_ID' with the genotype encoded:\n",
    "## this naming will be of use when transposing the table so that each group of SNPs will be indexed with their Sample_ID\n",
    "    \n",
    "    GT_categ = ['0/0','0/1','1/1']\n",
    "    OE = OrdinalEncoder(categories= [GT_categ], handle_unknown= 'use_encoded_value', unknown_value= -1)\n",
    "    df[Sample_ID] = pd.DataFrame(OE.fit_transform(df[['GT']]) )\n",
    "   \n",
    "    print(Sample_ID, df.shape)\n",
    "\n",
    "## combine CHROM + POS in one variable\n",
    "## Create new column called SNP_ID and insert it into df\n",
    "    \n",
    "    Head = df[\"CHROM\"] + '-' + df[\"POS\"].astype(str)   \n",
    "    df.insert(loc=1, column= 'SNP_ID', value= Head) \n",
    "    \n",
    "## Now use multiple filters on ur df to include only SNPs that has     \n",
    "## 'FILTER' == PASS and that has \n",
    "## 'AF' >= 0.2  .....???\n",
    "## 'QUAL' >= 20.01\n",
    "## 'CHROM' == ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "\n",
    "    df = df[(df['FILTER']=='PASS')  & (df['QUAL']>=20.1)]  \n",
    "    chromosome = ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "    df = df[df[\"CHROM\"].isin(chromosome)]\n",
    "    print(Sample_ID, df.shape)          ## Note that the df.shape now has fewer rows for SNPs that was filtered out\n",
    "    print ('GT: ')\n",
    "    print(df['GT'].value_counts())\n",
    "      \n",
    "    df = df[['SNP_ID','GT',Sample_ID]]\n",
    "    df = df.T  \n",
    "    df.columns= df.iloc[0]\n",
    "    df = df.drop(['SNP_ID','GT'])\n",
    "#     #print(Sample_ID, df.shape)\n",
    "    G2_coded_list.append(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "425e7fd6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%time\n",
    "## Dataframe containing all LABELTRANSFORMED Genotypes of All SNPs in all samples\n",
    "\n",
    "G2_coded = pd.concat(G2_coded_list,axis=0)\n",
    "G2_coded"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "374293c2",
   "metadata": {},
   "source": [
    "## Group_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2cd6b52d",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Extract Genotype Information for each and every single SNP in every single patient\n",
    "### and store these Informations from each patient in a list to then concatenate them in a single Dataframe \n",
    "\n",
    "\n",
    "G3_coded_list= []\n",
    "for file in tqdm(os.listdir(Gp3_Ready_SNPs_VCF)):\n",
    "    file_path = os.path.join(Gp3_Ready_SNPs_VCF,file)\n",
    "    Sample_ID = 'G3_' + '_'.join(os.path.basename(file).split('_')[0:3])\n",
    "\n",
    "## Read every single file in a separate df\n",
    "## print df.shape to get idea about ur df dimensions.\n",
    "    df =pd.read_csv(file_path, index_col='Unnamed: 0') \n",
    "    print(Sample_ID, df.shape)  \n",
    "    \n",
    "    \n",
    "    ## Before apply ordinal encoding which is additive encoding counting number of minor alleles i.e. \n",
    "    ## genotype \"0/0\" will be coded as '0' caz it contains zero minor alelles\n",
    "    ## genotype \"0/1\" will be coded as '1' caz it contains 1 minor alelles\n",
    "    ## genotype \"1/1\" will be coded as '2' caz it contains 2 minor alelles\n",
    "    ## first replace different genotypes as following:\n",
    "    print ('GT: ')\n",
    "    print(df['GT'].value_counts())\n",
    "    df['GT'] = df['GT'].str.replace('0/2','0/1')\n",
    "    df['GT'] = df['GT'].str.replace('1/2','1/1') \n",
    "    df['GT'] = df['GT'].str.replace('0|1','0/1', regex=False) \n",
    "    df['GT'] = df['GT'].str.replace('1|1','1/1', regex=False)\n",
    "    df['GT'] = df['GT'].str.replace('1|2','1/2', regex=False) \n",
    "    \n",
    "    \n",
    "## Apply OrdinalEncoder to the Genotype column with the following categories ordered as follow:\n",
    "## ordered categories ['0/0','0/1', '0/2','1/1', '1/2'] will take orders as ['0','1','2','3','4']\n",
    "## And add new column named 'Sample_ID' with the genotype encoded:\n",
    "## this naming will be of use when transposing the table so that each group of SNPs will be indexed with their Sample_ID\n",
    "    \n",
    "    GT_categ = ['0/0','0/1','1/1']\n",
    "    OE = OrdinalEncoder(categories= [GT_categ], handle_unknown= 'use_encoded_value', unknown_value= -1)\n",
    "    df[Sample_ID] = pd.DataFrame(OE.fit_transform(df[['GT']]) )\n",
    "   \n",
    "    print(Sample_ID, df.shape)\n",
    "\n",
    "## combine CHROM + POS in one variable\n",
    "## Create new column called SNP_ID and insert it into df\n",
    "    \n",
    "    Head = df[\"CHROM\"] + '-' + df[\"POS\"].astype(str)   \n",
    "    df.insert(loc=1, column= 'SNP_ID', value= Head) \n",
    "    \n",
    "## Now use multiple filters on ur df to include only SNPs that has     \n",
    "## 'FILTER' == PASS and that has \n",
    "## 'AF' >= 0.2  .....???\n",
    "## 'QUAL' >= 20.01\n",
    "## 'CHROM' == ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "\n",
    "    df = df[(df['FILTER']=='PASS')  & (df['QUAL']>=20.1)]  \n",
    "    chromosome = ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "    df = df[df[\"CHROM\"].isin(chromosome)]\n",
    "    print(Sample_ID, df.shape)          ## Note that the df.shape now has fewer rows for SNPs that was filtered out\n",
    "    print ('GT: ')\n",
    "    print(df['GT'].value_counts())\n",
    "    \n",
    "    \n",
    "    df = df[['SNP_ID','GT',Sample_ID]]\n",
    "    df = df.T  \n",
    "    df.columns= df.iloc[0]\n",
    "    df = df.drop(['SNP_ID','GT'])\n",
    "#     #print(Sample_ID, df.shape)\n",
    "    G3_coded_list.append(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4f4aca16",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "## Dataframe containing all LABELTRANSFORMED Genotypes of All SNPs in all samples\n",
    "\n",
    "G3_coded = pd.concat(G3_coded_list,axis=0)\n",
    "G3_coded"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8938d4f6",
   "metadata": {},
   "source": [
    "# Now we have 3 separate DF for the three groups of patients the plan is to use the first 2 groups for train and validation and save the 3rd group for testing"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dfe8b583",
   "metadata": {},
   "source": [
    "> ## Concatenate the first two groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05ceae74",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%time\n",
    "## Dataframe containing all Genotypes of All SNPs in all samples in G1 + G2\n",
    "\n",
    "G1_G2_Coded = pd.concat([G1_coded, G2_coded],axis=0)\n",
    "G1_G2_Coded"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "397a1601",
   "metadata": {},
   "source": [
    "> ## Sort by rows and by columns "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf003e05",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## Sort df by row index label\n",
    "G1_G2_Coded_Sorted = G1_G2_Coded.sort_index(axis=0)\n",
    "\n",
    "## Sort df by column index label\n",
    "G1_G2_Coded_Sorted = G1_G2_Coded_Sorted.sort_index(axis=1)\n",
    "G1_G2_Coded_Sorted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49fde74f",
   "metadata": {},
   "source": [
    "> ## Replace nan values in Genotype DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f2dfc99",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## from my point of view the null values could be replaced by the genotype '0/0'\n",
    "## as an example to explain the SNP = chr1-10002921\texists only in sample 'G2_Sample_42_SNPs' and NAN in all others\n",
    "## since it does not appear in others this means that their genotype at thie exact SNP are homozygous wildtype '0/0'\n",
    "## and thus were not called by the variant caller in those samples so we replase them by '0/0'\n",
    "\n",
    "G1_G2_Coded_NoNull = G1_G2_Coded_Sorted.fillna('0')\n",
    "G1_G2_Coded_NoNull"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a708a21f",
   "metadata": {},
   "outputs": [],
   "source": [
    "G1_G2_Coded_NoNull.dtypes.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cbaf95fb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "614a8c0f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "13affcca",
   "metadata": {},
   "source": [
    "> ## Combined DF containig: G1 & G2 coded SNPs Genotype + Clinical data for both groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c234465e",
   "metadata": {},
   "outputs": [],
   "source": [
    "G1_G2_coded_df = pd.merge(G1_G2_Coded_NoNull, G1_G2_clinical, left_index=True, right_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84014f8c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "G1_G2_coded_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e9a66aaf",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "3032971d",
   "metadata": {},
   "source": [
    "# Save the resulted CODED Genotype DataFrame in a .csv format "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46d3b862",
   "metadata": {},
   "outputs": [],
   "source": [
    "G1_G2_coded_DataFrames = '/home/phoebe/phd/ML_csv/'\n",
    "file_name = 'G1_G2_coded_df_Two_output.csv'\n",
    "file_out_path = os.path.join(G1_G2_coded_DataFrames , file_name)\n",
    "G1_G2_coded_df.to_csv(file_out_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f27d40bc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2ad9786",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "567af385",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c883033",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fc160d8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "2fd52caa",
   "metadata": {},
   "source": [
    "> ## Alternative Allele Dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83431e24",
   "metadata": {},
   "source": [
    "# Group_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b2f0dc0d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "### Extract Alternative Allele Information for each and every single SNP in every single patient\n",
    "### and store these Informations from each patient in a list to then concatenate them in a single Dataframe \n",
    "### containing all Alternative alleles at each SNP for every patient.\n",
    "\n",
    "df_ALT_list= []\n",
    "\n",
    "for file in tqdm(os.listdir(Gp2_Ready_SNPs_F1_VCF)):\n",
    "    file_path = os.path.join(Gp2_Ready_SNPs_F1_VCF,file)\n",
    "    Sample_ID = 'G2_' + '_'.join(os.path.basename(file).split('_')[0:3])\n",
    "\n",
    "## Read every single file in a separate df\n",
    "## print df.shape to get idea about ur df dimensions.\n",
    "    df =pd.read_csv(file_path, index_col='Unnamed: 0') \n",
    "    print(Sample_ID, df.shape)                         \n",
    "    \n",
    "## combine CHROM + POS in one variable\n",
    "## Create new column called SNP_ID and insert it into df\n",
    "    \n",
    "    Head = df[\"CHROM\"] + '-' + df[\"POS\"].astype(str)   \n",
    "    df.insert(loc=1, column= 'SNP_ID', value= Head) \n",
    "    \n",
    "## Now use multiple filters on ur df to include only SNPs that has     \n",
    "## 'FILTER' == PASS and that has \n",
    "## 'AF' >= 0.2  .....???\n",
    "## 'QUAL' >= 20.01\n",
    "## 'CHROM' == ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "\n",
    "    df = df[(df['FILTER']=='PASS')  & (df['QUAL']>=20.1)]  \n",
    "    chromosome = ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "    df = df[df[\"CHROM\"].isin(chromosome)]\n",
    "    print(Sample_ID, df.shape)          ## Note that the df.shape now has fewer rows for SNPs that was filtered out\n",
    "    df = df[['SNP_ID','REF','ALT']]\n",
    "    df = df.rename(columns = {'REF': Sample_ID+ '_REF', 'ALT': Sample_ID+ '_ALT'})\n",
    "    df = df.T  \n",
    "    df.columns= df.iloc[0]\n",
    "    df = df.drop(['SNP_ID'])\n",
    "    print(Sample_ID, df.shape)\n",
    "    df_ALT_list.append(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "72a96744",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%time\n",
    "## Dataframe containing all Alternative SNPs in All samples\n",
    "\n",
    "df3 = pd.concat(df_ALT_list,axis=0)\n",
    "df3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "19dca8e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Sort df by row index label\n",
    "df3_Sorted = df3.sort_index(axis=0)\n",
    "\n",
    "## Sort df by column index label\n",
    "df3_Sorted = df3_Sorted.sort_index(axis=1)\n",
    "df3_Sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f0ff71f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "df3_Sorted.columns.get_loc(\"chr1-10264\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8636d17a",
   "metadata": {},
   "outputs": [],
   "source": [
    "len(df3_Sorted.columns)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ad30039c",
   "metadata": {},
   "source": [
    ">> ## Replace nan values in Alternative allele DataFrame\n",
    "> ### Replace each column nan values with the reference allele of that SNP\n",
    ">>>  ## Very slow piece of code"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c6ce658",
   "metadata": {},
   "source": [
    "##  ## THIS METHOD T00K 12 HOURS TO COMPLETE\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d44cb8c6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%time\n",
    "\n",
    "## create a variable containing all elements of every col as pd.Series and Get the last valid value \n",
    "## because the df is sorted so for each patient their exists two rows first for Alt allele and second for Ref allele\n",
    "## caz I want to refill Nan with Ref allele so we are looking in evevry column for the 'last_valid_index()' \n",
    "## not the 'first_valid_index()' this is because the file is sorted and the Ref row came after the Alt row\n",
    "## Then use this value to replace nan's in this specific column: \n",
    "counter = 0\n",
    "for col in tqdm(df3_Sorted.columns):\n",
    "    s = pd.Series (df3_Sorted[col])\n",
    "    last_valid_Ref_value = s.loc[s.last_valid_index()]\n",
    "    print(col, last_valid_Ref_value)\n",
    "    df3_Sorted[[col]] = df3_Sorted[[col]].fillna(last_valid_Ref_value)\n",
    "    counter +=1"
   ]
  },
  {
   "cell_type": "raw",
   "id": "f770cda2",
   "metadata": {},
   "source": [
    "## create a variable containing all elements of every col as pd.Series and Get the first valid value \n",
    "## Then use this value to replace nan's in this specific column: \n",
    "\n",
    "\n",
    "s = pd.Series (df3_Sorted['chr1-14907'])\n",
    "#print(s)\n",
    "print (s.loc[s.last_valid_index()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00655838",
   "metadata": {},
   "outputs": [],
   "source": [
    "df3_Sorted.isnull().any().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8d59363d",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Drop all Ref rows from the DataFrame\n",
    "#%%time\n",
    "\n",
    "REF_ALLELE = df3_Sorted.filter(regex='REF', axis=0)\n",
    "REF_ALLELE.index\n",
    "df4 = df3_Sorted.drop(REF_ALLELE.index,  axis=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ee9d992",
   "metadata": {},
   "source": [
    "### Store the resulted Alternative Allele DataFrame in a .csv format "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6cca5277",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gp2_SNPs_DataFrames = '/home/phoebe/phd/ML_csv/GP_2/'\n",
    "file_name = 'Gp2_Alt_Allele_df.csv'\n",
    "file_out_path = os.path.join(Gp2_SNPs_DataFrames , file_name)\n",
    "df4.to_csv(file_out_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "89baa446",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gp2_Alt_Allele_df = pd.read_csv('/home/phoebe/phd/ML_csv/GP_2/Gp2_Alt_Allele_df.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ab0a9a3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "Gp2_Alt_Allele_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "246a0829",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3208c043",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df3.loc['G2_Sample_50_SNPs REF']['chrX-152870460']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff9b6afb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "29dff4c7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "f60e03e8",
   "metadata": {},
   "source": [
    "# ------------------------------------------------TEST-------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e4785fbe",
   "metadata": {},
   "source": [
    "# -------------------------------------------------- & ------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a4efede",
   "metadata": {},
   "source": [
    "# ---------------------------------------- Feature Encoding ----------------------------------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "106019d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "G2_Sample_5_SNPs = '/home/phoebe/phd/ML_csv/GP_2/Gp2_SNP_Variant_files/1_Gp2_Ready_SNPs_F1_VCF/Sample_5_SNPs_ML.csv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "709898d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Read a sample file:\n",
    "\n",
    "df = pd.read_csv(G2_Sample_5_SNPs, index_col= 'Unnamed: 0')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed3d9e70",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "696d2bf6",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[df[\"GT\"]== '0/0']\n",
    "df['QUAL']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bb4204cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "chromosome = ['chr1', 'chr2', 'chr3', 'chr9', 'chr22', 'chrX']\n",
    "df = df[df[\"CHROM\"].isin(chromosome)]\n",
    "df['Sample_info']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "edd81e63",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5f2de31",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.loc[df['CHROM']== 'chr1']  \n",
    "\n",
    "#df = df.loc[(df['FILTER']=='PASS') & (df['QUAL']>=20.1)]  \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "68757323",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "5f49c09c",
   "metadata": {},
   "source": [
    "> # Genotype Encoding using sklearn OrdinalEncoder\n",
    " ### In OrdinalEncoder you can arrange the categories in the order you want \n",
    " ### or other wise they will be ordered alphabetically"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a679d82d",
   "metadata": {},
   "outputs": [],
   "source": [
    "print(df['GT'].ndim)\n",
    "print(df['GT'].shape)\n",
    "print(df[['GT']].ndim)\n",
    "print(df[['GT']].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ffd2591e",
   "metadata": {},
   "outputs": [],
   "source": [
    "GT_categ = ['0/0','0/1', '0/2','1/1', '1/2']\n",
    "OE = OrdinalEncoder(categories= [GT_categ])\n",
    "#encoded_df = OE.fit_transform(df[['GT']]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11ca2cc2",
   "metadata": {},
   "outputs": [],
   "source": [
    "df['GT_coded'] = pd.DataFrame(OE.fit_transform(df[['GT']]) )\n",
    "df.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b36c1671",
   "metadata": {},
   "outputs": [],
   "source": [
    "print (df['GT'].dtype)\n",
    "print(df['GT'].unique())\n",
    "print(df['GT_coded'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20551144",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "raw",
   "id": "71f87a1a",
   "metadata": {},
   "source": [
    "OE = OrdinalEncoder(encoded_missing_value=-1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1532be57",
   "metadata": {},
   "outputs": [],
   "source": [
    "Head = df[\"CHROM\"] + '-' + df[\"POS\"].astype(str)   ## combine  SNP (chromosome no) + (exact location) in one variable\n",
    "df.insert(loc=1, column= 'SNP_ID', value= Head)    ## Create new column called SNP_ID and insert it into df\n",
    "\n",
    "## Try different combinations of filter to Filterout some SNPS:\n",
    "\n",
    "#df = df.loc[(df['FILTER']=='PASS')& (df['AF_alt1']>=0.5) & (df['QUAL']>=20.1)] \n",
    "df = df[(df['FILTER']=='PASS') & (df['QUAL']>=20.1)] \n",
    "\n",
    "## New df contain only 2 columns SNP_ID and the SNP genotype:    \n",
    "\n",
    "df = df[['SNP_ID','GT']]\n",
    "#print(df['GT'].value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d66db0f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd6097bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3bab8b67",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a3f38989",
   "metadata": {},
   "outputs": [],
   "source": [
    "df['GT'] = df['GT'].astype(str)\n",
    "df['GT'] = df['GT'].str.strip()\n",
    "df['GT'] = df['GT'].str.replace(',','')\n",
    "df['GT'] = df['GT'].str.rstrip(',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9cff1d5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df.GT.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5047c065",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "3ebba297",
   "metadata": {},
   "source": [
    "> # Genotype Encoding using sklearn LabelEncoder\n",
    "\n",
    "### ** This transformer should be used to encode target values, i.e. y, and not the input X.**\n",
    "### In LabelEncoder categories are labeled from 0 to n-1 categories alphabetically\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "735adace",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.preprocessing import LabelEncoder\n",
    "\n",
    "le = LabelEncoder()\n",
    "df['GT_codeLE'] = le.fit_transform(df.GT)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d461a5f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.GT_codeLE.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a545c7f6",
   "metadata": {},
   "source": [
    "> # Genotype Encoding using the pd.factorize method"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8dae0201",
   "metadata": {},
   "outputs": [],
   "source": [
    "df['GT_code']= pd.factorize(df.GT)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1e75e77",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.GT_code.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5430e295",
   "metadata": {},
   "source": [
    "> # Genotype Encoding using sklearn OneHotEncoder "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d6b3157",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.preprocessing import OneHotEncoder, OrdinalEncoder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a488115",
   "metadata": {},
   "outputs": [],
   "source": [
    "ohe = OneHotEncoder(sparse=False)\n",
    "df= ohe.fit_transform(df[['GT']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3b5a5c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ce2caff",
   "metadata": {},
   "outputs": [],
   "source": [
    "## To know which category is represented in the first col.,second col.,...etc\n",
    "\n",
    "ohe.categories_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46dcfe41",
   "metadata": {},
   "outputs": [],
   "source": [
    "ohe.get_feature_names_out()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e3a1c755",
   "metadata": {},
   "source": [
    "> # Genotype Encoding using sklearn ColumnTransformer\n",
    "https://www.youtube.com/watch?v=to2mukSyvLk    <<<  <u>Very informative tutorial about Encoding multiple features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0396db26",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.compose import ColumnTransformer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80e12a4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "categorical_feats = [\"GT\"]\n",
    "passthrough_feats = [\"SNP_ID\"] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b3fe03b",
   "metadata": {},
   "outputs": [],
   "source": [
    "ct = ColumnTransformer(\n",
    "    [\n",
    "        (\"onehot\", OneHotEncoder(sparse=False), categorical_feats),\n",
    "    ]\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d4126ca7",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.compose import make_column_transformer\n",
    "\n",
    "ct = make_column_transformer(    \n",
    "    (\"passthrough\", passthrough_feats),  # no transformations on the binary features    \n",
    "    (OneHotEncoder(), categorical_feats),  # OHE on categorical features\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "67c14988",
   "metadata": {},
   "outputs": [],
   "source": [
    "transformed = ct.fit_transform(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9584d514",
   "metadata": {},
   "outputs": [],
   "source": [
    "transformed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e16d5dec",
   "metadata": {},
   "outputs": [],
   "source": [
    "column_names = (\n",
    "    passthrough_feats    \n",
    "    + ct.named_transformers_[\"onehotencoder\"].get_feature_names_out().tolist()\n",
    ")\n",
    "column_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a709592a",
   "metadata": {},
   "outputs": [],
   "source": [
    "ct.named_transformers_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9882db70",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(transformed, columns=column_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b98b5708",
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = df.T\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ccd9408",
   "metadata": {},
   "outputs": [],
   "source": [
    "df1.columns= df1.iloc[0]\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d793692c",
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = df1.drop('SNP_ID')\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2431b59",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4292e0cb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f356b1ad",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43af6872",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "cad2456c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import dtale "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84c1fba5",
   "metadata": {},
   "outputs": [],
   "source": [
    "G3_sample_3  = '/home/phoebe/phd/ML_csv/GP_3/Gp3_INDEL_Variant_files/1_Gp3_Ready_INDELs_F2_VCF/sample_3_Indel_ML.csv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a831350d-4487-4f98-95bc-6ca70b431079",
   "metadata": {},
   "outputs": [],
   "source": [
    "example = pd.read_csv('/home/phoebe/phd/ML_csv/Group_3_TestGp.csv', index_col='Unnamed: 0')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "eab3bec9-5f84-4400-97c2-e5420f95eaca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>chr1-12783</th>\n",
       "      <th>chr1-13116</th>\n",
       "      <th>chr1-13118</th>\n",
       "      <th>chr1-14522</th>\n",
       "      <th>chr1-14542</th>\n",
       "      <th>chr1-14653</th>\n",
       "      <th>chr1-14775</th>\n",
       "      <th>chr1-14815</th>\n",
       "      <th>chr1-14907</th>\n",
       "      <th>...</th>\n",
       "      <th>chr22-50117010</th>\n",
       "      <th>chr22-50438148</th>\n",
       "      <th>chr22-50451702</th>\n",
       "      <th>chr22-50516420</th>\n",
       "      <th>chr22-50694007</th>\n",
       "      <th>chr22-50751547</th>\n",
       "      <th>chr22-50775822</th>\n",
       "      <th>chr22-50804107</th>\n",
       "      <th>chr22-51066632</th>\n",
       "      <th>chr22-51097011</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>G3_Sample_1_merged</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G3_Sample_2_merged</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G3_Sample_3_merged</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G3_Sample_4_merged</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>G3_Sample_5_merged</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>G3_Sample_6_merged</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G3_Sample_7_merged</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G3_Sample_8_merged</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G3_Sample_9_merged</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9 rows × 500717 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Unnamed: 0 chr1-12783 chr1-13116 chr1-13118 chr1-14522 chr1-14542  \\\n",
       "0  G3_Sample_1_merged        1/1        0/1        0/1        0/1        0/1   \n",
       "1  G3_Sample_2_merged        0/1        0/1        0/1        0/1        NaN   \n",
       "2  G3_Sample_3_merged        NaN        0/1        0/1        0/1        0/1   \n",
       "3  G3_Sample_4_merged        0/1        0/1        0/1        0/1        0/1   \n",
       "4  G3_Sample_5_merged        NaN        0/1        0/1        NaN        NaN   \n",
       "5  G3_Sample_6_merged        NaN        1/1        1/1        NaN        NaN   \n",
       "6  G3_Sample_7_merged        NaN        NaN        NaN        NaN        NaN   \n",
       "7  G3_Sample_8_merged        NaN        0/1        0/1        NaN        NaN   \n",
       "8  G3_Sample_9_merged        NaN        NaN        NaN        NaN        NaN   \n",
       "\n",
       "  chr1-14653 chr1-14775 chr1-14815 chr1-14907  ... chr22-50117010  \\\n",
       "0        0/1        NaN        NaN        0/1  ...            NaN   \n",
       "1        0/1        NaN        NaN        0/1  ...            NaN   \n",
       "2        0/1        NaN        NaN        0/1  ...            NaN   \n",
       "3        0/1        0/1        0/1        0/1  ...            NaN   \n",
       "4        0/1        NaN        NaN        0/1  ...            NaN   \n",
       "5        0/1        NaN        NaN        0/1  ...            1/1   \n",
       "6        0/1        NaN        NaN        0/1  ...            NaN   \n",
       "7        0/1        NaN        NaN        0/1  ...            NaN   \n",
       "8        1/1        NaN        0/1        0/1  ...            NaN   \n",
       "\n",
       "  chr22-50438148 chr22-50451702 chr22-50516420 chr22-50694007 chr22-50751547  \\\n",
       "0            NaN            NaN            NaN            NaN            NaN   \n",
       "1            NaN            NaN            NaN            NaN            NaN   \n",
       "2            NaN            NaN            NaN            NaN            NaN   \n",
       "3            NaN            NaN            NaN            NaN            NaN   \n",
       "4            NaN            NaN            NaN            NaN            NaN   \n",
       "5            1/1            0/1            1/1            0/1            0/1   \n",
       "6            NaN            NaN            NaN            NaN            NaN   \n",
       "7            NaN            NaN            NaN            NaN            NaN   \n",
       "8            NaN            NaN            NaN            NaN            NaN   \n",
       "\n",
       "  chr22-50775822 chr22-50804107 chr22-51066632 chr22-51097011  \n",
       "0            NaN            NaN            NaN            NaN  \n",
       "1            NaN            NaN            NaN            NaN  \n",
       "2            NaN            NaN            NaN            NaN  \n",
       "3            NaN            NaN            NaN            NaN  \n",
       "4            NaN            NaN            NaN            NaN  \n",
       "5            1/1            1/1            0/1            1/1  \n",
       "6            NaN            NaN            NaN            NaN  \n",
       "7            NaN            NaN            NaN            NaN  \n",
       "8            NaN            NaN            NaN            NaN  \n",
       "\n",
       "[9 rows x 500717 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "example\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85363624",
   "metadata": {},
   "outputs": [],
   "source": [
    "Sample_3_SNPs = pd.read_csv(G3_sample_3, index_col= 'Unnamed: 0')\n",
    "#dtale.show(Sample_3_SNPs).open_browser()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1607dbbb",
   "metadata": {},
   "outputs": [],
   "source": [
    "Sample_3_SNPs['GT'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3060b479",
   "metadata": {},
   "outputs": [],
   "source": [
    "Sample_3_SNPs['GT'] = Sample_3_SNPs['GT'].str.replace(\"1|1\",'1/1', regex=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "41287c3f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "dtale.show(example).open_browser()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4c01b72",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eebde1ad",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6df66165",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
